{"cells":[{"metadata":{},"id":"planned-twelve","cell_type":"markdown","source":"## Getting started with genomics data"},{"metadata":{},"id":"vanilla-classroom","cell_type":"markdown","source":"In this session we will start to look into some genomic data, using some of the most common tools used in the analysis of genome-wide data.\nWe ara now inside a **jupyter** notebook, we can have a look at https://jupyter.org/ to have a glimpse how jupyter is useful for data analysis.\n\nOn the left we can see a few different tabs: The \"folder\" tab allows us to explore the files in our working directory. We can see that there are ..."},{"metadata":{},"id":"deadly-musical","cell_type":"markdown","source":"Let's have a look at these files:\nWe have GreSalVL_b37_AllelesUpdatedNewMapPED.ped and GreSalVL_b37_AllelesUpdatedNewMapPED.map. These formats are used by **plink**, by far the most used software in the analysis of genotype array data.\nThe ped (for pedigree) file has the genotype data, while the map file has the information about ore Single Nucleotide Polymorphisms. We can look int it using some basic bash commands in the UNIX terminal, which is natively avaialble in MAC and Linux systems.\nGiven that the kernel of this notebook is based on python 3 we should use some magic to make some cells working exclusively in bash. Cell magic \"codes\" are simple string starting with \"%%\", which will give jupyter some extra instruction on how to set the cell we are working in."},{"metadata":{"trusted":true},"id":"three-mobile","cell_type":"code","source":"%%bash \nunzip -o plinkfilesEx1.zip","execution_count":59,"outputs":[{"output_type":"stream","text":"Archive:  plinkfilesEx1.zip\n  inflating: GreSalVL_b37_AllelesUpdatedNewMapPED.map  \n  inflating: GreSalVL_b37_AllelesUpdatedNewMapPED.ped  \n  inflating: plink                   \n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"indirect-temperature","cell_type":"code","source":"%%bash\n\nls -altrh\n","execution_count":60,"outputs":[{"output_type":"stream","text":"total 497M\n-rw-r--r-- 1 jovyan jovyan  807 Apr  4  2018 .profile\n-rw-r--r-- 1 jovyan jovyan 3.7K Apr  4  2018 .bashrc\n-rw-r--r-- 1 jovyan jovyan  220 Apr  4  2018 .bash_logout\ndrwxr-xr-x 1 root   root   4.0K May 30 06:30 ..\ndrwx------ 3 jovyan jovyan 4.0K May 30 09:34 .config\n-rwxr--r-- 1 jovyan jovyan  40M May 31 12:05 plink\n-rw-r--r-- 1 jovyan jovyan 248M May 31 12:05 GreSalVL_b37_AllelesUpdatedNewMapPED.ped\n-rw-r--r-- 1 jovyan jovyan 7.3M May 31 12:05 GreSalVL_b37_AllelesUpdatedNewMapPED.map\n-rw-r--r-- 1 jovyan jovyan  368 Jun  2 13:08 README.md\n-rw-r--r-- 1 jovyan jovyan  42M Jun  2 13:08 plinkfilesEx1.zip\ndrwxr-xr-x 8 jovyan jovyan 4.0K Jun  2 13:08 .git\ndrwxr-xr-x 3 jovyan jovyan 4.0K Jun  2 13:09 .local\ndrwxr-xr-x 3 jovyan jovyan 4.0K Jun  2 13:09 .cache\ndrwxr-xr-x 5 jovyan jovyan 4.0K Jun  2 13:11 .ipython\ndrwxr-xr-x 2 jovyan jovyan 4.0K Jun  2 13:13 .ipynb_checkpoints\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:01 GreSalVL_b37_AllelesUpdatedNewMapBED.nosex\n-rw-r--r-- 1 jovyan jovyan  16M Jun  2 15:01 GreSalVL_b37_AllelesUpdatedNewMapBED.bed\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:01 GreSalVL_b37_AllelesUpdatedNewMapBED.fam\n-rw-r--r-- 1 jovyan jovyan 8.4M Jun  2 15:01 GreSalVL_b37_AllelesUpdatedNewMapBED.bim\n-rw-r--r-- 1 jovyan jovyan 1.6K Jun  2 15:01 GreSalVL_b37_AllelesUpdatedNewMapBED.log\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam~\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed~\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim~\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.nosex\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.fam\n-rw-r--r-- 1 jovyan jovyan  15M Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bed\n-rw-r--r-- 1 jovyan jovyan 1.3K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.log\n-rw-r--r-- 1 jovyan jovyan 7.9M Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.nosex\n-rw-r--r-- 1 jovyan jovyan  13M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.lmiss\n-rw-r--r-- 1 jovyan jovyan 1.2K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.log\n-rw-r--r-- 1 jovyan jovyan  12K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.imiss\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.nosex\n-rw-r--r-- 1 jovyan jovyan   32 Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.irem\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.fam\n-rw-r--r-- 1 jovyan jovyan  15M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bed\n-rw-r--r-- 1 jovyan jovyan 1.6K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.log\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.nosex\n-rw-r--r-- 1 jovyan jovyan  11M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bed\n-rw-r--r-- 1 jovyan jovyan 3.9K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.fam\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bim\n-rw-r--r-- 1 jovyan jovyan 1.5K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.log\n-rw-r--r-- 1 jovyan jovyan 2.6K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.nosex\n-rw-r--r-- 1 jovyan jovyan  11M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bed\n-rw-r--r-- 1 jovyan jovyan 3.9K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.fam\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bim\n-rw-r--r-- 1 jovyan jovyan 1.5K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.log\n-rw-r--r-- 1 jovyan jovyan  976 Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.rel.id\n-rw-r--r-- 1 jovyan jovyan  976 Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.nosex\n-rw-r--r-- 1 jovyan jovyan 1.4M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.in\n-rw-r--r-- 1 jovyan jovyan 1.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.out\n-rw-r--r-- 1 jovyan jovyan 2.6K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.log\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.fam~\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bed~\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bim~\n-rw-r--r-- 1 jovyan jovyan 8.9K Jun  2 15:16 .jupyter-server-log.txt\n-rw-r--r-- 1 jovyan jovyan 374K Jun  2 15:16 Ex1.ipynb\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.nosex\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.fam\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bed\n-rw-r--r-- 1 jovyan jovyan 1.7K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.log\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bim\ndrwxr-xr-x 1 jovyan jovyan 4.0K Jun  2 15:17 .\n","name":"stdout"}]},{"metadata":{},"id":"elementary-objective","cell_type":"markdown","source":"Using the code `ls -altrh` we are listing all the files in our folder. Specifically:\n\n- `a` specifies that we want to list **a**ll the files\n- `l` specifies that we want a **l**ong version of the list\n- `t` specifies we want the files to be ordered by modification **t**ime\n- `r` just **r**everse the list\n- `h` makes the filesize **h**uman readable\n\nWe can just check all the oprions we can add for a given command just typing ``command --help``.\nLet's try typing `ls --help`"},{"metadata":{"trusted":true},"id":"entire-speech","cell_type":"code","source":"%%bash\nls --help","execution_count":61,"outputs":[{"output_type":"stream","text":"Usage: ls [OPTION]... [FILE]...\nList information about the FILEs (the current directory by default).\nSort entries alphabetically if none of -cftuvSUX nor --sort is specified.\n\nMandatory arguments to long options are mandatory for short options too.\n  -a, --all                  do not ignore entries starting with .\n  -A, --almost-all           do not list implied . and ..\n      --author               with -l, print the author of each file\n  -b, --escape               print C-style escapes for nongraphic characters\n      --block-size=SIZE      scale sizes by SIZE before printing them; e.g.,\n                               '--block-size=M' prints sizes in units of\n                               1,048,576 bytes; see SIZE format below\n  -B, --ignore-backups       do not list implied entries ending with ~\n  -c                         with -lt: sort by, and show, ctime (time of last\n                               modification of file status information);\n                               with -l: show ctime and sort by name;\n                               otherwise: sort by ctime, newest first\n  -C                         list entries by columns\n      --color[=WHEN]         colorize the output; WHEN can be 'always' (default\n                               if omitted), 'auto', or 'never'; more info below\n  -d, --directory            list directories themselves, not their contents\n  -D, --dired                generate output designed for Emacs' dired mode\n  -f                         do not sort, enable -aU, disable -ls --color\n  -F, --classify             append indicator (one of */=>@|) to entries\n      --file-type            likewise, except do not append '*'\n      --format=WORD          across -x, commas -m, horizontal -x, long -l,\n                               single-column -1, verbose -l, vertical -C\n      --full-time            like -l --time-style=full-iso\n  -g                         like -l, but do not list owner\n      --group-directories-first\n                             group directories before files;\n                               can be augmented with a --sort option, but any\n                               use of --sort=none (-U) disables grouping\n  -G, --no-group             in a long listing, don't print group names\n  -h, --human-readable       with -l and/or -s, print human readable sizes\n                               (e.g., 1K 234M 2G)\n      --si                   likewise, but use powers of 1000 not 1024\n  -H, --dereference-command-line\n                             follow symbolic links listed on the command line\n      --dereference-command-line-symlink-to-dir\n                             follow each command line symbolic link\n                               that points to a directory\n      --hide=PATTERN         do not list implied entries matching shell PATTERN\n                               (overridden by -a or -A)\n      --hyperlink[=WHEN]     hyperlink file names; WHEN can be 'always'\n                               (default if omitted), 'auto', or 'never'\n      --indicator-style=WORD  append indicator with style WORD to entry names:\n                               none (default), slash (-p),\n                               file-type (--file-type), classify (-F)\n  -i, --inode                print the index number of each file\n  -I, --ignore=PATTERN       do not list implied entries matching shell PATTERN\n  -k, --kibibytes            default to 1024-byte blocks for disk usage\n  -l                         use a long listing format\n  -L, --dereference          when showing file information for a symbolic\n                               link, show information for the file the link\n                               references rather than for the link itself\n  -m                         fill width with a comma separated list of entries\n  -n, --numeric-uid-gid      like -l, but list numeric user and group IDs\n  -N, --literal              print entry names without quoting\n  -o                         like -l, but do not list group information\n  -p, --indicator-style=slash\n                             append / indicator to directories\n  -q, --hide-control-chars   print ? instead of nongraphic characters\n      --show-control-chars   show nongraphic characters as-is (the default,\n                               unless program is 'ls' and output is a terminal)\n  -Q, --quote-name           enclose entry names in double quotes\n      --quoting-style=WORD   use quoting style WORD for entry names:\n                               literal, locale, shell, shell-always,\n                               shell-escape, shell-escape-always, c, escape\n  -r, --reverse              reverse order while sorting\n  -R, --recursive            list subdirectories recursively\n  -s, --size                 print the allocated size of each file, in blocks\n  -S                         sort by file size, largest first\n      --sort=WORD            sort by WORD instead of name: none (-U), size (-S),\n                               time (-t), version (-v), extension (-X)\n      --time=WORD            with -l, show time as WORD instead of default\n                               modification time: atime or access or use (-u);\n                               ctime or status (-c); also use specified time\n                               as sort key if --sort=time (newest first)\n      --time-style=STYLE     with -l, show times using style STYLE:\n                               full-iso, long-iso, iso, locale, or +FORMAT;\n                               FORMAT is interpreted like in 'date'; if FORMAT\n                               is FORMAT1<newline>FORMAT2, then FORMAT1 applies\n                               to non-recent files and FORMAT2 to recent files;\n                               if STYLE is prefixed with 'posix-', STYLE\n                               takes effect only outside the POSIX locale\n  -t                         sort by modification time, newest first\n  -T, --tabsize=COLS         assume tab stops at each COLS instead of 8\n  -u                         with -lt: sort by, and show, access time;\n                               with -l: show access time and sort by name;\n                               otherwise: sort by access time, newest first\n  -U                         do not sort; list entries in directory order\n  -v                         natural sort of (version) numbers within text\n  -w, --width=COLS           set output width to COLS.  0 means no limit\n  -x                         list entries by lines instead of by columns\n  -X                         sort alphabetically by entry extension\n  -Z, --context              print any security context of each file\n  -1                         list one file per line.  Avoid '\\n' with -q or -b\n      --help     display this help and exit\n      --version  output version information and exit\n\nThe SIZE argument is an integer and optional unit (example: 10K is 10*1024).\nUnits are K,M,G,T,P,E,Z,Y (powers of 1024) or KB,MB,... (powers of 1000).\n\nUsing color to distinguish file types is disabled both by default and\nwith --color=never.  With --color=auto, ls emits color codes only when\nstandard output is connected to a terminal.  The LS_COLORS environment\nvariable can change the settings.  Use the dircolors command to set it.\n\nExit status:\n 0  if OK,\n 1  if minor problems (e.g., cannot access subdirectory),\n 2  if serious trouble (e.g., cannot access command-line argument).\n\nGNU coreutils online help: <http://www.gnu.org/software/coreutils/>\nFull documentation at: <http://www.gnu.org/software/coreutils/ls>\nor available locally via: info '(coreutils) ls invocation'\n","name":"stdout"}]},{"metadata":{},"id":"listed-computer","cell_type":"markdown","source":"### 1. A first look into genetic data"},{"metadata":{},"id":"logical-artwork","cell_type":"markdown","source":"Ok, we are now ready to have a look at the genetic data using some basic bash command. Let's start with the map file. We can print the frst **n** lines using `head -n filename` "},{"metadata":{"trusted":true},"id":"turkish-uganda","cell_type":"code","source":"%%bash\n\nhead -n 10 GreSalVL_b37_AllelesUpdatedNewMapPED.map\n","execution_count":62,"outputs":[{"output_type":"stream","text":"0\t0_0\t0\t0\n0\t0_0_1\t0\t0\n0\t0_0_2\t0\t0\n0\t0_0_3\t0\t0\n0\t0_0_4\t0\t0\n0\t0_0_5\t0\t0\n0\t0_0_6\t0\t0\n0\t0_0_7\t0\t0\n0\t0_0_8\t0\t0\n0\t0_0_9\t0\t0\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"dirty-sleeping","cell_type":"code","source":"%%bash\n\ntail -n 10 GreSalVL_b37_AllelesUpdatedNewMapPED.map","execution_count":63,"outputs":[{"output_type":"stream","text":"25\t25_155228901\t0\t155228901\n25\t25_155228995\t0\t155228995\n25\t25_155229003\t0\t155229003\n25\t25_155229189\t0\t155229189\n25\t25_155230350\t0\t155230350\n25\t25_155230365\t0\t155230365\n25\t25_155232838\t0\t155232838\n25\t25_155233098\t0\t155233098\n25\t25_155233846\t0\t155233846\n25\t25_155234707\t0\t155234707\n","name":"stdout"}]},{"metadata":{},"id":"worldwide-duration","cell_type":"markdown","source":"By default, each line of the MAP file describes a single marker and must contain exactly 4 columns:\n- chromosome (1-22, X, Y or 0 if unplaced)\n- rs# or snp identifier\n- Genetic distance (morgans)\n- Base-pair position (bp units)"},{"metadata":{},"id":"paperback-sustainability","cell_type":"markdown","source":"We can count the total number of lines, and thus of SNPs using the command `wc -l <filename>`"},{"metadata":{"trusted":true},"id":"improved-estimate","cell_type":"code","source":"%%bash\n\nwc -l GreSalVL_b37_AllelesUpdatedNewMapPED.map","execution_count":64,"outputs":[{"output_type":"stream","text":"299140 GreSalVL_b37_AllelesUpdatedNewMapPED.map\n","name":"stdout"}]},{"metadata":{},"id":"developed-adolescent","cell_type":"markdown","source":"In our file there are ~300,000 SNPs. Let's have a look at the ped file. How many lines are stored in it?"},{"metadata":{"trusted":true},"id":"blocked-stamp","cell_type":"code","source":"%%bash\n\n#Let's print the number of lines in the ped file","execution_count":65,"outputs":[]},{"metadata":{},"id":"unexpected-voluntary","cell_type":"markdown","source":"We can also count the number of columns in our ped file using `awk`, which is a **domain specific language** for text processing and manipulation"},{"metadata":{"trusted":true},"id":"actual-basement","cell_type":"code","source":"%%bash\nawk '{print NF}' GreSalVL_b37_AllelesUpdatedNewMapPED.ped | sort -nu ","execution_count":66,"outputs":[{"output_type":"stream","text":"awk: program limit exceeded: maximum number of fields size=32767\n\tFILENAME=\"GreSalVL_b37_AllelesUpdatedNewMapPED.ped\" FNR=1 NR=1\n","name":"stderr"}]},{"metadata":{},"id":"available-fraction","cell_type":"markdown","source":"Given the high number of columns, it's better not to show only the firs 10 rows, but also the first n columns, that will be selected using the `cut` command.\nWe will combine the two commands using `|` (pipe), which as shown in the previous cell, combines different commands. "},{"metadata":{"trusted":true},"id":"miniature-remove","cell_type":"code","source":"%%bash\n\nhead -n 10 GreSalVL_b37_AllelesUpdatedNewMapPED.ped | cut -f 1-30 -d \" \"","execution_count":67,"outputs":[{"output_type":"stream","text":"PuBa GSM1094884 0 0 0 2 0 0 A A G G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094885 0 0 0 2 0 0 A A G G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094886 0 0 0 1 0 0 A A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094887 0 0 0 1 0 0 A A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094888 0 0 0 1 0 0 G A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094889 0 0 0 1 0 0 A A A A C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094890 0 0 0 1 0 0 G A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094891 0 0 0 1 0 0 A A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094892 0 0 0 1 0 0 A A A G C C 0 0 0 0 G G G G G G 0 0 G G G G\nPuBa GSM1094893 0 0 0 1 0 0 A A G G C C 0 0 0 0 G G G G G G 0 0 G G G G\n","name":"stdout"}]},{"metadata":{},"id":"danish-highlight","cell_type":"markdown","source":"The PED file is a white-space (space or tab) delimited file: the first six columns are mandatory:\n- Family ID\n- Individual ID\n- Paternal ID\n- Maternal ID\n- Sex (1=male; 2=female; other=unknown)\n- Phenotype\n\nfrom the 7th column the actual genotype for individuals is shown, with **0** or **-9** indicating missing data. This is a very simple format of genotype file, which is actually very resource consuming as can be seen by `ls`"},{"metadata":{"trusted":true},"id":"speaking-governor","cell_type":"code","source":"%%bash\nmyped=GreSalVL_b37_AllelesUpdatedNewMapPED.ped \nmb=`ls -altrh $myped |cut -f 5 -d \" \"`\n\necho \"The file $myped size is $mb\"","execution_count":68,"outputs":[{"output_type":"stream","text":"The file GreSalVL_b37_AllelesUpdatedNewMapPED.ped size is 248M\n","name":"stdout"}]},{"metadata":{},"id":"southern-potter","cell_type":"markdown","source":"Luckily, **plink** offers the possibility to convert our ped file in  binary files **bed** which codes as follows using **binary notation**:\n    \nFor the genotype data, each byte encodes up to four genotypes (2 bits per genoytpe). The coding is\n- 00  Homozygote \"1\"/\"1\"\n- 01  Heterozygote\n- 11  Homozygote \"2\"/\"2\"\n- 10  Missing genotype"},{"metadata":{"trusted":true},"id":"documented-jesus","cell_type":"code","source":"%%bash\n./plink --help","execution_count":69,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n\nIn the command line flag definitions that follow,\n  * <angle brackets> denote a required parameter, where the text between the\n    angle brackets describes its nature.\n  * ['square brackets + single-quotes'] denotes an optional modifier.  Use the\n    EXACT text in the quotes.\n  * [{bar|separated|braced|bracketed|values}] denotes a collection of mutually\n    exclusive optional modifiers (again, the exact text must be used).  When\n    there are no outer square brackets, one of the choices must be selected.\n  * ['quoted_text='<description of value>] denotes an optional modifier that\n    must begin with the quoted text, and be followed by a value with no\n    whitespace in between.  '|' may also be used here to indicate mutually\n    exclusive options.\n  * [square brackets without quotes or braces] denote an optional parameter,\n    where the text between the brackets describes its nature.\n  * An ellipsis (...) indicates that you may enter multiple parameters of the\n    specified type.\n\n  plink <input flag(s)...> [command flag(s)...] [other flag(s)...]\n  plink --help [flag name(s)...]\n\nMost PLINK runs require exactly one main input fileset.  The following flags\nare available for defining its form and location:\n\n  --bfile [prefix] : Specify .bed + .bim + .fam prefix (default 'plink').\n  --bed <filename> : Specify full name of .bed file.\n  --bim <filename> : Specify full name of .bim file.\n  --fam <filename> : Specify full name of .fam file.\n\n  --keep-autoconv  : With --file/--tfile/--lfile/--vcf/--bcf/--data/--23file,\n                     don't delete autogenerated binary fileset at end of run.\n\n  --file [prefix]  : Specify .ped + .map filename prefix (default 'plink').\n  --ped <filename> : Specify full name of .ped file.\n  --map <filename> : Specify full name of .map file.\n\n  --no-fid         : .fam/.ped file does not contain column 1 (family ID).\n  --no-parents     : .fam/.ped file does not contain columns 3-4 (parents).\n  --no-sex         : .fam/.ped file does not contain column 5 (sex).\n  --no-pheno       : .fam/.ped file does not contain column 6 (phenotype).\n\n  --tfile [prefix] : Specify .tped + .tfam filename prefix (default 'plink').\n  --tped <fname>   : Specify full name of .tped file.\n  --tfam <fname>   : Specify full name of .tfam file.\n\n  --lfile [prefix] : Specify .lgen + .map + .fam (long-format fileset) prefix.\n  --lgen <fname>   : Specify full name of .lgen file.\n  --reference <fn> : Specify default allele file accompanying .lgen input.\n  --allele-count   : When used with --lfile/--lgen + --reference, specifies\n                     that the .lgen file contains reference allele counts.\n\n  --vcf <filename> : Specify full name of .vcf or .vcf.gz file.\n  --bcf <filename> : Specify full name of BCF2 file.\n\n  --data [prefix]  : Specify Oxford .gen + .sample prefix (default 'plink').\n  --gen <filename> : Specify full name of .gen or .gen.gz file.\n  --bgen <f> ['snpid-chr'] : Specify full name of .bgen file.\n  --sample <fname> : Specify full name of .sample file.\n\n  --23file <fname> [FID] [IID] [sex] [pheno] [pat. ID] [mat. ID] :\n    Specify 23andMe input file.\n\n  --grm-gz [prfx]  : Specify .grm.gz + .grm.id (GCTA rel. matrix) prefix.\n  --grm-bin [prfx] : Specify .grm.bin + .grm.N.bin + .grm.id (GCTA triangular\n                     binary relationship matrix) filename prefix.\n\n  --dummy <sample ct> <SNP ct> [missing geno freq] [missing pheno freq]\n          [{acgt | 1234 | 12}] ['scalar-pheno']\n    This generates a fake input dataset with the specified number of samples\n    and SNPs.  By default, the missing genotype and phenotype frequencies are\n    zero, and genotypes are As and Bs (change the latter with\n    'acgt'/'1234'/'12').  The 'scalar-pheno' modifier causes a normally\n    distributed scalar phenotype to be generated instead of a binary one.\n\n  --simulate <simulation parameter file> [{tags | haps}] [{acgt | 1234 | 12}]\n  --simulate-qt <sim. parameter file> [{tags | haps}] [{acgt | 1234 | 12}]\n    --simulate generates a fake input dataset with disease-associated SNPs,\n    while --simulate-qt generates a dataset with quantitative trait loci.\n\nOutput files have names of the form 'plink.<extension>' by default.  You can\nchange the 'plink' prefix with\n\n  --out <prefix>   : Specify prefix for output files.\n\nMost runs also require at least one of the following commands:\n\n  --make-bed\n    Create a new binary fileset.  Unlike the automatic text-to-binary\n    converters (which only heed chromosome filters), this supports all of\n    PLINK's filtering flags.\n  --make-just-bim\n  --make-just-fam\n    Variants of --make-bed which only write a new .bim or .fam file.  Can be\n    used with only .bim/.fam input.\n    USE THESE CAUTIOUSLY.  It is very easy to desynchronize your binary\n    genotype data and your .bim/.fam indexes if you use these commands\n    improperly.  If you have any doubt, stick with --make-bed.\n\n  --recode <output format> [{01 | 12}] [{tab | tabx | spacex | bgz | gen-gz}]\n           ['include-alt'] ['omit-nonmale-y']\n    Create a new text fileset with all filters applied.  The following output\n    formats are supported:\n    * '23': 23andMe 4-column format.  This can only be used on a single\n      sample's data (--keep may be handy), and does not support multicharacter\n      allele codes.\n    * 'A': Sample-major additive (0/1/2) coding, suitable for loading from R.\n      If you need uncounted alleles to be named in the header line, add the\n      'include-alt' modifier.\n    * 'AD': Sample-major additive (0/1/2) + dominant (het=1/hom=0) coding.\n      Also supports 'include-alt'.\n    * 'A-transpose': Variant-major 0/1/2.\n    * 'beagle': Unphased per-autosome .dat and .map files, readable by early\n      BEAGLE versions.\n    * 'beagle-nomap': Single .beagle.dat file.\n    * 'bimbam': Regular BIMBAM format.\n    * 'bimbam-1chr': BIMBAM format, with a two-column .pos.txt file.  Does not\n      support multiple chromosomes.\n    * 'fastphase': Per-chromosome fastPHASE files, with\n      .chr-<chr #>.recode.phase.inp filename extensions.\n    * 'fastphase-1chr': Single .recode.phase.inp file.  Does not support\n      multiple chromosomes.\n    * 'HV': Per-chromosome Haploview files, with .chr-<chr #>{.ped,.info}\n      filename extensions.\n    * 'HV-1chr': Single Haploview .ped + .info file pair.  Does not support\n      multiple chromosomes.\n    * 'lgen': PLINK 1 long-format (.lgen + .fam + .map), loadable with --lfile.\n    * 'lgen-ref': .lgen + .fam + .map + .ref, loadable with --lfile +\n       --reference.\n    * 'list': Single genotype-based list, up to 4 lines per variant.  To omit\n      nonmale genotypes on the Y chromosome, add the 'omit-nonmale-y' modifier.\n    * 'rlist': .rlist + .fam + .map fileset, where the .rlist file is a\n      genotype-based list which omits the most common genotype for each\n      variant.  Also supports 'omit-nonmale-y'.\n    * 'oxford': Oxford-format .gen + .sample.  With the 'gen-gz' modifier, the\n      .gen file is gzipped.\n    * 'ped': PLINK 1 sample-major (.ped + .map), loadable with --file.\n    * 'compound-genotypes': Same as 'ped', except that the space between each\n      pair of same-variant allele codes is removed.\n    * 'structure': Structure-format.\n    * 'transpose': PLINK 1 variant-major (.tped + .tfam), loadable with\n      --tfile.\n    * 'vcf', 'vcf-fid', 'vcf-iid': VCFv4.2.  'vcf-fid' and 'vcf-iid' cause\n      family IDs or within-family IDs respectively to be used for the sample\n      IDs in the last header row, while 'vcf' merges both IDs and puts an\n      underscore between them.  If the 'bgz' modifier is added, the VCF file is\n      block-gzipped.\n      The A2 allele is saved as the reference and normally flagged as not based\n      on a real reference genome (INFO:PR).  When it is important for reference\n      alleles to be correct, you'll also want to include --a2-allele and\n      --real-ref-alleles in your command.\n    In addition,\n    * The '12' modifier causes A1 (usually minor) alleles to be coded as '1'\n      and A2 alleles to be coded as '2', while '01' maps A1 -> 0 and A2 -> 1.\n    * The 'tab' modifier makes the output mostly tab-delimited instead of\n      mostly space-delimited.  'tabx' and 'spacex' force all tabs and all\n      spaces, respectively.\n\n  --flip-scan ['verbose']\n    (alias: --flipscan)\n    LD-based scan for case/control strand inconsistency.\n\n  --write-covar\n    If a --covar file is loaded, --make-bed/--make-just-fam and --recode\n    automatically generate an updated version (with all filters applied).\n    However, if you do not wish to simultaneously generate a new genotype file,\n    you can use --write-covar to just produce a pruned covariate file.\n\n  --write-cluster ['omit-unassigned']\n    If clusters are specified with --within/--family, this generates a new\n    cluster file (with all filters applied).  The 'omit-unassigned' modifier\n    causes unclustered samples to be omitted from the file; otherwise their\n    cluster is 'NA'.\n\n  --write-set\n  --set-table\n    If sets have been defined, --write-set dumps 'END'-terminated set\n    membership lists to <output prefix>.set, while --set-table writes a\n    variant-by-set membership table to <output prefix>.set.table.\n\n  --merge <.ped filename> <.map filename>\n  --merge <text fileset prefix>\n  --bmerge <.bed filename> <.bim filename> <.fam filename>\n  --bmerge <binary fileset prefix>\n    Merge the given fileset with the initially loaded fileset, writing the\n    result to <output prefix>.bed + .bim + .fam.  (It is no longer necessary to\n    simultaneously specify --make-bed.)\n  --merge-list <filename>\n    Merge all filesets named in the text file with the reference fileset, if\n    one was specified.  (However, this can also be used *without* a reference;\n    in that case, the newly created fileset is then treated as the reference by\n    most other PLINK operations.)  The text file is interpreted as follows:\n    * If a line contains only one name, it is assumed to be the prefix for a\n      binary fileset.\n    * If a line contains exactly two names, they are assumed to be the full\n      filenames for a text fileset (.ped first, then .map).\n    * If a line contains exactly three names, they are assumed to be the full\n      filenames for a binary fileset (.bed, then .bim, then .fam).\n\n  --write-snplist\n  --list-23-indels\n    --write-snplist writes a .snplist file listing the names of all variants\n    which pass the filters and inclusion thresholds you've specified, while\n    --list-23-indels writes the subset with 23andMe-style indel calls (D/I\n    allele codes).\n\n  --list-duplicate-vars ['require-same-ref'] ['ids-only'] ['suppress-first']\n    --list-duplicate-vars writes a .dupvar file describing all groups of\n    variants with matching positions and allele codes.\n    * By default, A1/A2 allele assignments are ignored; use 'require-same-ref'\n      to override this.\n    * Normally, the report contains position and allele codes.  To remove them\n      (and produce a file directly usable with e.g. --extract/--exclude), use\n      'ids-only'.  Note that this command will fail in 'ids-only' mode if any\n      of the reported IDs are not unique.\n    * 'suppress-first' causes the first variant ID in each group to be omitted\n      from the report.\n\n  --freq [{counts | case-control}] ['gz']\n  --freqx ['gz']\n    --freq generates a basic allele frequency (or count, if the 'counts'\n    modifier is present) report.  This can be combined with --within/--family\n    to produce a cluster-stratified allele frequency/count report instead, or\n    the 'case-control' modifier to report case and control allele frequencies\n    separately.\n    --freqx generates a more detailed genotype count report, designed for use\n    with --read-freq.\n\n  --missing ['gz']\n    Generate sample- and variant-based missing data reports.  If clusters are\n    defined, the variant-based report is cluster-stratified.  'gz' causes the\n    output files to be gzipped.\n    Unlike most other commands, this doesn't treat het. haploids as missing.\n\n  --test-mishap\n    Check for association between missing calls and flanking haplotypes.\n\n  --hardy ['midp'] ['gz']\n    Generate a Hardy-Weinberg exact test p-value report.  (This does NOT\n    simultaneously filter on the p-value any more; use --hwe for that.)  With\n    the 'midp' modifier, the test applies the mid-p adjustment described in\n    Graffelman J, Moreno V (2013) The mid p-value in exact tests for\n    Hardy-Weinberg Equilibrium.\n\n  --mendel ['summaries-only']\n    Generate a Mendel error report.  The 'summaries-only' modifier causes the\n    .mendel file (listing every single error) to be skipped.\n\n  --het ['small-sample'] ['gz']\n  --ibc\n    Estimate inbreeding coefficients.  --het reports method-of-moments\n    estimates, while --ibc calculates all three values described in Yang J, Lee\n    SH, Goddard ME and Visscher PM (2011) GCTA: A Tool for Genome-wide Complex\n    Trait Analysis.  (That paper also describes the relationship matrix\n    computation we reimplement.)\n    * These functions require decent MAF estimates.  If there are very few\n      samples in your immediate fileset, --read-freq is practically mandatory\n      since imputed MAFs are wildly inaccurate in that case.\n    * They also assume the marker set is in approximate linkage equilibrium.\n    * By default, --het omits the n/(n-1) multiplier in Nei's expected\n      homozygosity formula.  The 'small-sample' modifier causes it to be\n      included, while forcing --het to use MAFs imputed from founders in the\n      immediate dataset.\n\n  --check-sex [female max F] [male min F]\n  --check-sex ycount [female max F] [male min F] [female max Y obs]\n                     [male min Y obs]\n  --check-sex y-only [female max Y obs] [male min Y obs]\n  --impute-sex [female max F] [male min F]\n  --impute-sex ycount [female max F] [male min F] [female max Y obs]\n                      [male min Y obs]\n  --impute-sex y-only [female max Y obs] [male min Y obs]\n    --check-sex normally compares sex assignments in the input dataset with\n    those imputed from X chromosome inbreeding coefficients.\n    * Make sure that the X chromosome pseudo-autosomal region has been split\n      off (with e.g. --split-x) before using this.\n    * You also need decent MAF estimates (so, with very few samples in your\n      immediate fileset, use --read-freq), and your marker set should be in\n      approximate linkage equilibrium.\n    * By default, F estimates smaller than 0.2 yield female calls, and values\n      larger than 0.8 yield male calls.  If you pass numeric parameter(s) to\n      --check-sex, the first two control these thresholds.\n    There are now two modes which consider Y chromosome data.\n    * In 'ycount' mode, gender is still imputed from the X chromosome, but\n      female calls are downgraded to ambiguous whenever more than 0 nonmissing\n      Y genotypes are present, and male calls are downgraded when fewer than 0\n      are present.  (Note that these are counts, not rates.)  These thresholds\n      are controllable with --check-sex ycount's optional 3rd and 4th numeric\n      parameters.\n    * In 'y-only' mode, gender is imputed from nonmissing Y genotype counts.\n      The male minimum threshold defaults to 1 instead of zero in this case.\n    --impute-sex changes sex assignments to the imputed values, and is\n    otherwise identical to --check-sex.  It must be used with\n    --make-bed/--recode/--write-covar.\n\n  --fst ['case-control']\n    (alias: --Fst)\n    Estimate Wright's Fst for each autosomal diploid variant using the method\n    introduced in Weir BS, Cockerham CC (1984) Estimating F-statistics for the\n    analysis of population structure, given a set of subpopulations defined via\n    --within.  Raw and weighted global means are also reported.\n    * If you're interested in the global means, it is usually best to perform\n      this calculation on a marker set in approximate linkage equilibrium.\n    * If you have only two subpopulations, you can represent them with\n      case/control status and use the 'case-control' modifier.\n\n  --indep <window size>['kb'] <step size (variant ct)> <VIF threshold>\n  --indep-pairwise <window size>['kb'] <step size (variant ct)> <r^2 threshold>\n  --indep-pairphase <window size>['kb'] <step size (variant ct)> <r^2 thresh>\n    Generate a list of markers in approximate linkage equilibrium.  With the\n    'kb' modifier, the window size is in kilobase instead of variant count\n    units.  (Pre-'kb' space is optional, i.e. \"--indep-pairwise 500 kb 5 0.5\"\n    and \"--indep-pairwise 500kb 5 0.5\" have the same effect.)\n    Note that you need to rerun PLINK using --extract or --exclude on the\n    .prune.in/.prune.out file to apply the list to another computation.\n\n  --r [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]\n      ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs']\n      ['yes-really']\n  --r2 [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]\n       ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs']\n       ['yes-really']\n    LD statistic reports.  --r yields raw inter-variant correlations, while\n    --r2 reports their squares.  You can request results for all pairs in\n    matrix format (if you specify 'bin' or one of the shape modifiers), all\n    pairs in table format ('inter-chr'), or a limited window in table format\n    (default).\n    * The 'gz' modifier causes the output text file to be gzipped.\n    * 'bin' causes the output matrix to be written in double-precision binary\n      format, while 'bin4' specifics single-precision binary.  The matrix is\n      square if no shape is explicitly specified.\n    * By default, text matrices are tab-delimited; 'spaces' switches this.\n    * 'in-phase' adds a column with in-phase allele pairs to table-formatted\n      reports.  (This cannot be used with very long allele codes.)\n    * 'dprime' adds the absolute value of Lewontin's D-prime statistic to\n      table-formatted reports, and forces both r/r^2 and D-prime to be based on\n      the maximum likelihood solution to the cubic equation discussed in Gaunt\n      T, Rodriguez S, Day I (2007) Cubic exact solutions for the estimation of\n      pairwise haplotype frequencies.\n      'dprime-signed' keeps the sign, while 'd' skips division by D_{max}.\n    * 'with-freqs' adds MAF columns to table-formatted reports.\n    * Since the resulting file can easily be huge, you're required to add the\n      'yes-really' modifier when requesting an unfiltered, non-distributed all\n      pairs computation on more than 400k variants.\n    * These computations can be subdivided with --parallel (even when the\n      'square' modifier is active).\n  --ld <variant ID> <variant ID> ['hwe-midp']\n    This displays haplotype frequencies, r^2, and D' for a single pair of\n    variants.  When there are multiple biologically possible solutions to the\n    haplotype frequency cubic equation, all are displayed (instead of just the\n    maximum likelihood solution identified by --r/--r2), along with HWE exact\n    test statistics.\n\n  --show-tags <filename>\n  --show-tags all\n    * If a file is specified, list all variants which tag at least one variant\n      named in the file.  (This will normally be a superset of the original\n      list, since a variant is considered to tag itself here.)\n    * If 'all' mode is specified, for each variant, each *other* variant which\n      tags it is reported.\n\n  --blocks ['no-pheno-req'] ['no-small-max-span']\n    Estimate haplotype blocks, via Haploview's interpretation of the block\n    definition suggested by Gabriel S et al. (2002) The Structure of Haplotype\n    Blocks in the Human Genome.\n    * Normally, samples with missing phenotypes are not considered by this\n      computation; the 'no-pheno-req' modifier lifts this restriction.\n    * Normally, size-2 blocks may not span more than 20kb, and size-3 blocks\n      are limited to 30kb.  The 'no-small-max-span' modifier removes these\n      limits.\n    The .blocks file is valid input for PLINK 1.07's --hap command.  However,\n    the --hap... family of flags has not been reimplemented in PLINK 1.9 due to\n    poor phasing accuracy relative to other software; for now, we recommend\n    using BEAGLE instead of PLINK for case/control haplotype association\n    analysis.  (You can use \"--recode beagle\" to export data to BEAGLE 3.3.)\n    We apologize for the inconvenience, and plan to develop variants of the\n    --hap... flags which handle pre-phased data effectively.\n\n  --distance [{square | square0 | triangle}] [{gz | bin | bin4}] ['ibs']\n             ['1-ibs'] ['allele-ct'] ['flat-missing']\n    Write a lower-triangular tab-delimited table of (weighted) genomic\n    distances in allele count units to <output prefix>.dist, and a list of the\n    corresponding sample IDs to <output prefix>.dist.id.  The first row of the\n    .dist file contains a single <genome 1-genome 2> distance, the second row\n    has the <genome 1-genome 3> and <genome 2-genome 3> distances in that\n    order, etc.\n    * It is usually best to perform this calculation on a marker set in\n      approximate linkage equilibrium.\n    * If the 'square' or 'square0' modifier is present, a square matrix is\n      written instead; 'square0' fills the upper right triangle with zeroes.\n    * If the 'gz' modifier is present, a compressed .dist.gz file is written\n      instead of a plain text file.\n    * If the 'bin' modifier is present, a binary (square) matrix of\n      double-precision floating point values, suitable for loading from R, is\n      instead written to <output prefix>.dist.bin.  ('bin4' specifies\n      single-precision numbers instead.)  This can be combined with 'square0'\n      if you still want the upper right zeroed out, or 'triangle' if you don't\n      want to pad the upper right at all.\n    * If the 'ibs' modifier is present, an identity-by-state matrix is written\n      to <output prefix>.mibs.  '1-ibs' causes distances expressed as genomic\n      proportions (i.e. 1 - IBS) to be written to <output prefix>.mdist.\n      Combine with 'allele-ct' if you want to generate the usual .dist file as\n      well.\n    * By default, distance rescaling in the presence of missing genotype calls\n      is sensitive to allele count distributions: if variant A contributes, on\n      average, twice as much to other pairwise distances as variant B, a\n      missing call at variant A will result in twice as large of a missingness\n      correction.  To turn this off (because e.g. your missing calls are highly\n      nonrandom), use the 'flat-missing' modifier.\n    * The computation can be subdivided with --parallel.\n  --distance-matrix\n  --ibs-matrix\n    These deprecated commands are equivalent to \"--distance 1-ibs flat-missing\n    square\" and \"--distance ibs flat-missing square\", respectively, except that\n    they generate space- instead of tab-delimited text matrices.\n\n  --make-rel [{square | square0 | triangle}] [{gz | bin | bin4}]\n             [{cov | ibc2 | ibc3}]\n    Write a lower-triangular variance-standardized realized relationship matrix\n    to <output prefix>.rel, and corresponding IDs to <output prefix>.rel.id.\n    * It is usually best to perform this calculation on a marker set in\n      approximate linkage equilibrium.\n    * 'square', 'square0', 'triangle', 'gz', 'bin', and 'bin4' act as they do\n      on --distance.\n    * The 'cov' modifier removes the variance standardization step, causing a\n      covariance matrix to be calculated instead.\n    * By default, the diagonal elements in the relationship matrix are based on\n      --ibc's Fhat1; use the 'ibc2' or 'ibc3' modifiers to base them on Fhat2\n      or Fhat3 instead.\n    * The computation can be subdivided with --parallel.\n  --make-grm-gz ['no-gz'] [{cov | ibc2 | ibc3}]\n  --make-grm-bin [{cov | ibc2 | ibc3}]\n    --make-grm-gz writes the relationships in GCTA's original gzipped list\n    format, which describes one pair per line, while --make-grm-bin writes them\n    in GCTA 1.1+'s single-precision triangular binary format.  Note that these\n    formats explicitly report the number of valid observations (where neither\n    sample has a missing call) for each pair, which is useful input for some\n    scripts.\n    These computations can be subdivided with --parallel.\n\n  --rel-cutoff [val]\n    (alias: --grm-cutoff)\n    Exclude one member of each pair of samples with relatedness greater than\n    the given cutoff value (default 0.025).  If no later operation will cause\n    the list of remaining samples to be written to disk, this will save it to\n    <output prefix>.rel.id.\n    Note that maximizing the remaining sample size is equivalent to the NP-hard\n    maximum independent set problem, so we use a greedy algorithm instead of\n    guaranteeing optimality.  (Use the --make-rel and --keep/--remove flags if\n    you want to try to do better.)\n\n  --ibs-test [permutation count]\n  --groupdist [iters] [d]\n    Given case/control phenotype data, these commands consider three subsets of\n    the distance matrix: pairs of affected samples, affected-unaffected pairs,\n    and pairs of unaffected samples.  Each of these subsets has a distribution\n    of pairwise genomic distances; --ibs-test uses permutation to estimate\n    p-values re: which types of pairs are most similar, while --groupdist\n    focuses on the differences between the centers of these distributions and\n    estimates standard errors via delete-d jackknife.\n\n  --regress-distance [iters] [d]\n    Linear regression of pairwise genomic distances on pairwise average\n    phenotypes and vice versa, using delete-d jackknife for standard errors.  A\n    scalar phenotype is required.\n    * With less than two parameters, d is set to <number of people>^0.6 rounded\n      down.  With no parameters, 100k iterations are run.\n  --regress-rel [iters] [d]\n    Linear regression of pairwise genomic relationships on pairwise average\n    phenotypes, and vice versa.  Defaults for iters and d are the same as for\n    --regress-distance.\n\n  --genome ['gz'] ['rel-check'] ['full'] ['unbounded'] ['nudge']\n    Generate an identity-by-descent report.\n    * It is usually best to perform this calculation on a marker set in\n      approximate linkage equilibrium.\n    * The 'rel-check' modifier excludes pairs of samples with different FIDs\n      from the final report.\n    * 'full' adds raw pairwise comparison data to the report.\n    * The P(IBD=0/1/2) estimator employed by this command sometimes yields\n      numbers outside the range [0,1]; by default, these are clipped.  The\n      'unbounded' modifier turns off this clipping.\n    * Then, when PI_HAT^2 < P(IBD=2), 'nudge' adjusts the final P(IBD=0/1/2)\n      estimates to a theoretically possible configuration.\n    * The computation can be subdivided with --parallel.\n\n  --homozyg [{group | group-verbose}] ['consensus-match'] ['extend']\n            ['subtract-1-from-lengths']\n  --homozyg-snp <min var count>\n  --homozyg-kb <min length>\n  --homozyg-density <max inverse density (kb/var)>\n  --homozyg-gap <max internal gap kb length>\n  --homozyg-het <max hets>\n  --homozyg-window-snp <scanning window size>\n  --homozyg-window-het <max hets in scanning window hit>\n  --homozyg-window-missing <max missing calls in scanning window hit>\n  --homozyg-window-threshold <min scanning window hit rate>\n    These commands request a set of run-of-homozygosity reports, and allow you\n    to customize how they are generated.\n    * If you're satisfied with all the default settings described below, just\n      use --homozyg with no modifiers.  Otherwise, --homozyg lets you change a\n      few binary settings:\n      * 'group[-verbose]' adds a report on pools of overlapping runs of\n        homozygosity.  (Automatically set when --homozyg-match is present.)\n      * With 'group[-verbose]', 'consensus-match' causes pairwise segmental\n        matches to be called based on the variants in the pool's consensus\n        segment, rather than the variants in the pairwise intersection.\n      * Due to how the scanning window algorithm works, it is possible for a\n        reported ROH to be adjacent to a few homozygous variants.  The 'extend'\n        modifier causes them to be included in the reported ROH if that\n        wouldn't cause a violation of the --homozyg-density bound.\n      * By default, segment bp lengths are calculated as <end bp position> -\n        <start bp position> + 1.  Therefore, reports normally differ slightly\n        from PLINK 1.07, which does not add 1 at the end.  For testing\n        purposes, you can use the 'subtract-1-from-lengths' modifier to apply\n        the old formula.\n    * By default, only runs of homozygosity containing at least 100 variants,\n      and of total length >= 1000 kilobases, are noted.  You can change these\n      minimums with --homozyg-snp and --homozyg-kb, respectively.\n    * By default, a ROH must have at least one variant per 50 kb on average;\n      change this bound with --homozyg-density.\n    * By default, if two consecutive variants are more than 1000 kb apart, they\n      cannot be in the same ROH; change this bound with --homozyg-gap.\n    * By default, a ROH can contain an unlimited number of heterozygous calls;\n      you can impose a limit with --homozyg-het.\n    * By default, the scanning window contains 50 variants; change this with\n      --homozyg-window-snp.\n    * By default, a scanning window hit can contain at most 1 heterozygous\n      call and 5 missing calls; change these limits with --homozyg-window-het\n      and --homozyg-window-missing, respectively.\n    * By default, for a variant to be eligible for inclusion in a ROH, the hit\n      rate of all scanning windows containing the variant must be at least\n      0.05; change this threshold with --homozyg-window-threshold.\n\n  --cluster ['cc'] [{group-avg | old-tiebreaks}] ['missing'] ['only2']\n    Cluster samples using a pairwise similarity statistic (normally IBS).\n    * The 'cc' modifier forces every cluster to have at least one case and one\n      control.\n    * The 'group-avg' modifier causes clusters to be joined based on average\n      instead of minimum pairwise similarity.\n    * The 'missing' modifier causes clustering to be based on\n      identity-by-missingness instead of identity-by-state, and writes a\n      space-delimited identity-by-missingness matrix to disk.\n    * The 'only2' modifier causes only a .cluster2 file (which is valid input\n      for --within) to be written; otherwise 2 other files will be produced.\n    * By default, IBS ties are not broken in the same manner as PLINK 1.07, so\n      final cluster solutions tend to differ.  This is generally harmless.\n      However, to simplify testing, you can use the 'old-tiebreaks' modifier to\n      force emulation of the old algorithm.\n\n  --pca [count] ['header'] ['tabs'] ['var-wts']\n    Calculates a variance-standardized relationship matrix (use\n    --make-rel/--make-grm-gz/--make-grm-bin to dump it), and extracts the top\n    20 principal components.\n    * It is usually best to perform this calculation on a marker set in\n      approximate linkage equilibrium.\n    * You can change the number of PCs by passing a numeric parameter.\n    * The 'header' modifier adds a header line to the .eigenvec output file.\n      (For compatibility with the GCTA flag of the same name, the default is no\n      header line.)\n    * The 'tabs' modifier causes the .eigenvec file(s) to be tab-delimited.\n    * The 'var-wts' modifier requests an additional .eigenvec.var file with PCs\n      expressed as variant weights instead of sample weights.\n\n  --neighbour <n1> <n2>\n    (alias: --neighbor)\n    Report IBS distances from each sample to their n1th- to n2th-nearest\n    neighbors, associated Z-scores, and the identities of those neighbors.\n    Useful for outlier detection.\n\n  --assoc ['perm' | 'mperm='<value>] ['perm-count'] [{fisher | fisher-midp}]\n          ['counts'] ['set-test']\n  --assoc ['perm' | 'mperm='<value>] ['perm-count'] ['qt-means'] ['lin']\n          ['set-test']\n  --model ['perm' | 'mperm='<value>] ['perm-count']\n          [{fisher | fisher-midp | trend-only}] ['set-test']\n          [{dom | rec | gen | trend}]\n    Basic association analysis report.\n    Given a case/control phenotype, --assoc performs a 1df chi-square allelic\n    test, while --model performs 4 other tests as well (1df dominant gene\n    action, 1df recessive gene action, 2df genotypic, Cochran-Armitage trend).\n    * With 'fisher'/'fisher-midp', Fisher's exact test is used to generate\n      p-values.  'fisher-midp' also applies Lancaster's mid-p adjustment.\n    * 'perm' causes an adaptive permutation test to be performed.\n    * 'mperm='<value> causes a max(T) permutation test with the specified\n      number of replications to be performed.\n    * 'perm-count' causes the permutation test report to include counts instead\n      of frequencies.\n    * 'counts' causes --assoc to report allele counts instead of frequencies.\n    * 'set-test' tests the significance of variant sets.  Requires permutation;\n      can be customized with --set-p/--set-r2/--set-max.\n    * 'dom', 'rec', 'gen', and 'trend' force the corresponding test to be used\n      as the basis for --model permutation.  (By default, the most significant\n      result among the allelic, dominant, and recessive tests is used.)\n    * 'trend-only' causes only the trend test to be performed.\n    Given a quantitative phenotype, --assoc normally performs a Wald test.\n    * In this case, the 'qt-means' modifier causes trait means and standard\n      deviations stratified by genotype to be reported as well.\n    * 'lin' causes the Lin statistic to be computed, and makes it the basis for\n      multiple-testing corrections and permutation tests.\n    Several other flags (most notably, --aperm) can be used to customize the\n    permutation test.\n\n  --mh ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n    (alias: --cmh)\n  --bd ['perm' | 'perm-bd' | 'mperm='<value>] ['perm-count'] ['set-test']\n  --mh2\n  --homog\n    Given a case/control phenotype and a set of clusters, --mh computes 2x2xK\n    Cochran-Mantel-Haenszel statistics for each variant, while --bd also\n    performs the Breslow-Day test for odds ratio homogeneity.  Permutation and\n    variant set testing based on the CMH (default) or Breslow-Day (when\n    'perm-bd' is present) statistic are supported.\n    The following similar analyses are also available:\n    * --mh2 swaps the roles of case/control status and cluster membership,\n      performing a phenotype-stratified IxJxK Cochran-Mantel-Haenszel test on\n      association between cluster assignments and genotypes.\n    * --homog executes an alternative to the Breslow-Day test, based on\n      partitioning of the chi-square statistic.\n\n  --gxe [covariate index]\n    Given both a quantitative phenotype and a case/control covariate loaded\n    with --covar defining two groups, --gxe compares the regression coefficient\n    derived from considering only members of one group to the regression\n    coefficient derived from considering only members of the other.  By\n    default, the first covariate in the --covar file defines the groups; use\n    e.g. \"--gxe 3\" to base them on the third covariate instead.\n\n  --linear ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n           [{genotypic | hethom | dominant | recessive | no-snp}]\n           ['hide-covar'] [{sex | no-x-sex}] ['interaction'] ['beta']\n           ['standard-beta'] ['intercept']\n  --logistic ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n             [{genotypic | hethom | dominant | recessive | no-snp}]\n             ['hide-covar'] [{sex | no-x-sex}] ['interaction'] ['beta']\n             ['intercept']\n    Multi-covariate association analysis on a quantitative (--linear) or\n    case/control (--logistic) phenotype.  Normally used with --covar.\n    * 'perm' normally causes an adaptive permutation test to be performed on\n      the main effect, while 'mperm='<value> starts a max(T) permutation test.\n    * 'perm-count' causes the permutation test report to include counts instead\n      of frequencies.\n    * 'set-test' tests the significance of variant sets.  Requires permutation;\n      can be customized with --set-p/--set-r2/--set-max.\n    * The 'genotypic' modifier adds an additive effect/dominance deviation 2df\n      joint test (0/1/2 and 0/1/0 coding), while 'hethom' uses 0/0/1 and 0/1/0\n      coding instead.  If permutation is also requested, these modifiers cause\n      permutation to be based on the joint test.\n    * 'dominant' and 'recessive' specify a model assuming full dominance or\n      recessiveness, respectively, for the A1 allele.\n    * 'no-snp' causes regression to be performed only on the phenotype and the\n      covariates, without reference to genomic data.  If permutation is also\n      requested, results are reported for all covariates.\n    * 'hide-covar' removes covariate-specific lines from the report.\n    * By default, sex (male = 1, female = 0) is automatically added as a\n      covariate on X chromosome variants, and nowhere else.  The 'sex' modifier\n      causes it to be added everywhere, while 'no-x-sex' excludes it.\n    * 'interaction' adds genotype x covariate interactions to the model.  This\n      cannot be used with the usual permutation tests; use --tests to define\n      the permutation test statistic instead.\n    * 'intercept' causes intercepts to be included in the main report.\n    * For logistic regressions, the 'beta' modifier causes regression\n      coefficients instead of odds ratios to be reported.\n    * With --linear, the 'standard-beta' modifier standardizes the phenotype\n      and all predictors to zero mean and unit variance before regression.\n\n  --dosage <allele dosage file> ['noheader'] ['skip0='<i>] ['skip1='<j>]\n           ['skip2='<k>] ['dose1'] ['format='<m>] ['Zout']\n           [{occur | standard-beta}] ['sex'] ['case-control-freqs']\n  --dosage <list file> list [{sepheader | noheader}] ['skip0='<i>]\n           ['skip1='<j>] ['skip2='<k>] ['dose1'] ['format='<m>] ['Zout']\n           [{occur | standard-beta}] ['sex'] ['case-control-freqs']\n  --write-dosage\n    Process (possibly gzipped) text files with variant-major allelic dosage\n    data.  This cannot be used with a regular input fileset; instead, you must\n    *only* specify a .fam and possibly a .map file, and you can't specify any\n    other commands.\n    * PLINK 2.0 will have first-class support for genotype probabilities.  An\n      equivalent data import flag will be provided then, and --dosage will be\n      retired.\n    * By default, --dosage assumes that only one allelic dosage file should be\n      loaded.  To specify multiple files,\n      1. create a master list with one entry per line.  There are normally two\n         supported formats for this list: just a filename per line, or variant\n         batch numbers in the first column and filenames in the second.\n      2. Provide the name of that list as the first --dosage parameter.\n      3. Add the 'list' modifier.\n    * By default, --dosage assumes the allelic dosage file(s) contain a header\n      line, which has 'SNP' in column i+1, 'A1' in column i+j+2, 'A2' in column\n      i+j+3, and sample FID/IIDs starting from column i+j+k+4.  (i/j/k are\n      normally zero, but can be changed with 'skip0', 'skip1', and 'skip2'\n      respectively.)  If such a header line is not present,\n      * when all samples appear in the same order as they do in the .fam file,\n        you can use the 'noheader' modifier.\n      * Otherwise, use the 'sepheader' modifier, and append sample ID filenames\n        to your 'list' file entries.\n    * The 'format=' modifier lets you specify the number of values used to\n      represent each dosage.  'format=1' normally indicates a single 0..2 A1\n      expected count; 'dose1' modifies this to a 0..1 frequency.  'format=2'\n      (the default) indicates a 0..1 homozygous A1 likelihood followed by a\n      0..1 het likelihood, while 'format=3' indicates 0..1 hom A1, 0..1 het,\n      0..1 hom A2.\n    * 'Zout' causes the output file to be gzipped.\n    * Normally, an association analysis is performed.  'standard-beta' and\n      'sex' behave as they are supposed to with --linear/--logistic.\n      'case-control-freqs' causes case and control allele frequencies to be\n      reported separately.\n    * There are three alternate modes which cause the association analysis to\n      be skipped.\n      * 'occur' requests a simple variant occurrence report.\n      * --write-dosage causes a simple merged file matching the 'format'\n        specification (not including 'dose1') to be generated.\n      * --score applies a linear scoring system to the dosages.\n\n  --lasso <h2 estimate> [min lambda] ['report-zeroes']\n    Estimate variant effect sizes via LASSO regression.  You must provide an\n    additive heritability estimate to calibrate the regression.\n    Note that this method may require a very large sample size (e.g. hundreds\n    of thousands) to be effective on complex polygenic traits.\n\n  --test-missing ['perm' | 'mperm='<value>] ['perm-count'] ['midp']\n    Check for association between missingness and case/control status, using\n    Fisher's exact test.  (Het. haploids are treated as missing.)\n    The 'midp' modifier causes Lancaster's mid-p adjustment to be applied.\n\n  --make-perm-pheno <ct>\n    Generate phenotype permutations and write them to disk, without invoking an\n    association test.\n\n  --tdt [{exact | exact-midp | poo}] ['perm' | 'mperm='<value>] ['perm-count']\n        [{parentdt1 | parentdt2 | pat | mat}] ['set-test']\n    Report transmission disequilibrium test statistics, given case/control\n    phenotypes and pedigree information.\n    * A Mendel error check is performed before the main tests; offending\n      genotypes are treated as missing by this analysis.\n    * By default, the basic TDT p-value is based on a chi-square test unless\n      you request the exact binomial test with 'exact' or 'exact-midp'.\n    * 'perm'/'mperm=' requests a family-based adaptive or max(T) permutation\n      test.  By default, the permutation test statistic is the basic TDT\n      p-value; 'parentdt1'/'parentdt2' cause parenTDT or combined test\n      p-values, respectively, to be considered instead.\n    * 'set-test' tests the significance of variant sets.  This cannot be used\n      with exact tests for now.\n    The 'poo' modifier causes a parent-of-origin analysis to be performed\n    instead, with transmissions from heterozygous fathers and heterozygous\n    mothers considered separately.\n    * The parent-of-origin analysis does not currently support exact tests.\n    * By default, the permutation test statistic is the absolute\n      parent-of-origin test Z score; 'pat'/'mat' cause paternal or maternal TDT\n      chi-square statistics, respectively, to be considered instead.\n\n  --qfam ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n  --qfam-parents ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n  --qfam-between ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n  --qfam-total ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n    QFAM family-based association test for quantitative traits.\n    * A Mendel error check is performed before the main tests; offending\n      genotypes are treated as missing by this analysis.\n    * This procedure requires permutation.  'perm' and 'perm-count' have the\n      usual meanings.  However, 'mperm='<value> just specifies a fixed number\n      of permutations; the method does not support a proper max(T) test.\n    * The 'emp-se' modifier adds BETA and EMP_SE (empirical standard error for\n      beta) fields to the .perm output file.\n\n  --annotate <PLINK report> ['attrib='<file>] ['ranges='<file>]\n             ['filter='<file>] ['snps='<file>] [{NA | prune}] ['block']\n             ['subset='<file>] ['minimal'] ['distance']\n    Add annotations to a variant-based PLINK report.  This requires an\n    annotation source:\n    * 'attrib='<file> specifies a (possibly gzipped) attribute file.\n    * 'ranges='<file> specifies a gene/range list file.\n    (Both source types can be specified simultaneously.)  The following options\n    are also supported:\n    * 'filter='<file> causes only variants within one of the ranges in the file\n      to be included in the new report.\n    * 'snps='<file> causes only variants named in the file to be included in\n      the new report.\n    * The 'NA' modifier causes unannotated variants to have 'NA' instead of '.'\n      in the new report's ANNOT column, while the 'prune' modifier excludes\n      them entirely.\n    * The 'block' modifier replaces the single ANNOT column with a 0/1-coded\n      column for each possible annotation.\n    * With 'ranges',\n      * 'subset='<file> causes only intervals named in the subset file to be\n        loaded from the ranges file.\n      * interval annotations normally come with a parenthesized signed distance\n        to the interval boundary (0 if the variant is located inside the\n        interval; this is always true without --border).  They can be excluded\n        with the 'minimal' modifier.\n      * the 'distance' modifier adds 'DIST' and 'SGN' columns describing signed\n        distance to the nearest interval.\n    * When --pfilter is present, high p-values are filtered out.\n\n  --clump <PLINK report filename(s)...>\n    Process association analysis report(s) with 'SNP' and p-value columns,\n    organizing results by LD-based clumps.  Multiple filenames can be separated\n    by spaces or commas.\n\n  --gene-report <PLINK report> <gene range file>\n    Generate a gene-based report from a variant-based report.\n    * When --pfilter is present, high p-values are filtered out.\n    * When --extract (without 'range') is present, only variants named in the\n      --extract file are considered.\n\n  --meta-analysis <PLINK report filenames...>\n  --meta-analysis <PLINK report filenames...> + [{logscale | qt}]\n                  [{no-map | no-allele}] ['study'] ['report-all']\n                  ['weighted-z']\n    Perform a meta-analysis on several variant-based reports with 'SNP' and\n    'SE' fields.\n    * Normally, an 'OR' odds ratio field must also be present in each input\n      file.  With 'logscale', 'BETA' log-odds values/regression coefficients\n      are expected instead, but the generated report will still contain odds\n      ratio estimates.  With 'qt', both input and output values are regression\n      betas.\n    * 'CHR', 'BP', and 'A1' fields are also normally required.  'no-map' causes\n      them to all be ignored, while 'no-allele' causes just 'A1' to be ignored.\n    * If 'A2' fields are present, and neither 'no-map' nor 'no-allele' was\n      specified, A1/A2 allele flips are handled properly.  Otherwise, A1\n      mismatches are thrown out.\n    * 'study' causes study-specific effect estimates to be collated in the\n      meta-analysis report.\n    * 'report-all' causes variants present in only a single input file to be\n      included in the meta-analysis report.\n    * 'weighted-z' requests weighted Z-score-based p-values (as computed by the\n      Abecasis Lab's METAL software) in addition to the usual inverse\n      variance-based analysis.  This requires P and effective sample size\n      fields.\n    * When --extract (without 'range') is present, only variants named in the\n      --extract file are considered.\n    * Unless 'no-map' is specified, chromosome filters are also respected.\n\n  --fast-epistasis [{boost | joint-effects | no-ueki}] ['case-only']\n                   [{set-by-set | set-by-all}] ['nop']\n  --epistasis [{set-by-set | set-by-all}]\n    Scan for epistatic interactions.  --fast-epistasis inspects 3x3 joint\n    genotype count tables and only applies to case/control phenotypes, while\n    --epistasis performs linear or logistic regression.\n    * By default, --fast-epistasis uses the PLINK 1.07 allele-based test.  Two\n      newer tests are now supported: 'boost' invokes the likelihood ratio test\n      introduced by Wan X et al. (2010) BOOST: A Fast Approach to Detecting\n      Gene-Gene Interactions in Genome-wide Case-Control Studies, while\n      'joint-effects' applies the joint effects test introduced in Ueki M,\n      Cordell HJ (2012) Improved statistics for genome-wide interaction\n      analysis.\n    * The original --fast-epistasis test normally applies the variance and\n      empty cell corrections suggested by Ueki and Cordell's paper.  To disable\n      them, use the 'no-ueki' modifier.\n    * 'case-only' requests a case-only instead of a case/control test.\n    * By default, all pairs of variants across the entire genome are tested.\n      To just test pairs of variants within a single set, add the 'set-by-set'\n      modifier and load exactly one set with --set/--make-set; with exactly two\n      sets loaded, all variants in one set are tested against all variants in\n      the other.  'set-by-all' tests all variants in one set against the entire\n      genome instead.\n    * 'nop' strips p-values from the main report.\n    * These computations can be subdivided with --parallel; however...\n  --epistasis-summary-merge <common file prefix> <ct>\n    When a --[fast-]epistasis job is subdivided with --parallel, the main\n    report can be assembled at the end by applying Unix 'cat' in the usual\n    manner, but the .summary.1, .summary.2, ... files may require a specialized\n    merge.  --epistasis-summary-merge takes care of the latter.\n\n  --twolocus <variant ID> <variant ID>\n    Two-locus joint genotype count report.\n\n  --score <filename> [i] [j] [k] ['header'] [{sum | no-sum}]\n          [{no-mean-imputation | center}] ['include-cnt'] ['double-dosage']\n    Apply a linear scoring system to each sample.\n    The input file should have one line per scored variant.  Variant IDs are\n    read from column #i, allele codes are read from column #j, and scores are\n    read from column #k, where i defaults to 1, j defaults to i+1, and k\n    defaults to j+1.\n    * The 'header' modifier causes the first nonempty line of the input file to\n      be ignored; otherwise, --score assumes there is no header line.\n    * By default, final scores are averages of the valid per-variant scores.\n      The 'sum' modifier causes sums to be reported instead.  (This cannot be\n      used with 'no-mean-imputation'.  And for backward compatibility, 'sum' is\n      automatically on with dosage data unless 'no-sum' is specified.)\n    * By default, copies of the unnamed allele contribute zero to score, while\n      missing genotypes contribute an amount proportional to the loaded (via\n      --read-freq) or imputed allele frequency.  To throw out missing\n      observations instead (decreasing the denominator in the final average\n      when this happens), use the 'no-mean-imputation' modifier.\n    * Alternatively, you can use the 'center' modifier to shift all scores to\n      mean zero.\n    * This command can be used with dosage data.  By default, the 'CNT' column\n      is omitted from the output file in this case; use 'include-cnt' to keep\n      it.  Also, note that scores are multiplied by 0..1 dosages, not 0..2\n      diploid allele counts, unless the 'double-dosage' modifier is present.\n\n  --R <R script file> ['debug']\n    Connect to a Rserve (preferably version 1.7 or later) background process,\n    and execute the Rplink function defined in the input file.  (Unless the\n    'debug' modifier is present; in that case, the R commands that PLINK would\n    have tried to execute are logged to a file.)\n\n  --write-var-ranges <block ct>\n    Divide the set of variants into equal-size blocks.  (Can be used with\n    --snps to split a job across multiple machines.)\n\nThe following other flags are supported.  (Order of operations is described at\nhttps://www.cog-genomics.org/plink/1.9/order .)\n  --script <fname> : Include command-line options from file.\n  --rerun [log]    : Rerun commands in log (default 'plink.log').\n  --version        : Display only version number before exiting.\n  --silent         : Suppress output to console.\n  --gplink         : Reserved for interoperation with gPLINK.\n  --missing-genotype <char> : Set missing genotype code (normally '0').\n  --double-id          : Set both FIDs and IIDs to the VCF/BCF sample ID.\n  --const-fid [ID]     : Set all FIDs to the given constant (default '0').\n  --id-delim [d]       : Parse sample IDs as <FID><d><IID> (default delim '_').\n  --vcf-idspace-to <c> : Convert spaces in sample IDs to the given character.\n  --biallelic-only ['strict'] ['list'] : Skip VCF variants with 2+ ALT alleles.\n  --vcf-min-qual <val>           : Skip VCF variants with low/missing QUAL.\n  --vcf-filter [exception(s)...] : Skip variants which have FILTER failures.\n  --vcf-require-gt               : Skip variants with no GT field.\n  --vcf-min-gq <val>             : No-call a genotype when GQ is below the\n                                   given threshold.\n  --vcf-min-gp <val>             : No-call a genotype when 0-1 scaled GP is\n                                   below the given threshold.\n  --vcf-half-call <m>  : Specify how '0/.' and similar VCF GT values should be\n                         handled.  The following four modes are supported:\n                         * 'error'/'e' (default) errors out and reports line #.\n                         * 'haploid'/'h' treats them as haploid calls.\n                         * 'missing'/'m' treats them as missing.\n                         * 'reference'/'r' treats the missing value as 0.\n  --oxford-single-chr <chr nm> : Specify single-chromosome .gen file with\n                                 ignorable first column.\n  --oxford-pheno-name <col nm> : Import named phenotype from the .sample file.\n  --hard-call-threshold <val>  : When an Oxford-format fileset is loaded, calls\n  --hard-call-threshold random   with uncertainty level greater than 0.1 are\n                                 normally treated as missing.  You can adjust\n                                 this threshold by providing a numeric\n                                 parameter, or randomize all calls with\n                                 'random'.\n  --missing-code [string list] : Comma-delimited list of missing phenotype\n    (alias: --missing_code)      values for Oxford-format filesets (def. 'NA').\n  --simulate-ncases <num>   : Set --simulate case count (default 1000).\n  --simulate-ncontrols <n>  : Set --simulate control count (default 1000).\n  --simulate-prevalence <p> : Set --simulate disease prevalence (default 0.01).\n  --simulate-n <num>        : Set --simulate-qt sample count (default 1000).\n  --simulate-label <prefix> : Set --simulate[-qt] FID/IID name prefix.\n  --simulate-missing <freq> : Set --simulate[-qt] missing genotype frequency.\n  --allow-extra-chr ['0']   : Permit unrecognized chromosome codes.  The '0'\n    (alias: --aec)            modifier causes them to be treated as if they had\n                              been set to zero.\n  --chr-set <autosome ct> ['no-x'] ['no-y'] ['no-xy'] ['no-mt'] :\n    Specify a nonhuman chromosome set.  The first parameter sets the number of\n    diploid autosome pairs if positive, or haploid chromosomes if negative.\n    Given diploid autosomes, the remaining modifiers indicate the absence of\n    the named non-autosomal chromosomes.\n  --cow/--dog/--horse/--mouse/--rice/--sheep : Shortcuts for those species.\n  --autosome-num <value>    : Alias for \"--chr-set <value> no-y no-xy no-mt\".\n  --cm-map <fname pattern> [chr] : Use SHAPEIT-format recombination maps to set\n                                   centimorgan positions.  To process more than\n                                   one chromosome, include a '@' in the first\n                                   parameter where the chrom. number belongs,\n                                   e.g. 'genetic_map_chr@_combined_b37.txt'.\n  --zero-cms         : Zero out centimorgan positions.\n  --allow-no-samples : Allow the input fileset to contain no samples.\n  --allow-no-vars    : Allow the input fileset to contain no variants.\n  --pheno <fname>  : Load phenotype data from the specified file, instead of\n                     using the values in the main input fileset.\n  --all-pheno      : For basic association tests, loop through all phenotypes\n                     in --pheno file.\n  --mpheno <n>     : Load phenotype from column (n+2) in --pheno file.\n  --pheno-name <c> : If --pheno file has a header row, use column with the\n                     given name.\n  --pheno-merge    : When the main input fileset contains an phenotype value\n                     for a sample, but the --pheno file does not, use the\n                     original value instead of treating the phenotype as\n                     missing.\n  --missing-phenotype <v> : Set missing phenotype value (normally -9).\n  --1                     : Expect case/control phenotypes to be coded as\n                            0 = control, 1 = case, instead of the usual\n                            0 = missing, 1 = control, 2 = case.  This also\n                            forces phenotypes to be interpreted as case/ctrl.\n  --make-pheno <fn> <val> : Define a new case/control phenotype.  If the val\n                            parameter is '*', all samples listed in the given\n                            file are cases, and everyone else is a control.\n                            (Note that, in some shells, it is necessary to\n                            surround the * with quotes.)\n                            Otherwise, all samples with third column entry\n                            equal to the val parameter are cases, and all other\n                            samples mentioned in the file are controls.\n  --tail-pheno <Lt> [Hbt] : Downcode a scalar phenotype to a case/control\n                            phenotype.  All samples with phenotype values\n                            greater than Hbt are cases, and all with values\n                            less than or equal to Lt are controls.  If Hbt is\n                            unspecified, it is equal to Lt; otherwise,\n                            in-between phenotype values are set to missing.\n  --covar <filename> ['keep-pheno-on-missing-cov'] : Specify covariate file.\n  --covar-name <...>       : Specify covariate(s) in --covar file by name.\n                             Separate multiple names with spaces or commas, and\n                             use dashes to designate ranges.\n  --covar-number <...>     : Specify covariate(s) in --covar file by index.\n  --no-const-covar         : Exclude constant covariates.\n  --allow-no-covars        : Allow no covariates to be loaded from --covar\n                             file.\n  --within <f> ['keep-NA'] : Specify initial cluster assignments.\n  --mwithin <n>            : Load cluster assignments from column n+2.\n  --family                 : Create a cluster for each family ID.\n  --loop-assoc <f> ['keep-NA']  : Run specified case/control association\n                                  commands once for each cluster in the file,\n                                  using cluster membership as the phenotype.\n  --set <filename>              : Load sets from a .set file.\n  --set-names <name(s)...>      : Load only sets named on the command line.\n                                  Use spaces to separate multiple names.\n  --subset <filename>           : Load only sets named in the given text file.\n  --set-collapse-all <set name> : Merge all sets.\n  --complement-sets             : Invert all sets.  (Names gain 'C_' prefixes.)\n  --make-set-complement-all <s> : --set-collapse-all + inversion.\n  --make-set <filename>         : Define sets from a list of named bp ranges.\n  --make-set-border <kbs>       : Stretch regions in --make-set file.\n  --make-set-collapse-group     : Define sets from groups instead of sets in\n                                  --make-set file.\n  --keep <filename>       : Exclude all samples not named in the file.\n  --remove <filename>     : Exclude all samples named in the file.\n  --keep-fam <filename>   : Exclude all families not named in the file.\n  --remove-fam <filename> : Exclude all families named in the file.\n  --extract ['range'] <f> : Exclude all variants not named in the file.\n  --exclude ['range'] <f> : Exclude all variants named in the file.\n  --keep-clusters <filename>          : These can be used individually or in\n  --keep-cluster-names <name(s)...>     combination to define a list of\n                                        clusters to keep; all samples not in a\n                                        cluster in that list are then excluded.\n                                        Use spaces to separate cluster names\n                                        for --keep-cluster-names.\n  --remove-clusters <filename>        : Exclude all clusters named in the file.\n  --remove-cluster-names <name(s)...> : Exclude the named clusters.\n  --gene <sets...> : Exclude variants not in a set named on the command line.\n                     (Separate multiple set names with spaces.)\n  --gene-all       : Exclude variants which aren't a member of any set.  (PLINK\n                     1.07 automatically did this under some circumstances.)\n  --attrib <f> [att lst] : Given a file assigning attributes to variants, and a\n  --attrib-indiv <f> [a]   comma-delimited list (with no whitespace) of\n                           attribute names, remove variants/samples which are\n                           either missing from the file or don't have any of\n                           the listed attributes.  If some attribute names in\n                           the list are preceded by '-', they are treated as\n                           \"negative match conditions\" instead: variants with\n                           at least one negative match attribute are removed.\n                           The first character in the list cannot be a '-', due\n                           to how command-line parsing works; add a comma in\n                           front to get around this.\n  --chr <chrs...>  : Exclude all variants not on the given chromosome(s).\n                     Valid choices for humans are 0 (unplaced), 1-22, X, Y, XY,\n                     and MT.  Separate multiple chromosomes with spaces and/or\n                     commas, and use a dash (no adjacent spaces permitted) to\n                     denote a range, e.g. \"--chr 1-4, 22, xy\".\n  --not-chr <...>  : Reverse of --chr (exclude variants on listed chromosomes).\n  --autosome       : Exclude all non-autosomal variants.\n  --autosome-xy    : Exclude all non-autosomal variants, except those with\n                     chromosome code XY (pseudo-autosomal region of X).\n  --snps-only ['just-acgt'] : Exclude non-SNP variants.  By default, SNP = both\n                              allele codes are single-character; 'just-acgt'\n                              restricts codes to {A,C,G,T,a,c,g,t,<missing>}.\n  --from <var ID>  : Use ID(s) to specify a variant range to load.  When used\n  --to   <var ID>    together, both variants must be on the same chromosome.\n  --snp  <var ID>  : Specify a single variant to load.\n  --exclude-snp <> : Specify a single variant to exclude.\n  --window  <kbs>  : With --snp or --exclude-snp, loads/excludes all variants\n                     within half the specified kb distance of the named one.\n  --from-bp <pos>  : Use physical position(s) to define a variant range to\n  --to-bp   <pos>    load.  --from-kb/--to-kb/--from-mb/--to-mb allow decimal\n  --from-kb <pos>    values.  You must also specify a single chromosome (using\n  --to-kb   <pos>    e.g. --chr) when using these flags.\n  --from-mb <pos>\n  --to-mb   <pos>\n  --snps <var IDs...>  : Use IDs to specify variant range(s) to load or\n  --exclude-snps <...>   exclude.  E.g. \"--snps rs1111-rs2222, rs3333, rs4444\".\n  --thin <p>       : Randomly remove variants, retaining each with prob. p.\n  --thin-count <n> : Randomly remove variants until n of them remain.\n  --bp-space <bps> : Remove variants so that each pair is no closer than the\n                     given bp distance.  (Equivalent to VCFtools --thin.)\n  --thin-indiv <p>         : Randomly remove samples, retaining with prob. p.\n  --thin-indiv-count <n>   : Randomly remove samples until n of them remain.\n  --filter <f> <val(s)...> : Exclude all samples without a 3rd column entry in\n                             the given file matching one of the given\n                             space-separated value(s).\n  --mfilter <n>            : Match against (n+2)th column instead.\n  --geno [val]     : Exclude variants with missing call frequencies greater\n                     than a threshold (default 0.1).  (Note that the default\n                     threshold is only applied if --geno is invoked without a\n                     parameter; when --geno is not invoked, no per-variant\n                     missing call frequency ceiling is enforced at all.  Other\n                     inclusion/exclusion default thresholds work the same way.)\n  --mind [val]     : Exclude samples with missing call frequencies greater than\n                     a threshold (default 0.1).\n  --oblig-missing <f1> <f2> : Specify blocks of missing genotype calls for\n                              --geno/--mind to ignore.  The first file should\n                              have variant IDs in the first column and block\n                              IDs in the second, while the second file should\n                              have FIDs in the first column, IIDs in the\n                              second, and block IDs in the third.\n  --prune             : Remove samples with missing phenotypes.\n  --maf [freq]        : Exclude variants with minor allele frequency lower than\n                        a threshold (default 0.01).\n  --max-maf <freq>    : Exclude variants with MAF greater than the threshold.\n  --mac <ct>          : Exclude variants with minor allele count lower than the\n    (alias: --min-ac)   given threshold.\n  --max-mac <ct>      : Exclude variants with minor allele count greater than\n    (alias: --max-ac)   the given threshold.\n  --maf-succ       : Rule of succession MAF estimation (used in EIGENSOFT).\n                     Given j observations of one allele and k >= j observations\n                     of the other, infer a MAF of (j+1) / (j+k+2), rather than\n                     the default j / (j+k).\n  --read-freq <fn> : Estimate MAFs and heterozygote frequencies from the given\n                     --freq[x] report, instead of the input fileset.\n  --hwe <p> ['midp'] ['include-nonctrl'] : Exclude variants with Hardy-Weinberg\n                                           equilibrium exact test p-values\n                                           below a threshold.\n  --me <t> <v> ['var-first'] : Filter out trios and variants with Mendel error\n                               rates exceeding the given thresholds.\n  --me-exclude-one [ratio]   : Make --me exclude only one sample per trio.\n  --qual-scores <f> [qcol] [IDcol] [skip] : Filter out variants with\n                                            out-of-range quality scores.\n                                            Default range is now [0, \\infty ).\n  --qual-threshold <min qual score>       : Set --qual-scores range floor.\n  --qual-max-threshold <max qual score>   : Set --qual-scores range ceiling.\n  --allow-no-sex   : Do not treat ambiguous-sex samples as having missing\n                     phenotypes in analysis commands.  (Automatic /w --no-sex.)\n  --must-have-sex  : Force ambiguous-sex phenotypes to missing on\n                     --make-bed/--make-just-fam/--recode/--write-covar.\n  --filter-cases       : Include only cases in the current analysis.\n  --filter-controls    : Include only controls.\n  --filter-males       : Include only males.\n  --filter-females     : Include only females.\n  --filter-founders    : Include only founders.\n  --filter-nonfounders : Include only nonfounders.\n  --nonfounders        : Include nonfounders in allele freq/HWE calculations.\n  --make-founders ['require-2-missing'] ['first'] :\n    Clear parental IDs for those with 1+ missing parent(s).\n  --recode-allele <fn> : With --recode A/A-transpose/AD, count alleles named in\n                         the file (otherwise A1 alleles are always counted).\n  --output-chr <MT code> : Set chromosome coding scheme in output files by\n                           providing the desired human mitochondrial code.\n                           (Options are '26', 'M', 'MT', '0M', 'chr26', 'chrM',\n                           and 'chrMT'.)\n  --output-missing-genotype <ch> : Set the code used to represent missing\n                                   genotypes in output files (normally the\n                                   --missing-genotype value).\n  --output-missing-phenotype <s> : Set the string used to represent missing\n                                   phenotypes in output files (normally the\n                                   --missing-phenotype value).\n  --zero-cluster <f> : In combination with --within/--family, set blocks of\n                       genotype calls to missing.  The input file should have\n                       variant IDs in the first column and cluster IDs in the\n                       second.  This must now be used with --make-bed and no\n                       other output commands.\n  --set-hh-missing       : Cause --make-bed and --recode to set heterozygous\n                           haploid genotypes to missing.\n  --set-mixed-mt-missing : Cause --make-bed and --recode to set mixed MT\n                           genotypes to missing.\n  --split-x <bp1> <bp2> ['no-fail']\n  --split-x <build> ['no-fail'] :\n    Changes chromosome code of all chrX variants with bp position <= bp1 or >=\n    bp2 to XY.  The following build codes are supported as shorthand:\n    * 'b36'/'hg18' = NCBI 36, 2709521/154584237\n    * 'b37'/'hg19' = GRCh37, 2699520/154931044\n    * 'b38'/'hg38' = GRCh38, 2781479/155701383\n    By default, PLINK errors out when no variants would be affected by\n    --split-x; the 'no-fail' modifier (useful in scripts) overrides this.\n  --merge-x ['no-fail'] : Merge XY chromosome back with X.\n  --set-me-missing      : Cause --make-bed to set Mendel errors to missing.\n  --fill-missing-a2     : Cause --make-bed to replace all missing calls with\n                          homozygous A2 calls.\n  --set-missing-var-ids <t>   : Given a template string with a '@' where the\n                                chromosome code should go and '#' where the bp\n                                coordinate belongs, --set-missing-var-ids\n                                assigns chromosome-and-bp-based IDs to unnamed\n                                variants.\n                                You may also use '$1' and '$2' to refer to\n                                allele names in the template string, and in\n                                fact this becomes essential when multiple\n                                variants share the same coordinate.\n  --new-id-max-allele-len <n> : Specify maximum number of leading characters\n                                from allele names to include in new variant IDs\n                                (default 23).\n  --missing-var-code <string> : Change unnamed variant code (default '.').\n  --update-chr  <f> [chrcol] [IDcol]  [skip] : Update variant chromosome codes.\n  --update-cm   <f> [cmcol]  [IDcol]  [skip] : Update centimorgan positions.\n  --update-map  <f> [bpcol]  [IDcol]  [skip] : Update variant bp positions.\n  --update-name <f> [newcol] [oldcol] [skip] : Update variant IDs.\n  --update-alleles <fname> : Update variant allele codes.\n  --allele1234 ['multichar'] : Interpret/recode A/C/G/T alleles as 1/2/3/4.\n                               With 'multichar', converts all A/C/G/Ts in\n                               allele names to 1/2/3/4s.\n  --alleleACGT ['multichar'] : Reverse of --allele1234.\n  --update-ids <f>     : Update sample IDs.\n  --update-parents <f> : Update parental IDs.\n  --update-sex <f> [n] : Update sexes.  Sex (1 or M = male, 2 or F = female, 0\n                         = missing) is loaded from column n+2 (default n is 1).\n  --flip <filename>    : Flip alleles (A<->T, C<->G) for SNP IDs in the file.\n  --flip-subset <fn>   : Only apply --flip to samples in --flip-subset file.\n  --flip-scan-window <ct+1> : Set --flip-scan max variant ct dist. (def. 10).\n  --flip-scan-window-kb <x> : Set --flip-scan max kb distance (default 1000).\n  --flip-scan-threshold <x> : Set --flip-scan min correlation (default 0.5).\n  --keep-allele-order  : Keep the allele order defined in the .bim file,\n  --real-ref-alleles     instead of forcing A2 to be the major allele.\n                         --real-ref-alleles also removes 'PR' from the INFO\n                         values emitted by --recode vcf{,-fid,-iid}.\n  --a1-allele <f> [a1col] [IDcol] [skip] : Force alleles in the file to A1.\n  --a2-allele <filename> [a2col] [IDcol] [skip] :\n    Force alleles in the file to A2.  (\"--a2-allele <VCF filename> 4 3 '#'\",\n    which scrapes reference allele assignments from a VCF file, is especially\n    useful.)\n  --indiv-sort <m> [f] : Specify FID/IID sort order.  The following four modes\n                         are supported:\n                         * 'none'/'0' keeps samples in the order they were\n                           loaded.  Default for non-merge operations.\n                         * 'natural'/'n' invokes 'natural sort', e.g.\n                           'id2' < 'ID3' < 'id10'.  Default when merging.\n                         * 'ascii'/'a' sorts in ASCII order, e.g.\n                           'ID3' < 'id10' < 'id2'.\n                         * 'file'/'f' uses the order in the given file (named\n                           in the second parameter).\n                         For now, only --merge/--bmerge/--merge-list and\n                         --make-bed/--make-just-fam respect this flag.\n  --with-phenotype ['no-parents'] [{no-sex | female-2}] :\n    Include more sample info in new .cov file.\n  --dummy-coding [N] ['no-round'] : Split categorical variables (n categories,\n                                    2 < n <= N, default N is 49) into n-1\n                                    binary dummy variables when writing\n                                    covariate file.\n  --merge-mode <n>   : Adjust --[b]merge/--merge-list behavior based on a\n                       numeric code.\n                       1 (default) = ignore missing calls, otherwise difference\n                                     -> missing\n                       2 = only overwrite originally missing calls\n                       3 = only overwrite when nonmissing in new file\n                       4/5 = never overwrite and always overwrite, respectively\n                       6 = report all mismatching calls without merging\n                       7 = report mismatching nonmissing calls without merging\n  --merge-equal-pos  : With --merge/--bmerge/--merge-list, merge variants with\n                       different names but identical positions.  (Exception:\n                       same-position chromosome code 0 variants aren't merged.)\n  --mendel-duos      : Make Mendel error checks consider samples with only one\n                       parent in the dataset.\n  --mendel-multigen  : Make Mendel error checks consider (great-)grandparental\n                       genotypes when parental genotype data is missing.\n  --ld-window <ct+1> : Set --r/--r2 max variant ct pairwise distance (usu. 10).\n  --ld-window-kb <x> : Set --r/--r2 max kb pairwise distance (usually 1000).\n  --ld-window-cm <x> : Set --r/--r2 max centimorgan pairwise distance.\n  --ld-window-r2 <x> : Set threshold for --r2 report inclusion (usually 0.2).\n  --ld-snp <var ID>  : Set first variant in all --r/--r2 pairs.\n  --ld-snps <vID...> : Restrict first --r/--r2 variant to the given ranges.\n  --ld-snp-list <f>  : Restrict first --r/--r2 var. to those named in the file.\n  --list-all         : Generate the 'all' mode report when using --show-tags in\n                       file mode.\n  --tag-kb <kbs>     : Set --show-tags max tag kb distance (default 250).\n  --tag-r2 <val>     : Set --show-tags min tag r-squared (default 0.8)\n  --tag-mode2        : Use two-column --show-tags (file mode) I/O format.\n  --ld-xchr <code>   : Set chrX model for --indep[-pairwise], --r/--r2,\n                       --flip-scan, and --show-tags.\n                       1 (default) = males coded 0/1, females 0/1/2 (A1 dosage)\n                       2 = males coded 0/2\n                       3 = males coded 0/2, but females given double weighting\n  --blocks-max-kb <kbs>      : Set --blocks maximum haploblock span (def. 200).\n  --blocks-min-maf <cutoff>  : Adjust --blocks MAF minimum (default 0.05).\n  --blocks-strong-lowci <x>  : Set --blocks \"strong LD\" CI thresholds (defaults\n  --blocks-strong-highci <x>   0.70 and 0.98).\n  --blocks-recomb-highci <x> : Set 'recombination' CI threshold (default 0.90).\n  --blocks-inform-frac <x>   : Force haploblock <strong LD pairs>:<total\n                               informative pairs> ratios to be larger than this\n                               value (default 0.95).\n  --distance-wts exp=<x>        : When computing genomic distances, assign each\n                                  variant a weight of (2q(1-q))^{-x}, where q\n                                  is the loaded or inferred MAF.\n  --read-dists <dist file> [id file] : Load a triangular binary distance matrix\n                                       instead of recalculating from scratch.\n  --ppc-gap <val>    : Minimum number of base pairs, in thousands, between\n                       informative pairs of markers used in --genome PPC test.\n                       500 if unspecified.\n  --min <cutoff>     : Specify minimum PI_HAT for inclusion in --genome report.\n  --max <cutoff>     : Specify maximum PI_HAT for inclusion in --genome report.\n  --homozyg-match <> : Set minimum concordance across jointly homozygous\n                       variants for a pairwise allelic match to be declared.\n  --pool-size <ct>   : Set minimum size of pools in \"--homozyg group\" report.\n  --read-genome <fn> : Load --genome report for --cluster/--neighbour, instead\n                       of recalculating IBS and PPC test p-values from scratch.\n  --ppc <p-val>    : Specify minimum PPC test p-value within a cluster.\n  --mc <max size>  : Specify maximum cluster size.\n  --mcc <c1> <c2>  : Specify maximum case and control counts per cluster.\n  --K <min count>  : Specify minimum cluster count.\n  --ibm <val>      : Specify minimum identity-by-missingness.\n  --match <f> [mv] : Use covariate values to restrict clustering.  Without\n                     --match-type, two samples can only be in the same cluster\n                     if all covariates match.  The optional second parameter\n                     specifies a covariate value to treat as missing.\n  --match-type <f> : Refine interpretation of --match file.  The --match-type\n                     file is expected to be a single line with as many entries\n                     as the --match file has covariates; '0' entries specify\n                     \"negative matches\" (i.e. samples with equal covariate\n                     values cannot be in the same cluster), '1' entries specify\n                     \"positive matches\" (default), and '-1' causes the\n                     corresponding covariate to be ignored.\n  --qmatch <f> [m] : Force all members of a cluster to have similar\n  --qt <fname>       quantitative covariate values.  The --qmatch file contains\n                     the covariate values, while the --qt file is a list of\n                     nonnegative tolerances (and '-1's marking covariates to\n                     skip).\n  --pca-cluster-names <...> : These can be used individually or in combination\n  --pca-clusters <fname>      to define a list of clusters to use in the basic\n                              --pca computation.  (--pca-cluster-names expects\n                              a space-delimited sequence of cluster names,\n                              while --pca-clusters expects a file with one\n                              cluster name per line.)  All samples outside\n                              those clusters will then be projected on to the\n                              calculated PCs.\n  --mds-plot <dims> ['by-cluster'] ['eigendecomp'] ['eigvals'] :\n    Multidimensional scaling analysis.  Requires --cluster.\n  --cell <thresh>  : Skip some --model tests when a contingency table entry is\n                     smaller than the given threshold.\n  --condition <var ID> [{dominant | recessive}] : Add one variant as a --linear\n                                                  or --logistic covariate.\n  --condition-list <f> [{dominant | recessive}] : Add variants named in the\n                                                  file as --linear/--logistic\n                                                  covariates.\n  --parameters <...>  : Include only the given covariates/interactions in the\n                        --linear/--logistic models, identified by a list of\n                        1-based indices and/or ranges of them.\n  --tests <all> [...] : Perform a (joint) test on the specified term(s) in the\n                        --linear/--logistic model, identified by 1-based\n                        indices and/or ranges of them.  If permutation was\n                        requested, it is based on this test.\n                        * Note that, when --parameters is also present, the\n                          indices refer to the terms remaining AFTER pruning by\n                          --parameters.\n                        * You can use \"--tests all\" to include all terms.\n  --vif <max VIF>     : Set VIF threshold for --linear multicollinearity check\n                        (default 50).\n  --xchr-model <code> : Set the X chromosome --linear/--logistic model.\n                        0 = skip sex and haploid chromosomes\n                        1 (default) = add sex as a covariate on X chromosome\n                        2 = code male genotypes 0/2 instead of 0/1\n                        3 = test for interaction between genotype and sex\n  --lasso-select-covars [cov(s)...] : Subject some or all covariates to LASSO\n                                      model selection.\n  --adjust ['gc'] ['log10'] ['qq-plot'] : Report some multiple-testing\n                                          corrections.\n  --lambda <val>   : Set genomic control lambda for --adjust.\n  --ci <size>      : Report confidence intervals for odds ratios.\n  --pfilter <val>  : Filter out association test results with higher p-values.\n  --aperm <min perms - 1> [max perms] [alpha] [beta] [init interval] [slope] :\n    Set up to six parameters controlling adaptive permutation tests.\n    * The first two control the minimum and maximum number of permutations that\n      may be run for each variant; default values are 5 and 1000000.\n    * The next two control the early termination condition.  A\n      100% * (1 - beta/2T) confidence interval is calculated for each empirical\n      p-value, where T is the total number of variants; whenever this\n      confidence interval doesn't contain alpha, the variant is exempted from\n      further permutation testing.  Default values are 0 and 1e-4.\n    * The last two control when the early termination condition is checked.  If\n      a check occurs at permutation #p, the next check occurs after\n      <slope>p + <init interval> more permutations (rounded down).  Default\n      initial interval is 1, and default slope is 0.001.\n  --mperm-save     : Save best max(T) permutation test statistics.\n  --mperm-save-all : Save all max(T) permutation test statistics.\n  --set-p <p-val>        : Adjust set test significant variant p-value ceiling\n                           (default 0.05).\n  --set-r2 [v] ['write'] : Adjust set test significant variant pairwise r^2\n                           ceiling (default 0.5).  'write' causes violating\n                          pairs to be dumped to <output prefix>.ldset.\n  --set-max <ct>         : Adjust set test maximum # of significant variants\n                           considered per set (default 5).\n  --set-test-lambda <v>  : Specify genomic control correction for set test.\n  --border <kbs>            : Extend --annotate range intervals by given # kbs.\n  --annotate-snp-field <nm> : Set --annotate variant ID field name.\n  --clump-p1 <pval> : Set --clump index var. p-value ceiling (default 1e-4).\n  --clump-p2 <pval> : Set --clump secondary p-value threshold (default 0.01).\n  --clump-r2 <r^2>  : Set --clump r^2 threshold (default 0.5).\n  --clump-kb <kbs>  : Set --clump kb radius (default 250).\n  --clump-snp-field <n...>  : Set --clump variant ID field name (default\n                              'SNP').  With multiple field names, earlier names\n                              take precedence over later ones.\n  --clump-field <name...>   : Set --clump p-value field name (default 'P').\n  --clump-allow-overlap     : Let --clump non-index vars. join multiple clumps.\n  --clump-verbose           : Request extended --clump report.\n  --clump-annotate <hdr...> : Include named extra fields in --clump-verbose and\n                              --clump-best reports.  (Field names can be\n                              separated with spaces or commas.)\n  --clump-range <filename>  : Report overlaps between clumps and regions.\n  --clump-range-border <kb> : Stretch regions in --clump-range file.\n  --clump-index-first       : Extract --clump index vars. from only first file.\n  --clump-replicate         : Exclude clumps which contain secondary results\n                              from only one file.\n  --clump-best              : Report best proxy for each --clump index var.\n  --meta-analysis-chr-field <n...> : Set --meta-analysis chromosome, variant\n  --meta-analysis-snp-field <n...>   ID, position, A1/A2 allele, p-value,\n  --meta-analysis-bp-field <n...>    standard error, and/or effective sample\n  --meta-analysis-a1-field <n...>    size field names.\n  --meta-analysis-a2-field <n...>    Defaults are 'CHR', 'SNP', 'BP', 'A1',\n  --meta-analysis-p-field <n...>     'A2', 'P', 'SE', and 'NMISS',\n  --meta-analysis-se-field <n...>    respectively.  When multiple parameters\n  --meta-analysis-ess-field <n...>   are given to these flags, earlier names\n                                     take precedence over later ones.\n                                     Note that, if the numbers of cases and\n                                     controls are unequal, effective sample\n                                     size should be\n                                       4 / (1/<# cases> + 1/<# controls>).\n  --meta-analysis-report-dups      : When a variant appears multiple times in\n                                     in the same file, report that.\n  --gene-list-border <kbs>   : Extend --gene-report regions by given # of kbs.\n  --gene-subset <filename>   : Specify gene name subset for --gene-report.\n  --gene-report-snp-field <> : Set --gene-report variant ID field name (default\n                               'SNP').  Only relevant with --extract.\n  --gap <kbs>      : Set \"--fast-epistasis case-only\" min. gap (default 1000).\n  --epi1 <p-value> : Set --[fast-]epistasis reporting threshold (default\n                     5e-6 for 'boost', 1e-4 otherwise).\n  --epi2 <p-value> : Set threshold for contributing to SIG_E count (def. 0.01).\n  --je-cellmin <n> : Set required number of observations per 3x3x2 contingency\n                     table cell for joint-effects test (default 5).\n  --q-score-range <range file> <data file> [i] [j] ['header'] :\n    Apply --score to subset(s) of variants in the primary score list based\n    on e.g. p-value ranges.\n    * The first file should have range labels in the first column, p-value\n      lower bounds in the second column, and upper bounds in the third column.\n      Lines with too few entries, or nonnumeric values in the second or third\n      column, are ignored.\n    * The second file should contain a variant ID and a p-value on each\n      nonempty line (except possibly the first).  Variant IDs are read from\n      column #i and p-values are read from column #j, where i defaults to 1 and\n      j defaults to i+1.  The 'header' modifier causes the first nonempty line\n      of this file to be skipped.\n  --R-port <port #>  : Connect to Rserve on a port other than 6311.\n  --R-host <host>    : Connect to Rserve host.\n  --R-socket <sock>  : Connect to Rserve socket.\n  --parallel <k> <n> : Divide the output matrix into n pieces, and only compute\n                       the kth piece.  The primary output file will have the\n                       piece number included in its name, e.g. plink.rel.13 or\n                       plink.rel.13.gz if k is 13.  Concatenating these files\n                       in order will yield the full matrix of interest.  (Yes,\n                       this can be done before unzipping.)\n                       N.B. This generally cannot be used to directly write a\n                       symmetric square matrix.  Choose square0 or triangle\n                       shape instead, and postprocess as necessary.\n  --memory <val>     : Set size, in MB, of initial workspace malloc attempt.\n                       (Practically mandatory when using GNU parallel.)\n  --threads <val>    : Set maximum number of concurrent threads.\n                       This has one known limitation: some BLAS/LAPACK linear\n                       algebra operations are multithreaded in a way that PLINK\n                       cannot control.  If this is problematic, you should\n                       recompile against single-threaded BLAS/LAPACK.\n  --d <char>         : Change variant/covariate range delimiter (normally '-').\n  --seed <val...>    : Set random number seed(s).  Each value must be an\n                       integer between 0 and 4294967295 inclusive.\n  --perm-batch-size <val> : Set number of permutations per batch for some\n                            permutation tests.\n  --native           : Allow Intel MKL to use processor-dependent code paths.\n  --output-min-p <p> : Specify minimum p-value to write to reports.\n  --debug            : Use slower, more crash-resistant logging method.\n\nPrimary methods paper:\nChang CC, Chow CC, Tellier LCAM, Vattikuti S, Purcell SM, Lee JJ (2015)\nSecond-generation PLINK: rising to the challenge of larger and richer datasets.\nGigaScience, 4.\n\nFor further documentation and support, consult the main webpage\n(https://www.cog-genomics.org/plink/1.9 ) and/or the mailing list\n(https://groups.google.com/d/forum/plink2-users ).\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"coral-theater","cell_type":"code","source":"%%bash\n./plink --file GreSalVL_b37_AllelesUpdatedNewMapPED --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBED","execution_count":70,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBED.log.\nOptions in effect:\n  --file GreSalVL_b37_AllelesUpdatedNewMapPED\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBED\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\nScanning .ped file... 0%\b\b1%\b\b1%\b\b2%\b\b2%\b\b3%\b\b3%\b\b4%\b\b4%\b\b5%\b\b5%\b\b5%\b\b6%\b\b6%\b\b7%\b\b7%\b\b8%\b\b8%\b\b9%\b\b9%\b\b10%\b\b\b10%\b\b\b11%\b\b\b11%\b\b\b11%\b\b\b12%\b\b\b12%\b\b\b13%\b\b\b13%\b\b\b14%\b\b\b14%\b\b\b15%\b\b\b15%\b\b\b16%\b\b\b16%\b\b\b17%\b\b\b17%\b\b\b17%\b\b\b18%\b\b\b18%\b\b\b19%\b\b\b19%\b\b\b20%\b\b\b20%\b\b\b21%\b\b\b21%\b\b\b22%\b\b\b22%\b\b\b23%\b\b\b23%\b\b\b23%\b\b\b24%\b\b\b24%\b\b\b25%\b\b\b25%\b\b\b26%\b\b\b26%\b\b\b27%\b\b\b27%\b\b\b28%\b\b\b28%\b\b\b29%\b\b\b29%\b\b\b29%\b\b\b30%\b\b\b30%\b\b\b31%\b\b\b31%\b\b\b32%\b\b\b32%\b\b\b33%\b\b\b33%\b\b\b34%\b\b\b34%\b\b\b35%\b\b\b35%\b\b\b35%\b\b\b36%\b\b\b36%\b\b\b37%\b\b\b37%\b\b\b38%\b\b\b38%\b\b\b39%\b\b\b39%\b\b\b40%\b\b\b40%\b\b\b41%\b\b\b41%\b\b\b41%\b\b\b42%\b\b\b42%\b\b\b43%\b\b\b43%\b\b\b44%\b\b\b44%\b\b\b45%\b\b\b45%\b\b\b46%\b\b\b46%\b\b\b47%\b\b\b47%\b\b\b47%\b\b\b48%\b\b\b48%\b\b\b49%\b\b\b49%\b\b\b50%\b\b\b50%\b\b\b51%\b\b\b51%\b\b\b52%\b\b\b52%\b\b\b52%\b\b\b53%\b\b\b53%\b\b\b54%\b\b\b54%\b\b\b55%\b\b\b55%\b\b\b56%\b\b\b56%\b\b\b57%\b\b\b57%\b\b\b58%\b\b\b58%\b\b\b58%\b\b\b59%\b\b\b59%\b\b\b60%\b\b\b60%\b\b\b61%\b\b\b61%\b\b\b62%\b\b\b62%\b\b\b63%\b\b\b63%\b\b\b64%\b\b\b64%\b\b\b64%\b\b\b65%\b\b\b65%\b\b\b66%\b\b\b66%\b\b\b67%\b\b\b67%\b\b\b68%\b\b\b68%\b\b\b69%\b\b\b69%\b\b\b70%\b\b\b70%\b\b\b70%\b\b\b71%\b\b\b71%\b\b\b72%\b\b\b72%\b\b\b73%\b\b\b73%\b\b\b74%\b\b\b74%\b\b\b75%\b\b\b75%\b\b\b76%\b\b\b76%\b\b\b76%\b\b\b77%\b\b\b77%\b\b\b78%\b\b\b78%\b\b\b79%\b\b\b79%\b\b\b80%\b\b\b80%\b\b\b81%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b83%\b\b\b84%\b\b\b84%\b\b\b85%\b\b\b85%\b\b\b86%\b\b\b86%\b\b\b87%\b\b\b87%\b\b\b88%\b\b\b88%\b\b\b88%\b\b\b89%\b\b\b89%\b\b\b90%\b\b\b90%\b\b\b91%\b\b\b91%\b\b\b92%\b\b\b92%\b\b\b93%\b\b\b93%\b\b\b94%\b\b\b94%\b\b\b94%\b\b\b95%\b\b\b95%\b\b\b96%\b\b\b96%\b\b\b97%\b\b\b97%\b\b\b98%\b\b\b98%\b\b\b99%\b\b\b99%\b\b\b100%\r.ped scan complete (for binary autoconversion).\nPerforming single-pass .bed write (299140 variants, 217 people).\n0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r--file: GreSalVL_b37_AllelesUpdatedNewMapBED-temporary.bed +\nGreSalVL_b37_AllelesUpdatedNewMapBED-temporary.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBED-temporary.fam written.\n299140 variants loaded from .bim file.\n217 people (0 males, 0 females, 217 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalVL_b37_AllelesUpdatedNewMapBED.nosex .\n217 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 217 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.976208.\n299140 variants and 217 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 166 are controls.\n--make-bed to GreSalVL_b37_AllelesUpdatedNewMapBED.bed +\nGreSalVL_b37_AllelesUpdatedNewMapBED.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBED.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\nWarning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\ntreat these as missing.\n","name":"stderr"}]},{"metadata":{"trusted":true},"id":"banned-drunk","cell_type":"code","source":"%%bash\nls -altrh","execution_count":71,"outputs":[{"output_type":"stream","text":"total 497M\n-rw-r--r-- 1 jovyan jovyan  807 Apr  4  2018 .profile\n-rw-r--r-- 1 jovyan jovyan 3.7K Apr  4  2018 .bashrc\n-rw-r--r-- 1 jovyan jovyan  220 Apr  4  2018 .bash_logout\ndrwxr-xr-x 1 root   root   4.0K May 30 06:30 ..\ndrwx------ 3 jovyan jovyan 4.0K May 30 09:34 .config\n-rwxr--r-- 1 jovyan jovyan  40M May 31 12:05 plink\n-rw-r--r-- 1 jovyan jovyan 248M May 31 12:05 GreSalVL_b37_AllelesUpdatedNewMapPED.ped\n-rw-r--r-- 1 jovyan jovyan 7.3M May 31 12:05 GreSalVL_b37_AllelesUpdatedNewMapPED.map\n-rw-r--r-- 1 jovyan jovyan  368 Jun  2 13:08 README.md\n-rw-r--r-- 1 jovyan jovyan  42M Jun  2 13:08 plinkfilesEx1.zip\ndrwxr-xr-x 8 jovyan jovyan 4.0K Jun  2 13:08 .git\ndrwxr-xr-x 3 jovyan jovyan 4.0K Jun  2 13:09 .local\ndrwxr-xr-x 3 jovyan jovyan 4.0K Jun  2 13:09 .cache\ndrwxr-xr-x 5 jovyan jovyan 4.0K Jun  2 13:11 .ipython\ndrwxr-xr-x 2 jovyan jovyan 4.0K Jun  2 13:13 .ipynb_checkpoints\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam~\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed~\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:08 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim~\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:09 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.nosex\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.fam\n-rw-r--r-- 1 jovyan jovyan  15M Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bed\n-rw-r--r-- 1 jovyan jovyan 1.3K Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.log\n-rw-r--r-- 1 jovyan jovyan 7.9M Jun  2 15:11 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.nosex\n-rw-r--r-- 1 jovyan jovyan  13M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.lmiss\n-rw-r--r-- 1 jovyan jovyan 1.2K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.log\n-rw-r--r-- 1 jovyan jovyan  12K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.imiss\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.nosex\n-rw-r--r-- 1 jovyan jovyan   32 Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.irem\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.fam\n-rw-r--r-- 1 jovyan jovyan  15M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bed\n-rw-r--r-- 1 jovyan jovyan 1.6K Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.log\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:13 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.nosex\n-rw-r--r-- 1 jovyan jovyan  11M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bed\n-rw-r--r-- 1 jovyan jovyan 3.9K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.fam\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bim\n-rw-r--r-- 1 jovyan jovyan 1.5K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.log\n-rw-r--r-- 1 jovyan jovyan 2.6K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.nosex\n-rw-r--r-- 1 jovyan jovyan  11M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bed\n-rw-r--r-- 1 jovyan jovyan 3.9K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.fam\n-rw-r--r-- 1 jovyan jovyan 7.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bim\n-rw-r--r-- 1 jovyan jovyan 1.5K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.log\n-rw-r--r-- 1 jovyan jovyan  976 Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.rel.id\n-rw-r--r-- 1 jovyan jovyan  976 Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.nosex\n-rw-r--r-- 1 jovyan jovyan 1.4M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.in\n-rw-r--r-- 1 jovyan jovyan 1.7M Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.out\n-rw-r--r-- 1 jovyan jovyan 2.6K Jun  2 15:14 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.log\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.fam~\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bed~\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:16 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bim~\n-rw-r--r-- 1 jovyan jovyan 8.9K Jun  2 15:16 .jupyter-server-log.txt\n-rw-r--r-- 1 jovyan jovyan 374K Jun  2 15:16 Ex1.ipynb\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.nosex\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.fam\n-rw-r--r-- 1 jovyan jovyan 6.6M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bed\n-rw-r--r-- 1 jovyan jovyan 1.7K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.log\n-rw-r--r-- 1 jovyan jovyan 3.6M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bim\n-rw-r--r-- 1 jovyan jovyan 3.4K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBED.nosex\n-rw-r--r-- 1 jovyan jovyan  16M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBED.bed\n-rw-r--r-- 1 jovyan jovyan 5.1K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBED.fam\n-rw-r--r-- 1 jovyan jovyan 8.4M Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBED.bim\n-rw-r--r-- 1 jovyan jovyan 1.6K Jun  2 15:17 GreSalVL_b37_AllelesUpdatedNewMapBED.log\ndrwxr-xr-x 1 jovyan jovyan  12K Jun  2 15:17 .\n","name":"stdout"}]},{"metadata":{},"id":"effective-powell","cell_type":"markdown","source":"Ok, from ~256 Mb of data we obtained 25 Mb. allowing us to analyse the same amount of genotypes extremely faster."},{"metadata":{},"id":"juvenile-advocacy","cell_type":"markdown","source":"### 2. Prepare the dataset ###"},{"metadata":{},"id":"changed-capitol","cell_type":"markdown","source":"One thing that we can look at is how many SNPs per chromosome are present. We can easily do it using some simple bash command"},{"metadata":{"trusted":true},"id":"valued-salmon","cell_type":"code","source":"%%bash\ncut -f 1 *.bim |uniq -c","execution_count":72,"outputs":[{"output_type":"stream","text":"  23094 1\n  24162 2\n  17766 3\n  14497 4\n  17856 5\n  16548 6\n  15684 7\n  15810 8\n  12232 9\n  13746 10\n  14369 11\n  14622 12\n   9011 13\n   9607 14\n  10082 15\n   9242 16\n   9659 17\n   7624 18\n   6889 19\n   7612 20\n   4332 21\n   4995 22\n  22572 1\n  23658 2\n  17378 3\n  14196 4\n  17467 5\n  16173 6\n  15313 7\n  15404 8\n  11874 9\n  13459 10\n  14047 11\n  14361 12\n   8813 13\n   9405 14\n   9814 15\n   8985 16\n   9410 17\n   7456 18\n   6693 19\n   7474 20\n   4246 21\n   4844 22\n  22572 1\n  23658 2\n  17378 3\n  14196 4\n  17467 5\n  16173 6\n  15313 7\n  15404 8\n  11874 9\n  13459 10\n  14047 11\n  14361 12\n   8813 13\n   9405 14\n   9814 15\n   8985 16\n   9410 17\n   7456 18\n   6693 19\n   7474 20\n   4246 21\n   4844 22\n  22551 1\n  23636 2\n  17368 3\n  14187 4\n  17451 5\n  16166 6\n  15297 7\n  15393 8\n  11865 9\n  13441 10\n  14034 11\n  14355 12\n   8808 13\n   9396 14\n   9800 15\n   8975 16\n   9401 17\n   7451 18\n   6684 19\n   7466 20\n   4245 21\n   4840 22\n   9952 1\n  10272 2\n   8168 3\n   7360 4\n   8062 5\n   7873 6\n   6895 7\n   6792 8\n   5660 9\n   6414 10\n   6184 11\n   6377 12\n   4576 13\n   4280 14\n   4107 15\n   4105 16\n   3972 17\n   3823 18\n   2888 19\n   3547 20\n   2026 21\n   2126 22\n   9952 1\n  10272 2\n   8168 3\n   7360 4\n   8062 5\n   7873 6\n   6895 7\n   6792 8\n   5660 9\n   6414 10\n   6184 11\n   6377 12\n   4576 13\n   4280 14\n   4107 15\n   4105 16\n   3972 17\n   3823 18\n   2888 19\n   3547 20\n   2026 21\n   2126 22\n   1082 0\n  23094 1\n  24162 2\n  17766 3\n  14497 4\n  17856 5\n  16548 6\n  15684 7\n  15810 8\n  12232 9\n  13746 10\n  14369 11\n  14622 12\n   9011 13\n   9607 14\n  10082 15\n   9242 16\n   9659 17\n   7624 18\n   6889 19\n   7612 20\n   4332 21\n   4995 22\n  15371 23\n   2481 24\n    767 25\n","name":"stdout"}]},{"metadata":{},"id":"veterinary-equilibrium","cell_type":"markdown","source":"We also would want to count how many different individuals are in our dataset, just to check if any of them has been reported more than once. How can we do it?"},{"metadata":{"trusted":true},"id":"prepared-schedule","cell_type":"code","source":"%%bash\n\n#Let's check if there are duplicate individuals","execution_count":73,"outputs":[]},{"metadata":{},"id":"macro-approach","cell_type":"markdown","source":"For this exercise we want to work only with autosomal data. So we need to find a way to extract only SNPs in autosomal chromosomes."},{"metadata":{"trusted":true},"id":"signal-proceeding","cell_type":"code","source":"%%bash\n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBED --autosome --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAuto ","execution_count":74,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.log.\nOptions in effect:\n  --autosome\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBED\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAuto\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n279439 out of 299140 variants loaded from .bim file.\n217 people (0 males, 0 females, 217 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.nosex .\n217 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 217 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.978587.\n279439 variants and 217 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 166 are controls.\n--make-bed to GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bed +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAuto.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{"trusted":true},"id":"lucky-detective","cell_type":"code","source":"%%bash\n\ncut -f1 GreSalVL_b37_AllelesUpdatedNewMapBEDAuto.bim|uniq -c","execution_count":75,"outputs":[{"output_type":"stream","text":"  23094 1\n  24162 2\n  17766 3\n  14497 4\n  17856 5\n  16548 6\n  15684 7\n  15810 8\n  12232 9\n  13746 10\n  14369 11\n  14622 12\n   9011 13\n   9607 14\n  10082 15\n   9242 16\n   9659 17\n   7624 18\n   6889 19\n   7612 20\n   4332 21\n   4995 22\n","name":"stdout"}]},{"metadata":{},"id":"intelligent-turkey","cell_type":"markdown","source":"Perfect, we now have a dataset containing only autosomal markers. Before starting the real analysis we want to do some quality control steps. Firts we want to have a look at the missing data."},{"metadata":{"trusted":true},"id":"harmful-milton","cell_type":"code","source":"%%bash\n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAuto --missing --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing","execution_count":76,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAuto\n  --missing\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n279439 variants loaded from .bim file.\n217 people (0 males, 0 females, 217 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.nosex .\n217 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 217 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.978587.\n--missing: Sample missing data report written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.imiss, and variant-based\nmissing data report written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.lmiss.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{"trusted":true},"id":"linear-complement","cell_type":"code","source":"%%bash \nhead -n 10 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.imiss","execution_count":77,"outputs":[{"output_type":"stream","text":" FID          IID MISS_PHENO   N_MISS   N_GENO   F_MISS\nPuBa   GSM1094884          Y     5268   279439  0.01885\nPuBa   GSM1094885          Y     5091   279439  0.01822\nPuBa   GSM1094886          Y     5147   279439  0.01842\nPuBa   GSM1094887          Y     5019   279439  0.01796\nPuBa   GSM1094888          Y     4912   279439  0.01758\nPuBa   GSM1094889          Y     5925   279439   0.0212\nPuBa   GSM1094890          Y     5099   279439  0.01825\nPuBa   GSM1094891          Y     4935   279439  0.01766\nPuBa   GSM1094892          Y     4925   279439  0.01762\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"gothic-nickname","cell_type":"code","source":"%%bash\nhead -n 10 GreSalVL_b37_AllelesUpdatedNewMapBEDAutoMissing.lmiss\n","execution_count":78,"outputs":[{"output_type":"stream","text":" CHR           SNP   N_MISS   N_GENO   F_MISS\n   1      1_752566        1      217 0.004608\n   1      1_753541        0      217        0\n   1      1_770216        0      217        0\n   1      1_776546       15      217  0.06912\n   1      1_781258        0      217        0\n   1      1_791853        3      217  0.01382\n   1      1_799463        1      217 0.004608\n   1      1_808769        0      217        0\n   1      1_816617        0      217        0\n","name":"stdout"}]},{"metadata":{},"id":"further-spoke","cell_type":"markdown","source":"As we can easily notice, some SNPs have a quite high proprtion of missing data. Let's remove them using the `--geno`option. Similarly, we will remove individuals with a high number of missing data using `--mind`"},{"metadata":{"scrolled":true,"trusted":true},"id":"aquatic-behalf","cell_type":"code","source":"%%bash\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAuto --geno 0.05 --mind 0.05 --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05","execution_count":79,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAuto\n  --geno 0.05\n  --make-bed\n  --mind 0.05\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n279439 variants loaded from .bim file.\n217 people (0 males, 0 females, 217 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.nosex .\n217 phenotype values loaded from .fam.\n2 people removed due to missing genotype data (--mind).\nIDs written to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.irem .\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 215 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate in remaining samples is 0.980489.\n6397 variants removed due to missing genotype data (--geno).\n273042 variants and 215 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 164 are controls.\n--make-bed to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bed +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{},"id":"optimum-practitioner","cell_type":"markdown","source":"Also, we may want to analyse only the controls."},{"metadata":{"trusted":true},"id":"functioning-welsh","cell_type":"code","source":"%%bash \n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05  --filter-controls --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls","execution_count":80,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05\n  --filter-controls\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n273042 variants loaded from .bim file.\n215 people (0 males, 0 females, 215 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.nosex .\n215 phenotype values loaded from .fam.\n51 people removed due to case/control status (--filter-controls).\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 164 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate in remaining samples is 0.998108.\n273042 variants and 164 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 164 are controls.\n--make-bed to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bed\n+ GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{},"id":"loved-video","cell_type":"markdown","source":"One common routine is to check for SNPs strongly deviating from Hardy-Weinberg equilibrium, which are usually an indication of technical artifact"},{"metadata":{"trusted":true},"id":"elementary-salad","cell_type":"code","source":"%%bash \n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls --hwe 1e-8 --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE","execution_count":81,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05Controls\n  --hwe 1e-8\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n273042 variants loaded from .bim file.\n164 people (0 males, 0 females, 164 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.nosex .\n164 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 164 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.998108.\n--hwe: 232 variants removed due to Hardy-Weinberg exact test.\n272810 variants and 164 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 164 are controls.\n--make-bed to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bed +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.bim +\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{},"id":"earlier-composition","cell_type":"markdown","source":"We also want to remove close relatives, filtering one random sample from any pair of individuals having a similarity higher than 12.5%"},{"metadata":{"trusted":true},"id":"conditional-angola","cell_type":"code","source":"%%bash \n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE --rel-cutoff .125 --allow-no-sex --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved","execution_count":82,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.log.\nOptions in effect:\n  --allow-no-sex\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWE\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved\n  --rel-cutoff .125\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n272810 variants loaded from .bim file.\n164 people (0 males, 0 females, 164 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.nosex\n.\n164 phenotype values loaded from .fam.\nUsing up to 8 threads (change this with --threads).\nBefore main variant filters, 164 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.998112.\n272810 variants and 164 people pass filters and QC (before --rel-cutoff).\nAmong remaining phenotypes, 0 are cases and 164 are controls.\n\r60 markers complete.\r120 markers complete.\r180 markers complete.\r240 markers complete.\r300 markers complete.\r360 markers complete.\r420 markers complete.\r480 markers complete.\r540 markers complete.\r600 markers complete.\r660 markers complete.\r720 markers complete.\r780 markers complete.\r840 markers complete.\r900 markers complete.\r960 markers complete.\r1020 markers complete.\r1080 markers complete.\r1140 markers complete.\r1200 markers complete.\r1260 markers complete.\r1320 markers complete.\r1380 markers complete.\r1440 markers complete.\r1500 markers complete.\r1560 markers complete.\r1620 markers complete.\r1680 markers complete.\r1740 markers complete.\r1800 markers complete.\r1860 markers complete.\r1920 markers complete.\r1980 markers complete.\r2040 markers complete.\r2100 markers complete.\r2160 markers complete.\r2220 markers complete.\r2280 markers complete.\r2340 markers complete.\r2400 markers complete.\r2460 markers complete.\r2520 markers complete.\r2580 markers complete.\r2640 markers complete.\r2700 markers complete.\r2760 markers complete.\r2820 markers complete.\r2880 markers complete.\r2940 markers complete.\r3000 markers complete.\r3060 markers complete.\r3120 markers complete.\r3180 markers complete.\r3240 markers complete.\r3300 markers complete.\r3360 markers complete.\r3420 markers complete.\r3480 markers complete.\r3540 markers complete.\r3600 markers complete.\r3660 markers complete.\r3720 markers complete.\r3780 markers complete.\r3840 markers complete.\r3900 markers complete.\r3960 markers complete.\r4020 markers complete.\r4080 markers complete.\r4140 markers complete.\r4200 markers complete.\r4260 markers complete.\r4320 markers complete.\r4380 markers complete.\r4440 markers complete.\r4500 markers complete.\r4560 markers complete.\r4620 markers complete.\r4680 markers complete.\r4740 markers complete.\r4800 markers complete.\r4860 markers complete.\r4920 markers complete.\r4980 markers complete.\r5040 markers complete.\r5100 markers complete.\r5160 markers complete.\r5220 markers complete.\r5280 markers complete.\r5340 markers complete.\r5400 markers complete.\r5460 markers complete.\r5520 markers complete.\r5580 markers complete.\r5640 markers complete.\r5700 markers complete.\r5760 markers complete.\r5820 markers complete.\r5880 markers complete.\r5940 markers complete.\r6000 markers complete.\r6060 markers complete.\r6120 markers complete.\r6180 markers complete.\r6240 markers complete.\r6300 markers complete.\r6360 markers complete.\r6420 markers complete.\r6480 markers complete.\r6540 markers complete.\r6600 markers complete.\r6660 markers complete.\r6720 markers complete.\r6780 markers complete.\r6840 markers complete.\r6900 markers complete.\r6960 markers complete.\r7020 markers complete.\r7080 markers complete.\r7140 markers complete.\r7200 markers complete.\r7260 markers complete.\r7320 markers complete.\r7380 markers complete.\r7440 markers complete.\r7500 markers complete.\r7560 markers complete.\r7620 markers complete.\r7680 markers complete.\r7740 markers complete.\r7800 markers complete.\r7860 markers complete.\r7920 markers complete.\r7980 markers complete.\r8040 markers complete.\r8100 markers complete.\r8160 markers complete.\r8220 markers complete.\r8280 markers complete.\r8340 markers complete.\r8400 markers complete.\r8460 markers complete.\r8520 markers complete.\r8580 markers complete.\r8640 markers complete.\r8700 markers complete.\r8760 markers complete.\r8820 markers complete.\r8880 markers complete.\r8940 markers complete.\r9000 markers complete.\r9060 markers complete.\r9120 markers complete.\r9180 markers complete.\r9240 markers complete.\r9300 markers complete.\r9360 markers complete.\r9420 markers complete.\r9480 markers complete.\r9540 markers complete.\r9600 markers complete.\r9660 markers complete.\r9720 markers complete.\r9780 markers complete.\r9840 markers complete.\r9900 markers complete.\r9960 markers complete.\r10020 markers complete.\r10080 markers complete.\r10140 markers complete.\r10200 markers complete.\r10260 markers complete.\r10320 markers complete.\r10380 markers complete.\r10440 markers complete.\r10500 markers complete.\r10560 markers complete.\r10620 markers complete.\r10680 markers complete.\r10740 markers complete.\r10800 markers complete.\r10860 markers complete.\r10920 markers complete.\r10980 markers complete.\r11040 markers complete.\r11100 markers complete.\r11160 markers complete.\r11220 markers complete.\r11280 markers complete.\r11340 markers complete.\r11400 markers complete.\r11460 markers complete.\r11520 markers complete.\r11580 markers complete.\r11640 markers complete.\r11700 markers complete.\r11760 markers complete.\r11820 markers complete.\r11880 markers complete.\r11940 markers complete.\r12000 markers complete.\r12060 markers complete.\r12120 markers complete.\r12180 markers complete.\r12240 markers complete.\r12300 markers complete.\r12360 markers complete.\r12420 markers complete.\r12480 markers complete.\r12540 markers complete.\r12600 markers complete.\r12660 markers complete.\r12720 markers complete.\r12780 markers complete.\r12840 markers complete.\r12900 markers complete.\r12960 markers complete.\r13020 markers complete.\r13080 markers complete.\r13140 markers complete.\r13200 markers complete.\r13260 markers complete.\r13320 markers complete.\r13380 markers complete.\r13440 markers complete.\r13500 markers complete.\r13560 markers complete.\r13620 markers complete.\r13680 markers complete.\r13740 markers complete.\r13800 markers complete.\r13860 markers complete.\r13920 markers complete.\r13980 markers complete.\r14040 markers complete.\r14100 markers complete.\r14160 markers complete.\r14220 markers complete.\r14280 markers complete.\r14340 markers complete.\r14400 markers complete.\r14460 markers complete.\r14520 markers complete.\r14580 markers complete.\r14640 markers complete.\r14700 markers complete.\r14760 markers complete.\r14820 markers complete.\r14880 markers complete.\r14940 markers complete.\r15000 markers complete.\r15060 markers complete.\r15120 markers complete.\r15180 markers complete.\r15240 markers complete.\r15300 markers complete.\r15360 markers complete.\r15420 markers complete.\r15480 markers complete.\r15540 markers complete.\r15600 markers complete.\r15660 markers complete.\r15720 markers complete.\r15780 markers complete.\r15840 markers complete.\r15900 markers complete.\r15960 markers complete.\r16020 markers complete.\r16080 markers complete.\r16140 markers complete.\r16200 markers complete.\r16260 markers complete.\r16320 markers complete.\r16380 markers complete.\r16440 markers complete.\r16500 markers complete.\r16560 markers complete.\r16620 markers complete.\r16680 markers complete.\r16740 markers complete.\r16800 markers complete.\r16860 markers complete.\r16920 markers complete.\r16980 markers complete.\r17040 markers complete.\r17100 markers complete.\r17160 markers complete.\r17220 markers complete.\r17280 markers complete.\r17340 markers complete.\r17400 markers complete.\r17460 markers complete.\r17520 markers complete.\r17580 markers complete.\r17640 markers complete.\r17700 markers complete.\r17760 markers complete.\r17820 markers complete.\r17880 markers complete.\r17940 markers complete.\r18000 markers complete.\r18060 markers complete.\r18120 markers complete.\r18180 markers complete.\r18240 markers complete.\r18300 markers complete.\r18360 markers complete.\r18420 markers complete.\r18480 markers complete.\r18540 markers complete.\r18600 markers complete.\r18660 markers complete.\r18720 markers complete.\r18780 markers complete.\r18840 markers complete.\r18900 markers complete.\r18960 markers complete.\r19020 markers complete.\r19080 markers complete.\r19140 markers complete.\r19200 markers complete.\r19260 markers complete.\r19320 markers complete.\r19380 markers complete.\r19440 markers complete.\r19500 markers complete.\r19560 markers complete.\r19620 markers complete.\r19680 markers complete.\r19740 markers complete.\r19800 markers complete.\r19860 markers complete.\r19920 markers complete.\r19980 markers complete.\r20040 markers complete.\r20100 markers complete.\r20160 markers complete.\r20220 markers complete.\r20280 markers complete.\r20340 markers complete.\r20400 markers complete.\r20460 markers complete.\r20520 markers complete.\r20580 markers complete.\r20640 markers complete.\r20700 markers complete.\r20760 markers complete.\r20820 markers complete.\r20880 markers complete.\r20940 markers complete.\r21000 markers complete.\r21060 markers complete.\r21120 markers complete.\r21180 markers complete.\r21240 markers complete.\r21300 markers complete.\r21360 markers complete.\r21420 markers complete.\r21480 markers complete.\r21540 markers complete.\r21600 markers complete.\r21660 markers complete.\r21720 markers complete.\r21780 markers complete.\r21840 markers complete.\r21900 markers complete.\r21960 markers complete.\r22020 markers complete.\r22080 markers complete.\r22140 markers complete.\r22200 markers complete.\r22260 markers complete.\r22320 markers complete.\r22380 markers complete.\r22440 markers complete.\r22500 markers complete.\r22560 markers complete.\r22620 markers complete.\r22680 markers complete.\r22740 markers complete.\r22800 markers complete.\r22860 markers complete.\r22920 markers complete.\r22980 markers complete.\r23040 markers complete.\r23100 markers complete.\r23160 markers complete.\r23220 markers complete.\r23280 markers complete.\r23340 markers complete.\r23400 markers complete.\r23460 markers complete.\r23520 markers complete.\r23580 markers complete.\r23640 markers complete.\r23700 markers complete.\r23760 markers complete.\r23820 markers complete.\r23880 markers complete.\r23940 markers complete.\r24000 markers complete.\r24060 markers complete.\r24120 markers complete.\r24180 markers complete.\r24240 markers complete.\r24300 markers complete.\r24360 markers complete.\r24420 markers complete.\r24480 markers complete.\r24540 markers complete.\r24600 markers complete.\r24660 markers complete.\r24720 markers complete.\r24780 markers complete.\r24840 markers complete.\r24900 markers complete.\r24960 markers complete.\r25020 markers complete.\r25080 markers complete.\r25140 markers complete.\r25200 markers complete.\r25260 markers complete.\r25320 markers complete.\r25380 markers complete.\r25440 markers complete.\r25500 markers complete.\r25560 markers complete.\r25620 markers complete.\r25680 markers complete.\r25740 markers complete.\r25800 markers complete.\r25860 markers complete.\r25920 markers complete.\r25980 markers complete.\r26040 markers complete.\r26100 markers complete.\r26160 markers complete.\r26220 markers complete.\r26280 markers complete.\r26340 markers complete.\r26400 markers complete.\r26460 markers complete.\r26520 markers complete.\r26580 markers complete.\r26640 markers complete.\r26700 markers complete.\r26760 markers complete.\r26820 markers complete.\r26880 markers complete.\r26940 markers complete.\r27000 markers complete.\r27060 markers complete.\r27120 markers complete.\r27180 markers complete.\r27240 markers complete.\r27300 markers complete.\r27360 markers complete.\r27420 markers complete.\r27480 markers complete.\r27540 markers complete.\r27600 markers complete.\r27660 markers complete.\r27720 markers complete.\r27780 markers complete.\r27840 markers complete.\r27900 markers complete.\r27960 markers complete.\r28020 markers complete.\r28080 markers complete.\r28140 markers complete.\r28200 markers complete.\r28260 markers complete.\r28320 markers complete.\r28380 markers complete.\r28440 markers complete.\r28500 markers complete.\r28560 markers complete.\r28620 markers complete.\r28680 markers complete.\r28740 markers complete.\r28800 markers complete.\r28860 markers complete.\r28920 markers complete.\r28980 markers complete.\r29040 markers complete.\r29100 markers complete.\r29160 markers complete.\r29220 markers complete.\r29280 markers complete.\r29340 markers complete.\r29400 markers complete.\r29460 markers complete.\r29520 markers complete.\r29580 markers complete.\r29640 markers complete.\r29700 markers complete.\r29760 markers complete.\r29820 markers complete.\r29880 markers complete.\r29940 markers complete.\r30000 markers complete.\r30060 markers complete.\r30120 markers complete.\r30180 markers complete.\r30240 markers complete.\r30300 markers complete.\r30360 markers complete.\r30420 markers complete.\r30480 markers complete.\r30540 markers complete.\r30600 markers complete.\r30660 markers complete.\r30720 markers complete.\r30780 markers complete.\r30840 markers complete.\r30900 markers complete.\r30960 markers complete.\r31020 markers complete.\r31080 markers complete.\r31140 markers complete.\r31200 markers complete.\r31260 markers complete.\r31320 markers complete.\r31380 markers complete.\r31440 markers complete.\r31500 markers complete.\r31560 markers complete.\r31620 markers complete.\r31680 markers complete.\r31740 markers complete.\r31800 markers complete.\r31860 markers complete.\r31920 markers complete.\r31980 markers complete.\r32040 markers complete.\r32100 markers complete.\r32160 markers complete.\r32220 markers complete.\r32280 markers complete.\r32340 markers complete.\r32400 markers complete.\r32460 markers complete.\r32520 markers complete.\r32580 markers complete.\r32640 markers complete.\r32700 markers complete.\r32760 markers complete.\r32820 markers complete.\r32880 markers complete.\r32940 markers complete.\r33000 markers complete.\r33060 markers complete.\r33120 markers complete.\r33180 markers complete.\r33240 markers complete.\r33300 markers complete.\r33360 markers complete.\r33420 markers complete.\r33480 markers complete.\r33540 markers complete.\r33600 markers complete.\r33660 markers complete.\r33720 markers complete.\r33780 markers complete.\r33840 markers complete.\r33900 markers complete.\r33960 markers complete.\r34020 markers complete.\r34080 markers complete.\r34140 markers complete.\r34200 markers complete.\r34260 markers complete.\r34320 markers complete.\r34380 markers complete.\r34440 markers complete.\r34500 markers complete.\r34560 markers complete.\r34620 markers complete.\r34680 markers complete.\r34740 markers complete.\r34800 markers complete.\r34860 markers complete.\r34920 markers complete.\r34980 markers complete.\r35040 markers complete.\r35100 markers complete.\r35160 markers complete.\r35220 markers complete.\r35280 markers complete.\r35340 markers complete.\r35400 markers complete.\r35460 markers complete.\r35520 markers complete.\r35580 markers complete.\r35640 markers complete.\r35700 markers complete.\r35760 markers complete.\r35820 markers complete.\r35880 markers complete.\r35940 markers complete.\r36000 markers complete.\r36060 markers complete.\r36120 markers complete.\r36180 markers complete.\r36240 markers complete.\r36300 markers complete.\r36360 markers complete.\r36420 markers complete.\r36480 markers complete.\r36540 markers complete.\r36600 markers complete.\r36660 markers complete.\r36720 markers complete.\r36780 markers complete.\r36840 markers complete.\r36900 markers complete.\r36960 markers complete.\r37020 markers complete.\r37080 markers complete.\r37140 markers complete.\r37200 markers complete.\r37260 markers complete.\r37320 markers complete.\r37380 markers complete.\r37440 markers complete.\r37500 markers complete.\r37560 markers complete.\r37620 markers complete.\r37680 markers complete.\r37740 markers complete.\r37800 markers complete.\r37860 markers complete.\r37920 markers complete.\r37980 markers complete.\r38040 markers complete.\r38100 markers complete.\r38160 markers complete.\r38220 markers complete.\r38280 markers complete.\r38340 markers complete.\r38400 markers complete.\r38460 markers complete.\r38520 markers complete.\r38580 markers complete.\r38640 markers complete.\r38700 markers complete.\r38760 markers complete.\r38820 markers complete.\r38880 markers complete.\r38940 markers complete.\r39000 markers complete.\r39060 markers complete.\r39120 markers complete.\r39180 markers complete.\r39240 markers complete.\r39300 markers complete.\r39360 markers complete.\r39420 markers complete.\r39480 markers complete.\r39540 markers complete.\r39600 markers complete.\r39660 markers complete.\r39720 markers complete.\r39780 markers complete.\r39840 markers complete.\r39900 markers complete.\r39960 markers complete.\r40020 markers complete.\r40080 markers complete.\r40140 markers complete.\r40200 markers complete.\r40260 markers complete.\r40320 markers complete.\r40380 markers complete.\r40440 markers complete.\r40500 markers complete.\r40560 markers complete.\r40620 markers complete.\r40680 markers complete.\r40740 markers complete.\r40800 markers complete.\r40860 markers complete.\r40920 markers complete.\r40980 markers complete.\r41040 markers complete.\r41100 markers complete.\r41160 markers complete.\r41220 markers complete.\r41280 markers complete.\r41340 markers complete.\r41400 markers complete.\r41460 markers complete.\r41520 markers complete.\r41580 markers complete.\r41640 markers complete.\r41700 markers complete.\r41760 markers complete.\r41820 markers complete.\r41880 markers complete.\r41940 markers complete.\r42000 markers complete.\r42060 markers complete.\r42120 markers complete.\r42180 markers complete.\r42240 markers complete.\r42300 markers complete.\r42360 markers complete.\r42420 markers complete.\r42480 markers complete.\r42540 markers complete.\r42600 markers complete.\r42660 markers complete.\r42720 markers complete.\r42780 markers complete.\r42840 markers complete.\r42900 markers complete.\r42960 markers complete.\r43020 markers complete.\r43080 markers complete.\r43140 markers complete.\r43200 markers complete.\r43260 markers complete.\r43320 markers complete.\r43380 markers complete.\r43440 markers complete.\r43500 markers complete.\r43560 markers complete.\r43620 markers complete.\r43680 markers complete.\r43740 markers complete.\r43800 markers complete.\r43860 markers complete.\r43920 markers complete.\r43980 markers complete.\r44040 markers complete.\r44100 markers complete.\r44160 markers complete.\r44220 markers complete.\r44280 markers complete.\r44340 markers complete.\r44400 markers complete.\r44460 markers complete.\r44520 markers complete.\r44580 markers complete.\r44640 markers complete.\r44700 markers complete.\r44760 markers complete.\r44820 markers complete.\r44880 markers complete.\r44940 markers complete.\r45000 markers complete.\r45060 markers complete.\r45120 markers complete.\r45180 markers complete.\r45240 markers complete.\r45300 markers complete.\r45360 markers complete.\r45420 markers complete.\r45480 markers complete.\r45540 markers complete.\r45600 markers complete.\r45660 markers complete.\r45720 markers complete.\r45780 markers complete.\r45840 markers complete.\r45900 markers complete.\r45960 markers complete.\r46020 markers complete.\r46080 markers complete.\r46140 markers complete.\r46200 markers complete.\r46260 markers complete.\r46320 markers complete.\r46380 markers complete.\r46440 markers complete.\r46500 markers complete.\r46560 markers complete.\r46620 markers complete.\r46680 markers complete.\r46740 markers complete.\r46800 markers complete.\r46860 markers complete.\r46920 markers complete.\r46980 markers complete.\r47040 markers complete.\r47100 markers complete.\r47160 markers complete.\r47220 markers complete.\r47280 markers complete.\r47340 markers complete.\r47400 markers complete.\r47460 markers complete.\r47520 markers complete.\r47580 markers complete.\r47640 markers complete.\r47700 markers complete.\r47760 markers complete.\r47820 markers complete.\r47880 markers complete.\r47940 markers complete.\r48000 markers complete.\r48060 markers complete.\r48120 markers complete.\r48180 markers complete.\r48240 markers complete.\r48300 markers complete.\r48360 markers complete.\r48420 markers complete.\r48480 markers complete.\r48540 markers complete.\r48600 markers complete.\r48660 markers complete.\r48720 markers complete.\r48780 markers complete.\r48840 markers complete.\r48900 markers complete.\r48960 markers complete.\r49020 markers complete.\r49080 markers complete.\r49140 markers complete.\r49200 markers complete.\r49260 markers complete.\r49320 markers complete.\r49380 markers complete.\r49440 markers complete.\r49500 markers complete.\r49560 markers complete.\r49620 markers complete.\r49680 markers complete.\r49740 markers complete.\r49800 markers complete.\r49860 markers complete.\r49920 markers complete.\r49980 markers complete.\r50040 markers complete.\r50100 markers complete.\r50160 markers complete.\r50220 markers complete.\r50280 markers complete.\r50340 markers complete.\r50400 markers complete.\r50460 markers complete.\r50520 markers complete.\r50580 markers complete.\r50640 markers complete.\r50700 markers complete.\r50760 markers complete.\r50820 markers complete.\r50880 markers complete.\r50940 markers complete.\r51000 markers complete.\r51060 markers complete.\r51120 markers complete.\r51180 markers complete.\r51240 markers complete.\r51300 markers complete.\r51360 markers complete.\r51420 markers complete.\r51480 markers complete.\r51540 markers complete.\r51600 markers complete.\r51660 markers complete.\r51720 markers complete.\r51780 markers complete.\r51840 markers complete.\r51900 markers complete.\r51960 markers complete.\r52020 markers complete.\r52080 markers complete.\r52140 markers complete.\r52200 markers complete.\r52260 markers complete.\r52320 markers complete.\r52380 markers complete.\r52440 markers complete.\r52500 markers complete.\r52560 markers complete.\r52620 markers complete.\r52680 markers complete.\r52740 markers complete.\r52800 markers complete.\r52860 markers complete.\r52920 markers complete.\r52980 markers complete.\r53040 markers complete.\r53100 markers complete.\r53160 markers complete.\r53220 markers complete.\r53280 markers complete.\r53340 markers complete.\r53400 markers complete.\r53460 markers complete.\r53520 markers complete.\r53580 markers complete.\r53640 markers complete.\r53700 markers complete.\r53760 markers complete.\r53820 markers complete.\r53880 markers complete.\r53940 markers complete.\r54000 markers complete.\r54060 markers complete.\r54120 markers complete.\r54180 markers complete.\r54240 markers complete.\r54300 markers complete.\r54360 markers complete.\r54420 markers complete.\r54480 markers complete.\r54540 markers complete.\r54600 markers complete.\r54660 markers complete.\r54720 markers complete.\r54780 markers complete.\r54840 markers complete.\r54900 markers complete.\r54960 markers complete.\r55020 markers complete.\r55080 markers complete.\r55140 markers complete.\r55200 markers complete.\r55260 markers complete.\r55320 markers complete.\r55380 markers complete.\r55440 markers complete.\r55500 markers complete.\r55560 markers complete.\r55620 markers complete.\r55680 markers complete.\r55740 markers complete.\r55800 markers complete.\r55860 markers complete.\r55920 markers complete.\r55980 markers complete.\r56040 markers complete.\r56100 markers complete.\r56160 markers complete.\r56220 markers complete.\r56280 markers complete.\r56340 markers complete.\r56400 markers complete.\r56460 markers complete.\r56520 markers complete.\r56580 markers complete.\r56640 markers complete.\r56700 markers complete.\r56760 markers complete.\r56820 markers complete.\r56880 markers complete.\r56940 markers complete.\r57000 markers complete.\r57060 markers complete.\r57120 markers complete.\r57180 markers complete.\r57240 markers complete.\r57300 markers complete.\r57360 markers complete.\r57420 markers complete.\r57480 markers complete.\r57540 markers complete.\r57600 markers complete.\r57660 markers complete.\r57720 markers complete.\r57780 markers complete.\r57840 markers complete.\r57900 markers complete.\r57960 markers complete.\r58020 markers complete.\r58080 markers complete.\r58140 markers complete.\r58200 markers complete.\r58260 markers complete.\r58320 markers complete.\r58380 markers complete.\r58440 markers complete.\r58500 markers complete.\r58560 markers complete.\r58620 markers complete.\r58680 markers complete.\r58740 markers complete.\r58800 markers complete.\r58860 markers complete.\r58920 markers complete.\r58980 markers complete.\r59040 markers complete.\r59100 markers complete.\r59160 markers complete.\r59220 markers complete.\r59280 markers complete.\r59340 markers complete.\r59400 markers complete.\r59460 markers complete.\r59520 markers complete.\r59580 markers complete.\r59640 markers complete.\r59700 markers complete.\r59760 markers complete.\r59820 markers complete.\r59880 markers complete.\r59940 markers complete.\r60000 markers complete.\r60060 markers complete.\r60120 markers complete.\r60180 markers complete.\r60240 markers complete.\r60300 markers complete.\r60360 markers complete.\r60420 markers complete.\r60480 markers complete.\r60540 markers complete.\r60600 markers complete.\r60660 markers complete.\r60720 markers complete.\r60780 markers complete.\r60840 markers complete.\r60900 markers complete.\r60960 markers complete.\r61020 markers complete.\r61080 markers complete.\r61140 markers complete.\r61200 markers complete.\r61260 markers complete.\r61320 markers complete.\r61380 markers complete.\r61440 markers complete.\r61500 markers complete.\r61560 markers complete.\r61620 markers complete.\r61680 markers complete.\r61740 markers complete.\r61800 markers complete.\r61860 markers complete.\r61920 markers complete.\r61980 markers complete.\r62040 markers complete.\r62100 markers complete.\r62160 markers complete.\r62220 markers complete.\r62280 markers complete.\r62340 markers complete.\r62400 markers complete.\r62460 markers complete.\r62520 markers complete.\r62580 markers complete.\r62640 markers complete.\r62700 markers complete.\r62760 markers complete.\r62820 markers complete.\r62880 markers complete.\r62940 markers complete.\r63000 markers complete.\r63060 markers complete.\r63120 markers complete.\r63180 markers complete.\r63240 markers complete.\r63300 markers complete.\r63360 markers complete.\r63420 markers complete.\r63480 markers complete.\r63540 markers complete.\r63600 markers complete.\r63660 markers complete.\r63720 markers complete.\r63780 markers complete.\r63840 markers complete.\r63900 markers complete.\r63960 markers complete.\r64020 markers complete.\r64080 markers complete.\r64140 markers complete.\r64200 markers complete.\r64260 markers complete.\r64320 markers complete.\r64380 markers complete.\r64440 markers complete.\r64500 markers complete.\r64560 markers complete.\r64620 markers complete.\r64680 markers complete.\r64740 markers complete.\r64800 markers complete.\r64860 markers complete.\r64920 markers complete.\r64980 markers complete.\r65040 markers complete.\r65100 markers complete.\r65160 markers complete.\r65220 markers complete.\r65280 markers complete.\r65340 markers complete.\r65400 markers complete.\r65460 markers complete.\r65520 markers complete.\r65580 markers complete.\r65640 markers complete.\r65700 markers complete.\r65760 markers complete.\r65820 markers complete.\r65880 markers complete.\r65940 markers complete.\r66000 markers complete.\r66060 markers complete.\r66120 markers complete.\r66180 markers complete.\r66240 markers complete.\r66300 markers complete.\r66360 markers complete.\r66420 markers complete.\r66480 markers complete.\r66540 markers complete.\r66600 markers complete.\r66660 markers complete.\r66720 markers complete.\r66780 markers complete.\r66840 markers complete.\r66900 markers complete.\r66960 markers complete.\r67020 markers complete.\r67080 markers complete.\r67140 markers complete.\r67200 markers complete.\r67260 markers complete.\r67320 markers complete.\r67380 markers complete.\r67440 markers complete.\r67500 markers complete.\r67560 markers complete.\r67620 markers complete.\r67680 markers complete.\r67740 markers complete.\r67800 markers complete.\r67860 markers complete.\r67920 markers complete.\r67980 markers complete.\r68040 markers complete.\r68100 markers complete.\r68160 markers complete.\r68220 markers complete.\r68280 markers complete.\r68340 markers complete.\r68400 markers complete.\r68460 markers complete.\r68520 markers complete.\r68580 markers complete.\r68640 markers complete.\r68700 markers complete.\r68760 markers complete.\r68820 markers complete.\r68880 markers complete.\r68940 markers complete.\r69000 markers complete.\r69060 markers complete.\r69120 markers complete.\r69180 markers complete.\r69240 markers complete.\r69300 markers complete.\r69360 markers complete.\r69420 markers complete.\r69480 markers complete.\r69540 markers complete.\r69600 markers complete.\r69660 markers complete.\r69720 markers complete.\r69780 markers complete.\r69840 markers complete.\r69900 markers complete.\r69960 markers complete.\r70020 markers complete.\r70080 markers complete.\r70140 markers complete.\r70200 markers complete.\r70260 markers complete.\r70320 markers complete.\r70380 markers complete.\r70440 markers complete.\r70500 markers complete.\r70560 markers complete.\r70620 markers complete.\r70680 markers complete.\r70740 markers complete.\r70800 markers complete.\r70860 markers complete.\r70920 markers complete.\r70980 markers complete.\r71040 markers complete.\r71100 markers complete.\r71160 markers complete.\r71220 markers complete.\r71280 markers complete.\r71340 markers complete.\r71400 markers complete.\r71460 markers complete.\r71520 markers complete.\r71580 markers complete.\r71640 markers complete.\r71700 markers complete.\r71760 markers complete.\r71820 markers complete.\r71880 markers complete.\r71940 markers complete.\r72000 markers complete.\r72060 markers complete.\r72120 markers complete.\r72180 markers complete.\r72240 markers complete.\r72300 markers complete.\r72360 markers complete.\r72420 markers complete.\r72480 markers complete.\r72540 markers complete.\r72600 markers complete.\r72660 markers complete.\r72720 markers complete.\r72780 markers complete.\r72840 markers complete.\r72900 markers complete.\r72960 markers complete.\r73020 markers complete.\r73080 markers complete.\r73140 markers complete.\r73200 markers complete.\r73260 markers complete.\r73320 markers complete.\r73380 markers complete.\r73440 markers complete.\r73500 markers complete.\r73560 markers complete.\r73620 markers complete.\r73680 markers complete.\r73740 markers complete.\r73800 markers complete.\r73860 markers complete.\r73920 markers complete.\r73980 markers complete.\r74040 markers complete.\r74100 markers complete.\r74160 markers complete.\r74220 markers complete.\r74280 markers complete.\r74340 markers complete.\r74400 markers complete.\r74460 markers complete.\r74520 markers complete.\r74580 markers complete.\r74640 markers complete.\r74700 markers complete.\r74760 markers complete.\r74820 markers complete.\r74880 markers complete.\r74940 markers complete.\r75000 markers complete.\r75060 markers complete.\r75120 markers complete.\r75180 markers complete.\r75240 markers complete.\r75300 markers complete.\r75360 markers complete.\r75420 markers complete.\r75480 markers complete.\r75540 markers complete.\r75600 markers complete.\r75660 markers complete.\r75720 markers complete.\r75780 markers complete.\r75840 markers complete.\r75900 markers complete.\r75960 markers complete.\r76020 markers complete.\r76080 markers complete.\r76140 markers complete.\r76200 markers complete.\r76260 markers complete.\r76320 markers complete.\r76380 markers complete.\r76440 markers complete.\r76500 markers complete.\r76560 markers complete.\r76620 markers complete.\r76680 markers complete.\r76740 markers complete.\r76800 markers complete.\r76860 markers complete.\r76920 markers complete.\r76980 markers complete.\r77040 markers complete.\r77100 markers complete.\r77160 markers complete.\r77220 markers complete.\r77280 markers complete.\r77340 markers complete.\r77400 markers complete.\r77460 markers complete.\r77520 markers complete.\r77580 markers complete.\r77640 markers complete.\r77700 markers complete.\r77760 markers complete.\r77820 markers complete.\r77880 markers complete.\r77940 markers complete.\r78000 markers complete.\r78060 markers complete.\r78120 markers complete.\r78180 markers complete.\r78240 markers complete.\r78300 markers complete.\r78360 markers complete.\r78420 markers complete.\r78480 markers complete.\r78540 markers complete.\r78600 markers complete.\r78660 markers complete.\r78720 markers complete.\r78780 markers complete.\r78840 markers complete.\r78900 markers complete.\r78960 markers complete.\r79020 markers complete.\r79080 markers complete.\r79140 markers complete.\r79200 markers complete.\r79260 markers complete.\r79320 markers complete.\r79380 markers complete.\r79440 markers complete.\r79500 markers complete.\r79560 markers complete.\r79620 markers complete.\r79680 markers complete.\r79740 markers complete.\r79800 markers complete.\r79860 markers complete.\r79920 markers complete.\r79980 markers complete.\r80040 markers complete.\r80100 markers complete.\r80160 markers complete.\r80220 markers complete.\r80280 markers complete.\r80340 markers complete.\r80400 markers complete.\r80460 markers complete.\r80520 markers complete.\r80580 markers complete.\r80640 markers complete.\r80700 markers complete.\r80760 markers complete.\r80820 markers complete.\r80880 markers complete.\r80940 markers complete.\r81000 markers complete.\r81060 markers complete.\r81120 markers complete.\r81180 markers complete.\r81240 markers complete.\r81300 markers complete.\r81360 markers complete.\r81420 markers complete.\r81480 markers complete.\r81540 markers complete.\r81600 markers complete.\r81660 markers complete.\r81720 markers complete.\r81780 markers complete.\r81840 markers complete.\r81900 markers complete.\r81960 markers complete.\r82020 markers complete.\r82080 markers complete.\r82140 markers complete.\r82200 markers complete.\r82260 markers complete.\r82320 markers complete.\r82380 markers complete.\r82440 markers complete.\r82500 markers complete.\r82560 markers complete.\r82620 markers complete.\r82680 markers complete.\r82740 markers complete.\r82800 markers complete.\r82860 markers complete.\r82920 markers complete.\r82980 markers complete.\r83040 markers complete.\r83100 markers complete.\r83160 markers complete.\r83220 markers complete.\r83280 markers complete.\r83340 markers complete.\r83400 markers complete.\r83460 markers complete.\r83520 markers complete.\r83580 markers complete.\r83640 markers complete.\r83700 markers complete.\r83760 markers complete.\r83820 markers complete.\r83880 markers complete.\r83940 markers complete.\r84000 markers complete.\r84060 markers complete.\r84120 markers complete.\r84180 markers complete.\r84240 markers complete.\r84300 markers complete.\r84360 markers complete.\r84420 markers complete.\r84480 markers complete.\r84540 markers complete.\r84600 markers complete.\r84660 markers complete.\r84720 markers complete.\r84780 markers complete.\r84840 markers complete.\r84900 markers complete.\r84960 markers complete.\r85020 markers complete.\r85080 markers complete.\r85140 markers complete.\r85200 markers complete.\r85260 markers complete.\r85320 markers complete.\r85380 markers complete.\r85440 markers complete.\r85500 markers complete.\r85560 markers complete.\r85620 markers complete.\r85680 markers complete.\r85740 markers complete.\r85800 markers complete.\r85860 markers complete.\r85920 markers complete.\r85980 markers complete.\r86040 markers complete.\r86100 markers complete.\r86160 markers complete.\r86220 markers complete.\r86280 markers complete.\r86340 markers complete.\r86400 markers complete.\r86460 markers complete.\r86520 markers complete.\r86580 markers complete.\r86640 markers complete.\r86700 markers complete.\r86760 markers complete.\r86820 markers complete.\r86880 markers complete.\r86940 markers complete.\r87000 markers complete.\r87060 markers complete.\r87120 markers complete.\r87180 markers complete.\r87240 markers complete.\r87300 markers complete.\r87360 markers complete.\r87420 markers complete.\r87480 markers complete.\r87540 markers complete.\r87600 markers complete.\r87660 markers complete.\r87720 markers complete.\r87780 markers complete.\r87840 markers complete.\r87900 markers complete.\r87960 markers complete.\r88020 markers complete.\r88080 markers complete.\r88140 markers complete.\r88200 markers complete.\r88260 markers complete.\r88320 markers complete.\r88380 markers complete.\r88440 markers complete.\r88500 markers complete.\r88560 markers complete.\r88620 markers complete.\r88680 markers complete.\r88740 markers complete.\r88800 markers complete.\r88860 markers complete.\r88920 markers complete.\r88980 markers complete.\r89040 markers complete.\r89100 markers complete.\r89160 markers complete.\r89220 markers complete.\r89280 markers complete.\r89340 markers complete.\r89400 markers complete.\r89460 markers complete.\r89520 markers complete.\r89580 markers complete.\r89640 markers complete.\r89700 markers complete.\r89760 markers complete.\r89820 markers complete.\r89880 markers complete.\r89940 markers complete.\r90000 markers complete.\r90060 markers complete.\r90120 markers complete.\r90180 markers complete.\r90240 markers complete.\r90300 markers complete.\r90360 markers complete.\r90420 markers complete.\r90480 markers complete.\r90540 markers complete.\r90600 markers complete.\r90660 markers complete.\r90720 markers complete.\r90780 markers complete.\r90840 markers complete.\r90900 markers complete.\r90960 markers complete.\r91020 markers complete.\r91080 markers complete.\r91140 markers complete.\r91200 markers complete.\r91260 markers complete.\r91320 markers complete.\r91380 markers complete.\r91440 markers complete.\r91500 markers complete.\r91560 markers complete.\r91620 markers complete.\r91680 markers complete.\r91740 markers complete.\r91800 markers complete.\r91860 markers complete.\r91920 markers complete.\r91980 markers complete.\r92040 markers complete.\r92100 markers complete.\r92160 markers complete.\r92220 markers complete.\r92280 markers complete.\r92340 markers complete.\r92400 markers complete.\r92460 markers complete.\r92520 markers complete.\r92580 markers complete.\r92640 markers complete.\r92700 markers complete.\r92760 markers complete.\r92820 markers complete.\r92880 markers complete.\r92940 markers complete.\r93000 markers complete.\r93060 markers complete.\r93120 markers complete.\r93180 markers complete.\r93240 markers complete.\r93300 markers complete.\r93360 markers complete.\r93420 markers complete.\r93480 markers complete.\r93540 markers complete.\r93600 markers complete.\r93660 markers complete.\r93720 markers complete.\r93780 markers complete.\r93840 markers complete.\r93900 markers complete.\r93960 markers complete.\r94020 markers complete.\r94080 markers complete.\r94140 markers complete.\r94200 markers complete.\r94260 markers complete.\r94320 markers complete.\r94380 markers complete.\r94440 markers complete.\r94500 markers complete.\r94560 markers complete.\r94620 markers complete.\r94680 markers complete.\r94740 markers complete.\r94800 markers complete.\r94860 markers complete.\r94920 markers complete.\r94980 markers complete.\r95040 markers complete.\r95100 markers complete.\r95160 markers complete.\r95220 markers complete.\r95280 markers complete.\r95340 markers complete.\r95400 markers complete.\r95460 markers complete.\r95520 markers complete.\r95580 markers complete.\r95640 markers complete.\r95700 markers complete.\r95760 markers complete.\r95820 markers complete.\r95880 markers complete.\r95940 markers complete.\r96000 markers complete.\r96060 markers complete.\r96120 markers complete.\r96180 markers complete.\r96240 markers complete.\r96300 markers complete.\r96360 markers complete.\r96420 markers complete.\r96480 markers complete.\r96540 markers complete.\r96600 markers complete.\r96660 markers complete.\r96720 markers complete.\r96780 markers complete.\r96840 markers complete.\r96900 markers complete.\r96960 markers complete.\r97020 markers complete.\r97080 markers complete.\r97140 markers complete.\r97200 markers complete.\r97260 markers complete.\r97320 markers complete.\r97380 markers complete.\r97440 markers complete.\r97500 markers complete.\r97560 markers complete.\r97620 markers complete.\r97680 markers complete.\r97740 markers complete.\r97800 markers complete.\r97860 markers complete.\r97920 markers complete.\r97980 markers complete.\r98040 markers complete.\r98100 markers complete.\r98160 markers complete.\r98220 markers complete.\r98280 markers complete.\r98340 markers complete.\r98400 markers complete.\r98460 markers complete.\r98520 markers complete.\r98580 markers complete.\r98640 markers complete.\r98700 markers complete.\r98760 markers complete.\r98820 markers complete.\r98880 markers complete.\r98940 markers complete.\r99000 markers complete.\r99060 markers complete.\r99120 markers complete.\r99180 markers complete.\r99240 markers complete.\r99300 markers complete.\r99360 markers complete.\r99420 markers complete.\r99480 markers complete.\r99540 markers complete.\r99600 markers complete.\r99660 markers complete.\r99720 markers complete.\r99780 markers complete.\r99840 markers complete.\r99900 markers complete.\r99960 markers complete.\r100020 markers complete.\r100080 markers complete.\r100140 markers complete.\r100200 markers complete.\r100260 markers complete.\r100320 markers complete.\r100380 markers complete.\r100440 markers complete.\r100500 markers complete.\r100560 markers complete.\r100620 markers complete.\r100680 markers complete.\r100740 markers complete.\r100800 markers complete.\r100860 markers complete.\r100920 markers complete.\r100980 markers complete.\r101040 markers complete.\r101100 markers complete.\r101160 markers complete.\r101220 markers complete.\r101280 markers complete.\r101340 markers complete.\r101400 markers complete.\r101460 markers complete.\r101520 markers complete.\r101580 markers complete.\r101640 markers complete.\r101700 markers complete.\r101760 markers complete.\r101820 markers complete.\r101880 markers complete.\r101940 markers complete.\r102000 markers complete.\r102060 markers complete.\r102120 markers complete.\r102180 markers complete.\r102240 markers complete.\r102300 markers complete.\r102360 markers complete.\r102420 markers complete.\r102480 markers complete.\r102540 markers complete.\r102600 markers complete.\r102660 markers complete.\r102720 markers complete.\r102780 markers complete.\r102840 markers complete.\r102900 markers complete.\r102960 markers complete.\r103020 markers complete.\r103080 markers complete.\r103140 markers complete.\r103200 markers complete.\r103260 markers complete.\r103320 markers complete.\r103380 markers complete.\r103440 markers complete.\r103500 markers complete.\r103560 markers complete.\r103620 markers complete.\r103680 markers complete.\r103740 markers complete.\r103800 markers complete.\r103860 markers complete.\r103920 markers complete.\r103980 markers complete.\r104040 markers complete.\r104100 markers complete.\r104160 markers complete.\r104220 markers complete.\r104280 markers complete.\r104340 markers complete.\r104400 markers complete.\r104460 markers complete.\r104520 markers complete.\r104580 markers complete.\r104640 markers complete.\r104700 markers complete.\r104760 markers complete.\r104820 markers complete.\r104880 markers complete.\r104940 markers complete.\r105000 markers complete.\r105060 markers complete.\r105120 markers complete.\r105180 markers complete.\r105240 markers complete.\r105300 markers complete.\r105360 markers complete.\r105420 markers complete.\r105480 markers complete.\r105540 markers complete.\r105600 markers complete.\r105660 markers complete.\r105720 markers complete.\r105780 markers complete.\r105840 markers complete.\r105900 markers complete.\r105960 markers complete.\r106020 markers complete.\r106080 markers complete.\r106140 markers complete.\r106200 markers complete.\r106260 markers complete.\r106320 markers complete.\r106380 markers complete.\r106440 markers complete.\r106500 markers complete.\r106560 markers complete.\r106620 markers complete.\r106680 markers complete.\r106740 markers complete.\r106800 markers complete.\r106860 markers complete.\r106920 markers complete.\r106980 markers complete.\r107040 markers complete.\r107100 markers complete.\r107160 markers complete.\r107220 markers complete.\r107280 markers complete.\r107340 markers complete.\r107400 markers complete.\r107460 markers complete.\r107520 markers complete.\r107580 markers complete.\r107640 markers complete.\r107700 markers complete.\r107760 markers complete.\r107820 markers complete.\r107880 markers complete.\r107940 markers complete.\r108000 markers complete.\r108060 markers complete.\r108120 markers complete.\r108180 markers complete.\r108240 markers complete.\r108300 markers complete.\r108360 markers complete.\r108420 markers complete.\r108480 markers complete.\r108540 markers complete.\r108600 markers complete.\r108660 markers complete.\r108720 markers complete.\r108780 markers complete.\r108840 markers complete.\r108900 markers complete.\r108960 markers complete.\r109020 markers complete.\r109080 markers complete.\r109140 markers complete.\r109200 markers complete.\r109260 markers complete.\r109320 markers complete.\r109380 markers complete.\r109440 markers complete.\r109500 markers complete.\r109560 markers complete.\r109620 markers complete.\r109680 markers complete.\r109740 markers complete.\r109800 markers complete.\r109860 markers complete.\r109920 markers complete.\r109980 markers complete.\r110040 markers complete.\r110100 markers complete.\r110160 markers complete.\r110220 markers complete.\r110280 markers complete.\r110340 markers complete.\r110400 markers complete.\r110460 markers complete.\r110520 markers complete.\r110580 markers complete.\r110640 markers complete.\r110700 markers complete.\r110760 markers complete.\r110820 markers complete.\r110880 markers complete.\r110940 markers complete.\r111000 markers complete.\r111060 markers complete.\r111120 markers complete.\r111180 markers complete.\r111240 markers complete.\r111300 markers complete.\r111360 markers complete.\r111420 markers complete.\r111480 markers complete.\r111540 markers complete.\r111600 markers complete.\r111660 markers complete.\r111720 markers complete.\r111780 markers complete.\r111840 markers complete.\r111900 markers complete.\r111960 markers complete.\r112020 markers complete.\r112080 markers complete.\r112140 markers complete.\r112200 markers complete.\r112260 markers complete.\r112320 markers complete.\r112380 markers complete.\r112440 markers complete.\r112500 markers complete.\r112560 markers complete.\r112620 markers complete.\r112680 markers complete.\r112740 markers complete.\r112800 markers complete.\r112860 markers complete.\r112920 markers complete.\r112980 markers complete.\r113040 markers complete.\r113100 markers complete.\r113160 markers complete.\r113220 markers complete.\r113280 markers complete.\r113340 markers complete.\r113400 markers complete.\r113460 markers complete.\r113520 markers complete.\r113580 markers complete.\r113640 markers complete.\r113700 markers complete.\r113760 markers complete.\r113820 markers complete.\r113880 markers complete.\r113940 markers complete.\r114000 markers complete.\r114060 markers complete.\r114120 markers complete.\r114180 markers complete.\r114240 markers complete.\r114300 markers complete.\r114360 markers complete.\r114420 markers complete.\r114480 markers complete.\r114540 markers complete.\r114600 markers complete.\r114660 markers complete.\r114720 markers complete.\r114780 markers complete.\r114840 markers complete.\r114900 markers complete.\r114960 markers complete.\r115020 markers complete.\r115080 markers complete.\r115140 markers complete.\r115200 markers complete.\r115260 markers complete.\r115320 markers complete.\r115380 markers complete.\r115440 markers complete.\r115500 markers complete.\r115560 markers complete.\r115620 markers complete.\r115680 markers complete.\r115740 markers complete.\r115800 markers complete.\r115860 markers complete.\r115920 markers complete.\r115980 markers complete.\r116040 markers complete.\r116100 markers complete.\r116160 markers complete.\r116220 markers complete.\r116280 markers complete.\r116340 markers complete.\r116400 markers complete.\r116460 markers complete.\r116520 markers complete.\r116580 markers complete.\r116640 markers complete.\r116700 markers complete.\r116760 markers complete.\r116820 markers complete.\r116880 markers complete.\r116940 markers complete.\r117000 markers complete.\r117060 markers complete.\r117120 markers complete.\r117180 markers complete.\r117240 markers complete.\r117300 markers complete.\r117360 markers complete.\r117420 markers complete.\r117480 markers complete.\r117540 markers complete.\r117600 markers complete.\r117660 markers complete.\r117720 markers complete.\r117780 markers complete.\r117840 markers complete.\r117900 markers complete.\r117960 markers complete.\r118020 markers complete.\r118080 markers complete.\r118140 markers complete.\r118200 markers complete.\r118260 markers complete.\r118320 markers complete.\r118380 markers complete.\r118440 markers complete.\r118500 markers complete.\r118560 markers complete.\r118620 markers complete.\r118680 markers complete.\r118740 markers complete.\r118800 markers complete.\r118860 markers complete.\r118920 markers complete.\r118980 markers complete.\r119040 markers complete.\r119100 markers complete.\r119160 markers complete.\r119220 markers complete.\r119280 markers complete.\r119340 markers complete.\r119400 markers complete.\r119460 markers complete.\r119520 markers complete.\r119580 markers complete.\r119640 markers complete.\r119700 markers complete.\r119760 markers complete.\r119820 markers complete.\r119880 markers complete.\r119940 markers complete.\r120000 markers complete.\r120060 markers complete.\r120120 markers complete.\r120180 markers complete.\r120240 markers complete.\r120300 markers complete.\r120360 markers complete.\r120420 markers complete.\r120480 markers complete.\r120540 markers complete.\r120600 markers complete.\r120660 markers complete.\r120720 markers complete.\r120780 markers complete.\r120840 markers complete.\r120900 markers complete.\r120960 markers complete.\r121020 markers complete.\r121080 markers complete.\r121140 markers complete.\r121200 markers complete.\r121260 markers complete.\r121320 markers complete.\r121380 markers complete.\r121440 markers complete.\r121500 markers complete.\r121560 markers complete.\r121620 markers complete.\r121680 markers complete.\r121740 markers complete.\r121800 markers complete.\r121860 markers complete.\r121920 markers complete.\r121980 markers complete.\r122040 markers complete.\r122100 markers complete.\r122160 markers complete.\r122220 markers complete.\r122280 markers complete.\r122340 markers complete.\r122400 markers complete.\r122460 markers complete.\r122520 markers complete.\r122580 markers complete.\r122640 markers complete.\r122700 markers complete.\r122760 markers complete.\r122820 markers complete.\r122880 markers complete.\r122940 markers complete.\r123000 markers complete.\r123060 markers complete.\r123120 markers complete.\r123180 markers complete.\r123240 markers complete.\r123300 markers complete.\r123360 markers complete.\r123420 markers complete.\r123480 markers complete.\r123540 markers complete.\r123600 markers complete.\r123660 markers complete.\r123720 markers complete.\r123780 markers complete.\r123840 markers complete.\r123900 markers complete.\r123960 markers complete.\r124020 markers complete.\r124080 markers complete.\r124140 markers complete.\r124200 markers complete.\r124260 markers complete.\r124320 markers complete.\r124380 markers complete.\r124440 markers complete.\r124500 markers complete.\r124560 markers complete.\r124620 markers complete.\r124680 markers complete.\r124740 markers complete.\r124800 markers complete.\r124860 markers complete.\r124920 markers complete.\r124980 markers complete.\r125040 markers complete.\r125100 markers complete.\r125160 markers complete.\r125220 markers complete.\r125280 markers complete.\r125340 markers complete.\r125400 markers complete.\r125460 markers complete.\r125520 markers complete.\r125580 markers complete.\r125640 markers complete.\r125700 markers complete.\r125760 markers complete.\r125820 markers complete.\r125880 markers complete.\r125940 markers complete.\r126000 markers complete.\r126060 markers complete.\r126120 markers complete.\r126180 markers complete.\r126240 markers complete.\r126300 markers complete.\r126360 markers complete.\r126420 markers complete.\r126480 markers complete.\r126540 markers complete.\r126600 markers complete.\r126660 markers complete.\r126720 markers complete.\r126780 markers complete.\r126840 markers complete.\r126900 markers complete.\r126960 markers complete.\r127020 markers complete.\r127080 markers complete.\r127140 markers complete.\r127200 markers complete.\r127260 markers complete.\r127320 markers complete.\r127380 markers complete.\r127440 markers complete.\r127500 markers complete.\r127560 markers complete.\r127620 markers complete.\r127680 markers complete.\r127740 markers complete.\r127800 markers complete.\r127860 markers complete.\r127920 markers complete.\r127980 markers complete.\r128040 markers complete.\r128100 markers complete.\r128160 markers complete.\r128220 markers complete.\r128280 markers complete.\r128340 markers complete.\r128400 markers complete.\r128460 markers complete.\r128520 markers complete.\r128580 markers complete.\r128640 markers complete.\r128700 markers complete.\r128760 markers complete.\r128820 markers complete.\r128880 markers complete.\r128940 markers complete.\r129000 markers complete.\r129060 markers complete.\r129120 markers complete.\r129180 markers complete.\r129240 markers complete.\r129300 markers complete.\r129360 markers complete.\r129420 markers complete.\r129480 markers complete.\r129540 markers complete.\r129600 markers complete.\r129660 markers complete.\r129720 markers complete.\r129780 markers complete.\r129840 markers complete.\r129900 markers complete.\r129960 markers complete.\r130020 markers complete.\r130080 markers complete.\r130140 markers complete.\r130200 markers complete.\r130260 markers complete.\r130320 markers complete.\r130380 markers complete.\r130440 markers complete.\r130500 markers complete.\r130560 markers complete.\r130620 markers complete.\r130680 markers complete.\r130740 markers complete.\r130800 markers complete.\r130860 markers complete.\r130920 markers complete.\r130980 markers complete.\r131040 markers complete.\r131100 markers complete.\r131160 markers complete.\r131220 markers complete.\r131280 markers complete.\r131340 markers complete.\r131400 markers complete.\r131460 markers complete.\r131520 markers complete.\r131580 markers complete.\r131640 markers complete.\r131700 markers complete.\r131760 markers complete.\r131820 markers complete.\r131880 markers complete.\r131940 markers complete.\r132000 markers complete.\r132060 markers complete.\r132120 markers complete.\r132180 markers complete.\r132240 markers complete.\r132300 markers complete.\r132360 markers complete.\r132420 markers complete.\r132480 markers complete.\r132540 markers complete.\r132600 markers complete.\r132660 markers complete.\r132720 markers complete.\r132780 markers complete.\r132840 markers complete.\r132900 markers complete.\r132960 markers complete.\r133020 markers complete.\r133080 markers complete.\r133140 markers complete.\r133200 markers complete.\r133260 markers complete.\r133320 markers complete.\r133380 markers complete.\r133440 markers complete.\r133500 markers complete.\r133560 markers complete.\r133620 markers complete.\r133680 markers complete.\r133740 markers complete.\r133800 markers complete.\r133860 markers complete.\r133920 markers complete.\r133980 markers complete.\r134040 markers complete.\r134100 markers complete.\r134160 markers complete.\r134220 markers complete.\r134280 markers complete.\r134340 markers complete.\r134400 markers complete.\r134460 markers complete.\r134520 markers complete.\r134580 markers complete.\r134640 markers complete.\r134700 markers complete.\r134760 markers complete.\r134820 markers complete.\r134880 markers complete.\r134940 markers complete.\r135000 markers complete.\r135060 markers complete.\r135120 markers complete.\r135180 markers complete.\r135240 markers complete.\r135300 markers complete.\r135360 markers complete.\r135420 markers complete.\r135480 markers complete.\r135540 markers complete.\r135600 markers complete.\r135660 markers complete.\r135720 markers complete.\r135780 markers complete.\r135840 markers complete.\r135900 markers complete.\r135960 markers complete.\r136020 markers complete.\r136080 markers complete.\r136140 markers complete.\r136200 markers complete.\r136260 markers complete.\r136320 markers complete.\r136380 markers complete.\r136440 markers complete.\r136500 markers complete.\r136560 markers complete.\r136620 markers complete.\r136680 markers complete.\r136740 markers complete.\r136800 markers complete.\r136860 markers complete.\r136920 markers complete.\r136980 markers complete.\r137040 markers complete.\r137100 markers complete.\r137160 markers complete.\r137220 markers complete.\r137280 markers complete.\r137340 markers complete.\r137400 markers complete.\r137460 markers complete.\r137520 markers complete.\r137580 markers complete.\r137640 markers complete.\r137700 markers complete.\r137760 markers complete.\r137820 markers complete.\r137880 markers complete.\r137940 markers complete.\r138000 markers complete.\r138060 markers complete.\r138120 markers complete.\r138180 markers complete.\r138240 markers complete.\r138300 markers complete.\r138360 markers complete.\r138420 markers complete.\r138480 markers complete.\r138540 markers complete.\r138600 markers complete.\r138660 markers complete.\r138720 markers complete.\r138780 markers complete.\r138840 markers complete.\r138900 markers complete.\r138960 markers complete.\r139020 markers complete.\r139080 markers complete.\r139140 markers complete.\r139200 markers complete.\r139260 markers complete.\r139320 markers complete.\r139380 markers complete.\r139440 markers complete.\r139500 markers complete.\r139560 markers complete.\r139620 markers complete.\r139680 markers complete.\r139740 markers complete.\r139800 markers complete.\r139860 markers complete.\r139920 markers complete.\r139980 markers complete.\r140040 markers complete.\r140100 markers complete.\r140160 markers complete.\r140220 markers complete.\r140280 markers complete.\r140340 markers complete.\r140400 markers complete.\r140460 markers complete.\r140520 markers complete.\r140580 markers complete.\r140640 markers complete.\r140700 markers complete.\r140760 markers complete.\r140820 markers complete.\r140880 markers complete.\r140940 markers complete.\r141000 markers complete.\r141060 markers complete.\r141120 markers complete.\r141180 markers complete.\r141240 markers complete.\r141300 markers complete.\r141360 markers complete.\r141420 markers complete.\r141480 markers complete.\r141540 markers complete.\r141600 markers complete.\r141660 markers complete.\r141720 markers complete.\r141780 markers complete.\r141840 markers complete.\r141900 markers complete.\r141960 markers complete.\r142020 markers complete.\r142080 markers complete.\r142140 markers complete.\r142200 markers complete.\r142260 markers complete.\r142320 markers complete.\r142380 markers complete.\r142440 markers complete.\r142500 markers complete.\r142560 markers complete.\r142620 markers complete.\r142680 markers complete.\r142740 markers complete.\r142800 markers complete.\r142860 markers complete.\r142920 markers complete.\r142980 markers complete.\r143040 markers complete.\r143100 markers complete.\r143160 markers complete.\r143220 markers complete.\r143280 markers complete.\r143340 markers complete.\r143400 markers complete.\r143460 markers complete.\r143520 markers complete.\r143580 markers complete.\r143640 markers complete.\r143700 markers complete.\r143760 markers complete.\r143820 markers complete.\r143880 markers complete.\r143940 markers complete.\r144000 markers complete.\r144060 markers complete.\r144120 markers complete.\r144180 markers complete.\r144240 markers complete.\r144300 markers complete.\r144360 markers complete.\r144420 markers complete.\r144480 markers complete.\r144540 markers complete.\r144600 markers complete.\r144660 markers complete.\r144720 markers complete.\r144780 markers complete.\r144840 markers complete.\r144900 markers complete.\r144960 markers complete.\r145020 markers complete.\r145080 markers complete.\r145140 markers complete.\r145200 markers complete.\r145260 markers complete.\r145320 markers complete.\r145380 markers complete.\r145440 markers complete.\r145500 markers complete.\r145560 markers complete.\r145620 markers complete.\r145680 markers complete.\r145740 markers complete.\r145800 markers complete.\r145860 markers complete.\r145920 markers complete.\r145980 markers complete.\r146040 markers complete.\r146100 markers complete.\r146160 markers complete.\r146220 markers complete.\r146280 markers complete.\r146340 markers complete.\r146400 markers complete.\r146460 markers complete.\r146520 markers complete.\r146580 markers complete.\r146640 markers complete.\r146700 markers complete.\r146760 markers complete.\r146820 markers complete.\r146880 markers complete.\r146940 markers complete.\r147000 markers complete.\r147060 markers complete.\r147120 markers complete.\r147180 markers complete.\r147240 markers complete.\r147300 markers complete.\r147360 markers complete.\r147420 markers complete.\r147480 markers complete.\r147540 markers complete.\r147600 markers complete.\r147660 markers complete.\r147720 markers complete.\r147780 markers complete.\r147840 markers complete.\r147900 markers complete.\r147960 markers complete.\r148020 markers complete.\r148080 markers complete.\r148140 markers complete.\r148200 markers complete.\r148260 markers complete.\r148320 markers complete.\r148380 markers complete.\r148440 markers complete.\r148500 markers complete.\r148560 markers complete.\r148620 markers complete.\r148680 markers complete.\r148740 markers complete.\r148800 markers complete.\r148860 markers complete.\r148920 markers complete.\r148980 markers complete.\r149040 markers complete.\r149100 markers complete.\r149160 markers complete.\r149220 markers complete.\r149280 markers complete.\r149340 markers complete.\r149400 markers complete.\r149460 markers complete.\r149520 markers complete.\r149580 markers complete.\r149640 markers complete.\r149700 markers complete.\r149760 markers complete.\r149820 markers complete.\r149880 markers complete.\r149940 markers complete.\r150000 markers complete.\r150060 markers complete.\r150120 markers complete.\r150180 markers complete.\r150240 markers complete.\r150300 markers complete.\r150360 markers complete.\r150420 markers complete.\r150480 markers complete.\r150540 markers complete.\r150600 markers complete.\r150660 markers complete.\r150720 markers complete.\r150780 markers complete.\r150840 markers complete.\r150900 markers complete.\r150960 markers complete.\r151020 markers complete.\r151080 markers complete.\r151140 markers complete.\r151200 markers complete.\r151260 markers complete.\r151320 markers complete.\r151380 markers complete.\r151440 markers complete.\r151500 markers complete.\r151560 markers complete.\r151620 markers complete.\r151680 markers complete.\r151740 markers complete.\r151800 markers complete.\r151860 markers complete.\r151920 markers complete.\r151980 markers complete.\r152040 markers complete.\r152100 markers complete.\r152160 markers complete.\r152220 markers complete.\r152280 markers complete.\r152340 markers complete.\r152400 markers complete.\r152460 markers complete.\r152520 markers complete.\r152580 markers complete.\r152640 markers complete.\r152700 markers complete.\r152760 markers complete.\r152820 markers complete.\r152880 markers complete.\r152940 markers complete.\r153000 markers complete.\r153060 markers complete.\r153120 markers complete.\r153180 markers complete.\r153240 markers complete.\r153300 markers complete.\r153360 markers complete.\r153420 markers complete.\r153480 markers complete.\r153540 markers complete.\r153600 markers complete.\r153660 markers complete.\r153720 markers complete.\r153780 markers complete.\r153840 markers complete.\r153900 markers complete.\r153960 markers complete.\r154020 markers complete.\r154080 markers complete.\r154140 markers complete.\r154200 markers complete.\r154260 markers complete.\r154320 markers complete.\r154380 markers complete.\r154440 markers complete.\r154500 markers complete.\r154560 markers complete.\r154620 markers complete.\r154680 markers complete.\r154740 markers complete.\r154800 markers complete.\r154860 markers complete.\r154920 markers complete.\r154980 markers complete.\r155040 markers complete.\r155100 markers complete.\r155160 markers complete.\r155220 markers complete.\r155280 markers complete.\r155340 markers complete.\r155400 markers complete.\r155460 markers complete.\r155520 markers complete.\r155580 markers complete.\r155640 markers complete.\r155700 markers complete.\r155760 markers complete.\r155820 markers complete.\r155880 markers complete.\r155940 markers complete.\r156000 markers complete.\r156060 markers complete.\r156120 markers complete.\r156180 markers complete.\r156240 markers complete.\r156300 markers complete.\r156360 markers complete.\r156420 markers complete.\r156480 markers complete.\r156540 markers complete.\r156600 markers complete.\r156660 markers complete.\r156720 markers complete.\r156780 markers complete.\r156840 markers complete.\r156900 markers complete.\r156960 markers complete.\r157020 markers complete.\r157080 markers complete.\r157140 markers complete.\r157200 markers complete.\r157260 markers complete.\r157320 markers complete.\r157380 markers complete.\r157440 markers complete.\r157500 markers complete.\r157560 markers complete.\r157620 markers complete.\r157680 markers complete.\r157740 markers complete.\r157800 markers complete.\r157860 markers complete.\r157920 markers complete.\r157980 markers complete.\r158040 markers complete.\r158100 markers complete.\r158160 markers complete.\r158220 markers complete.\r158280 markers complete.\r158340 markers complete.\r158400 markers complete.\r158460 markers complete.\r158520 markers complete.\r158580 markers complete.\r158640 markers complete.\r158700 markers complete.\r158760 markers complete.\r158820 markers complete.\r158880 markers complete.\r158940 markers complete.\r159000 markers complete.\r159060 markers complete.\r159120 markers complete.\r159180 markers complete.\r159240 markers complete.\r159300 markers complete.\r159360 markers complete.\r159420 markers complete.\r159480 markers complete.\r159540 markers complete.\r159600 markers complete.\r159660 markers complete.\r159720 markers complete.\r159780 markers complete.\r159840 markers complete.\r159900 markers complete.\r159960 markers complete.\r160020 markers complete.\r160080 markers complete.\r160140 markers complete.\r160200 markers complete.\r160260 markers complete.\r160320 markers complete.\r160380 markers complete.\r160440 markers complete.\r160500 markers complete.\r160560 markers complete.\r160620 markers complete.\r160680 markers complete.\r160740 markers complete.\r160800 markers complete.\r160860 markers complete.\r160920 markers complete.\r160980 markers complete.\r161040 markers complete.\r161100 markers complete.\r161160 markers complete.\r161220 markers complete.\r161280 markers complete.\r161340 markers complete.\r161400 markers complete.\r161460 markers complete.\r161520 markers complete.\r161580 markers complete.\r161640 markers complete.\r161700 markers complete.\r161760 markers complete.\r161820 markers complete.\r161880 markers complete.\r161940 markers complete.\r162000 markers complete.\r162060 markers complete.\r162120 markers complete.\r162180 markers complete.\r162240 markers complete.\r162300 markers complete.\r162360 markers complete.\r162420 markers complete.\r162480 markers complete.\r162540 markers complete.\r162600 markers complete.\r162660 markers complete.\r162720 markers complete.\r162780 markers complete.\r162840 markers complete.\r162900 markers complete.\r162960 markers complete.\r163020 markers complete.\r163080 markers complete.\r163140 markers complete.\r163200 markers complete.\r163260 markers complete.\r163320 markers complete.\r163380 markers complete.\r163440 markers complete.\r163500 markers complete.\r163560 markers complete.\r163620 markers complete.\r163680 markers complete.\r163740 markers complete.\r163800 markers complete.\r163860 markers complete.\r163920 markers complete.\r163980 markers complete.\r164040 markers complete.\r164100 markers complete.\r164160 markers complete.\r164220 markers complete.\r164280 markers complete.\r164340 markers complete.\r164400 markers complete.\r164460 markers complete.\r164520 markers complete.\r164580 markers complete.\r164640 markers complete.\r164700 markers complete.\r164760 markers complete.\r164820 markers complete.\r164880 markers complete.\r164940 markers complete.\r165000 markers complete.\r165060 markers complete.\r165120 markers complete.\r165180 markers complete.\r165240 markers complete.\r165300 markers complete.\r165360 markers complete.\r165420 markers complete.\r165480 markers complete.\r165540 markers complete.\r165600 markers complete.\r165660 markers complete.\r165720 markers complete.\r165780 markers complete.\r165840 markers complete.\r165900 markers complete.\r165960 markers complete.\r166020 markers complete.\r166080 markers complete.\r166140 markers complete.\r166200 markers complete.\r166260 markers complete.\r166320 markers complete.\r166380 markers complete.\r166440 markers complete.\r166500 markers complete.\r166560 markers complete.\r166620 markers complete.\r166680 markers complete.\r166740 markers complete.\r166800 markers complete.\r166860 markers complete.\r166920 markers complete.\r166980 markers complete.\r167040 markers complete.\r167100 markers complete.\r167160 markers complete.\r167220 markers complete.\r167280 markers complete.\r167340 markers complete.\r167400 markers complete.\r167460 markers complete.\r167520 markers complete.\r167580 markers complete.\r167640 markers complete.\r167700 markers complete.\r167760 markers complete.\r167820 markers complete.\r167880 markers complete.\r167940 markers complete.\r168000 markers complete.\r168060 markers complete.\r168120 markers complete.\r168180 markers complete.\r168240 markers complete.\r168300 markers complete.\r168360 markers complete.\r168420 markers complete.\r168480 markers complete.\r168540 markers complete.\r168600 markers complete.\r168660 markers complete.\r168720 markers complete.\r168780 markers complete.\r168840 markers complete.\r168900 markers complete.\r168960 markers complete.\r169020 markers complete.\r169080 markers complete.\r169140 markers complete.\r169200 markers complete.\r169260 markers complete.\r169320 markers complete.\r169380 markers complete.\r169440 markers complete.\r169500 markers complete.\r169560 markers complete.\r169620 markers complete.\r169680 markers complete.\r169740 markers complete.\r169800 markers complete.\r169860 markers complete.\r169920 markers complete.\r169980 markers complete.\r170040 markers complete.\r170100 markers complete.\r170160 markers complete.\r170220 markers complete.\r170280 markers complete.\r170340 markers complete.\r170400 markers complete.\r170460 markers complete.\r170520 markers complete.\r170580 markers complete.\r170640 markers complete.\r170700 markers complete.\r170760 markers complete.\r170820 markers complete.\r170880 markers complete.\r170940 markers complete.\r171000 markers complete.\r171060 markers complete.\r171120 markers complete.\r171180 markers complete.\r171240 markers complete.\r171300 markers complete.\r171360 markers complete.\r171420 markers complete.\r171480 markers complete.\r171540 markers complete.\r171600 markers complete.\r171660 markers complete.\r171720 markers complete.\r171780 markers complete.\r171840 markers complete.\r171900 markers complete.\r171960 markers complete.\r172020 markers complete.\r172080 markers complete.\r172140 markers complete.\r172200 markers complete.\r172260 markers complete.\r172320 markers complete.\r172380 markers complete.\r172440 markers complete.\r172500 markers complete.\r172560 markers complete.\r172620 markers complete.\r172680 markers complete.\r172740 markers complete.\r172800 markers complete.\r172860 markers complete.\r172920 markers complete.\r172980 markers complete.\r173040 markers complete.\r173100 markers complete.\r173160 markers complete.\r173220 markers complete.\r173280 markers complete.\r173340 markers complete.\r173400 markers complete.\r173460 markers complete.\r173520 markers complete.\r173580 markers complete.\r173640 markers complete.\r173700 markers complete.\r173760 markers complete.\r173820 markers complete.\r173880 markers complete.\r173940 markers complete.\r174000 markers complete.\r174060 markers complete.\r174120 markers complete.\r174180 markers complete.\r174240 markers complete.\r174300 markers complete.\r174360 markers complete.\r174420 markers complete.\r174480 markers complete.\r174540 markers complete.\r174600 markers complete.\r174660 markers complete.\r174720 markers complete.\r174780 markers complete.\r174840 markers complete.\r174900 markers complete.\r174960 markers complete.\r175020 markers complete.\r175080 markers complete.\r175140 markers complete.\r175200 markers complete.\r175260 markers complete.\r175320 markers complete.\r175380 markers complete.\r175440 markers complete.\r175500 markers complete.\r175560 markers complete.\r175620 markers complete.\r175680 markers complete.\r175740 markers complete.\r175800 markers complete.\r175860 markers complete.\r175920 markers complete.\r175980 markers complete.\r176040 markers complete.\r176100 markers complete.\r176160 markers complete.\r176220 markers complete.\r176280 markers complete.\r176340 markers complete.\r176400 markers complete.\r176460 markers complete.\r176520 markers complete.\r176580 markers complete.\r176640 markers complete.\r176700 markers complete.\r176760 markers complete.\r176820 markers complete.\r176880 markers complete.\r176940 markers complete.\r177000 markers complete.\r177060 markers complete.\r177120 markers complete.\r177180 markers complete.\r177240 markers complete.\r177300 markers complete.\r177360 markers complete.\r177420 markers complete.\r177480 markers complete.\r177540 markers complete.\r177600 markers complete.\r177660 markers complete.\r177720 markers complete.\r177780 markers complete.\r177840 markers complete.\r177900 markers complete.\r177960 markers complete.\r178020 markers complete.\r178080 markers complete.\r178140 markers complete.\r178200 markers complete.\r178260 markers complete.\r178320 markers complete.\r178380 markers complete.\r178440 markers complete.\r178500 markers complete.\r178560 markers complete.\r178620 markers complete.\r178680 markers complete.\r178740 markers complete.\r178800 markers complete.\r178860 markers complete.\r178920 markers complete.\r178980 markers complete.\r179040 markers complete.\r179100 markers complete.\r179160 markers complete.\r179220 markers complete.\r179280 markers complete.\r179340 markers complete.\r179400 markers complete.\r179460 markers complete.\r179520 markers complete.\r179580 markers complete.\r179640 markers complete.\r179700 markers complete.\r179760 markers complete.\r179820 markers complete.\r179880 markers complete.\r179940 markers complete.\r180000 markers complete.\r180060 markers complete.\r180120 markers complete.\r180180 markers complete.\r180240 markers complete.\r180300 markers complete.\r180360 markers complete.\r180420 markers complete.\r180480 markers complete.\r180540 markers complete.\r180600 markers complete.\r180660 markers complete.\r180720 markers complete.\r180780 markers complete.\r180840 markers complete.\r180900 markers complete.\r180960 markers complete.\r181020 markers complete.\r181080 markers complete.\r181140 markers complete.\r181200 markers complete.\r181260 markers complete.\r181320 markers complete.\r181380 markers complete.\r181440 markers complete.\r181500 markers complete.\r181560 markers complete.\r181620 markers complete.\r181680 markers complete.\r181740 markers complete.\r181800 markers complete.\r181860 markers complete.\r181920 markers complete.\r181980 markers complete.\r182040 markers complete.\r182100 markers complete.\r182160 markers complete.\r182220 markers complete.\r182280 markers complete.\r182340 markers complete.\r182400 markers complete.\r182460 markers complete.\r182520 markers complete.\r182580 markers complete.\r182640 markers complete.\r182700 markers complete.\r182760 markers complete.\r182820 markers complete.\r182880 markers complete.\r182940 markers complete.\r183000 markers complete.\r183060 markers complete.\r183120 markers complete.\r183180 markers complete.\r183240 markers complete.\r183300 markers complete.\r183360 markers complete.\r183420 markers complete.\r183480 markers complete.\r183540 markers complete.\r183600 markers complete.\r183660 markers complete.\r183720 markers complete.\r183780 markers complete.\r183840 markers complete.\r183900 markers complete.\r183960 markers complete.\r184020 markers complete.\r184080 markers complete.\r184140 markers complete.\r184200 markers complete.\r184260 markers complete.\r184320 markers complete.\r184380 markers complete.\r184440 markers complete.\r184500 markers complete.\r184560 markers complete.\r184620 markers complete.\r184680 markers complete.\r184740 markers complete.\r184800 markers complete.\r184860 markers complete.\r184920 markers complete.\r184980 markers complete.\r185040 markers complete.\r185100 markers complete.\r185160 markers complete.\r185220 markers complete.\r185280 markers complete.\r185340 markers complete.\r185400 markers complete.\r185460 markers complete.\r185520 markers complete.\r185580 markers complete.\r185640 markers complete.\r185700 markers complete.\r185760 markers complete.\r185820 markers complete.\r185880 markers complete.\r185940 markers complete.\r186000 markers complete.\r186060 markers complete.\r186120 markers complete.\r186180 markers complete.\r186240 markers complete.\r186300 markers complete.\r186360 markers complete.\r186420 markers complete.\r186480 markers complete.\r186540 markers complete.\r186600 markers complete.\r186660 markers complete.\r186720 markers complete.\r186780 markers complete.\r186840 markers complete.\r186900 markers complete.\r186960 markers complete.\r187020 markers complete.\r187080 markers complete.\r187140 markers complete.\r187200 markers complete.\r187260 markers complete.\r187320 markers complete.\r187380 markers complete.\r187440 markers complete.\r187500 markers complete.\r187560 markers complete.\r187620 markers complete.\r187680 markers complete.\r187740 markers complete.\r187800 markers complete.\r187860 markers complete.\r187920 markers complete.\r187980 markers complete.\r188040 markers complete.\r188100 markers complete.\r188160 markers complete.\r188220 markers complete.\r188280 markers complete.\r188340 markers complete.\r188400 markers complete.\r188460 markers complete.\r188520 markers complete.\r188580 markers complete.\r188640 markers complete.\r188700 markers complete.\r188760 markers complete.\r188820 markers complete.\r188880 markers complete.\r188940 markers complete.\r189000 markers complete.\r189060 markers complete.\r189120 markers complete.\r189180 markers complete.\r189240 markers complete.\r189300 markers complete.\r189360 markers complete.\r189420 markers complete.\r189480 markers complete.\r189540 markers complete.\r189600 markers complete.\r189660 markers complete.\r189720 markers complete.\r189780 markers complete.\r189840 markers complete.\r189900 markers complete.\r189960 markers complete.\r190020 markers complete.\r190080 markers complete.\r190140 markers complete.\r190200 markers complete.\r190260 markers complete.\r190320 markers complete.\r190380 markers complete.\r190440 markers complete.\r190500 markers complete.\r190560 markers complete.\r190620 markers complete.\r190680 markers complete.\r190740 markers complete.\r190800 markers complete.\r190860 markers complete.\r190920 markers complete.\r190980 markers complete.\r191040 markers complete.\r191100 markers complete.\r191160 markers complete.\r191220 markers complete.\r191280 markers complete.\r191340 markers complete.\r191400 markers complete.\r191460 markers complete.\r191520 markers complete.\r191580 markers complete.\r191640 markers complete.\r191700 markers complete.\r191760 markers complete.\r191820 markers complete.\r191880 markers complete.\r191940 markers complete.\r192000 markers complete.\r192060 markers complete.\r192120 markers complete.\r192180 markers complete.\r192240 markers complete.\r192300 markers complete.\r192360 markers complete.\r192420 markers complete.\r192480 markers complete.\r192540 markers complete.\r192600 markers complete.\r192660 markers complete.\r192720 markers complete.\r192780 markers complete.\r192840 markers complete.\r192900 markers complete.\r192960 markers complete.\r193020 markers complete.\r193080 markers complete.\r193140 markers complete.\r193200 markers complete.\r193260 markers complete.\r193320 markers complete.\r193380 markers complete.\r193440 markers complete.\r193500 markers complete.\r193560 markers complete.\r193620 markers complete.\r193680 markers complete.\r193740 markers complete.\r193800 markers complete.\r193860 markers complete.\r193920 markers complete.\r193980 markers complete.\r194040 markers complete.\r194100 markers complete.\r194160 markers complete.\r194220 markers complete.\r194280 markers complete.\r194340 markers complete.\r194400 markers complete.\r194460 markers complete.\r194520 markers complete.\r194580 markers complete.\r194640 markers complete.\r194700 markers complete.\r194760 markers complete.\r194820 markers complete.\r194880 markers complete.\r194940 markers complete.\r195000 markers complete.\r195060 markers complete.\r195120 markers complete.\r195180 markers complete.\r195240 markers complete.\r195300 markers complete.\r195360 markers complete.\r195420 markers complete.\r195480 markers complete.\r195540 markers complete.\r195600 markers complete.\r195660 markers complete.\r195720 markers complete.\r195780 markers complete.\r195840 markers complete.\r195900 markers complete.\r195960 markers complete.\r196020 markers complete.\r196080 markers complete.\r196140 markers complete.\r196200 markers complete.\r196260 markers complete.\r196320 markers complete.\r196380 markers complete.\r196440 markers complete.\r196500 markers complete.\r196560 markers complete.\r196620 markers complete.\r196680 markers complete.\r196740 markers complete.\r196800 markers complete.\r196860 markers complete.\r196920 markers complete.\r196980 markers complete.\r197040 markers complete.\r197100 markers complete.\r197160 markers complete.\r197220 markers complete.\r197280 markers complete.\r197340 markers complete.\r197400 markers complete.\r197460 markers complete.\r197520 markers complete.\r197580 markers complete.\r197640 markers complete.\r197700 markers complete.\r197760 markers complete.\r197820 markers complete.\r197880 markers complete.\r197940 markers complete.\r198000 markers complete.\r198060 markers complete.\r198120 markers complete.\r198180 markers complete.\r198240 markers complete.\r198300 markers complete.\r198360 markers complete.\r198420 markers complete.\r198480 markers complete.\r198540 markers complete.\r198600 markers complete.\r198660 markers complete.\r198720 markers complete.\r198780 markers complete.\r198840 markers complete.\r198900 markers complete.\r198960 markers complete.\r199020 markers complete.\r199080 markers complete.\r199140 markers complete.\r199200 markers complete.\r199260 markers complete.\r199320 markers complete.\r199380 markers complete.\r199440 markers complete.\r199500 markers complete.\r199560 markers complete.\r199620 markers complete.\r199680 markers complete.\r199740 markers complete.\r199800 markers complete.\r199860 markers complete.\r199920 markers complete.\r199980 markers complete.\r200040 markers complete.\r200100 markers complete.\r200160 markers complete.\r200220 markers complete.\r200280 markers complete.\r200340 markers complete.\r200400 markers complete.\r200460 markers complete.\r200520 markers complete.\r200580 markers complete.\r200640 markers complete.\r200700 markers complete.\r200760 markers complete.\r200820 markers complete.\r200880 markers complete.\r200940 markers complete.\r201000 markers complete.\r201060 markers complete.\r201120 markers complete.\r201180 markers complete.\r201240 markers complete.\r201300 markers complete.\r201360 markers complete.\r201420 markers complete.\r201480 markers complete.\r201540 markers complete.\r201600 markers complete.\r201660 markers complete.\r201720 markers complete.\r201780 markers complete.\r201840 markers complete.\r201900 markers complete.\r201960 markers complete.\r202020 markers complete.\r202080 markers complete.\r202140 markers complete.\r202200 markers complete.\r202260 markers complete.\r202320 markers complete.\r202380 markers complete.\r202440 markers complete.\r202500 markers complete.\r202560 markers complete.\r202620 markers complete.\r202680 markers complete.\r202740 markers complete.\r202800 markers complete.\r202860 markers complete.\r202920 markers complete.\r202980 markers complete.\r203040 markers complete.\r203100 markers complete.\r203160 markers complete.\r203220 markers complete.\r203280 markers complete.\r203340 markers complete.\r203400 markers complete.\r203460 markers complete.\r203520 markers complete.\r203580 markers complete.\r203640 markers complete.\r203700 markers complete.\r203760 markers complete.\r203820 markers complete.\r203880 markers complete.\r203940 markers complete.\r204000 markers complete.\r204060 markers complete.\r204120 markers complete.\r204180 markers complete.\r204240 markers complete.\r204300 markers complete.\r204360 markers complete.\r204420 markers complete.\r204480 markers complete.\r204540 markers complete.\r204600 markers complete.\r204660 markers complete.\r204720 markers complete.\r204780 markers complete.\r204840 markers complete.\r204900 markers complete.\r204960 markers complete.\r205020 markers complete.\r205080 markers complete.\r205140 markers complete.\r205200 markers complete.\r205260 markers complete.\r205320 markers complete.\r205380 markers complete.\r205440 markers complete.\r205500 markers complete.\r205560 markers complete.\r205620 markers complete.\r205680 markers complete.\r205740 markers complete.\r205800 markers complete.\r205860 markers complete.\r205920 markers complete.\r205980 markers complete.\r206040 markers complete.\r206100 markers complete.\r206160 markers complete.\r206220 markers complete.\r206280 markers complete.\r206340 markers complete.\r206400 markers complete.\r206460 markers complete.\r206520 markers complete.\r206580 markers complete.\r206640 markers complete.\r206700 markers complete.\r206760 markers complete.\r206820 markers complete.\r206880 markers complete.\r206940 markers complete.\r207000 markers complete.\r207060 markers complete.\r207120 markers complete.\r207180 markers complete.\r207240 markers complete.\r207300 markers complete.\r207360 markers complete.\r207420 markers complete.\r207480 markers complete.\r207540 markers complete.\r207600 markers complete.\r207660 markers complete.\r207720 markers complete.\r207780 markers complete.\r207840 markers complete.\r207900 markers complete.\r207960 markers complete.\r208020 markers complete.\r208080 markers complete.\r208140 markers complete.\r208200 markers complete.\r208260 markers complete.\r208320 markers complete.\r208380 markers complete.\r208440 markers complete.\r208500 markers complete.\r208560 markers complete.\r208620 markers complete.\r208680 markers complete.\r208740 markers complete.\r208800 markers complete.\r208860 markers complete.\r208920 markers complete.\r208980 markers complete.\r209040 markers complete.\r209100 markers complete.\r209160 markers complete.\r209220 markers complete.\r209280 markers complete.\r209340 markers complete.\r209400 markers complete.\r209460 markers complete.\r209520 markers complete.\r209580 markers complete.\r209640 markers complete.\r209700 markers complete.\r209760 markers complete.\r209820 markers complete.\r209880 markers complete.\r209940 markers complete.\r210000 markers complete.\r210060 markers complete.\r210120 markers complete.\r210180 markers complete.\r210240 markers complete.\r210300 markers complete.\r210360 markers complete.\r210420 markers complete.\r210480 markers complete.\r210540 markers complete.\r210600 markers complete.\r210660 markers complete.\r210720 markers complete.\r210780 markers complete.\r210840 markers complete.\r210900 markers complete.\r210960 markers complete.\r211020 markers complete.\r211080 markers complete.\r211140 markers complete.\r211200 markers complete.\r211260 markers complete.\r211320 markers complete.\r211380 markers complete.\r211440 markers complete.\r211500 markers complete.\r211560 markers complete.\r211620 markers complete.\r211680 markers complete.\r211740 markers complete.\r211800 markers complete.\r211860 markers complete.\r211920 markers complete.\r211980 markers complete.\r212040 markers complete.\r212100 markers complete.\r212160 markers complete.\r212220 markers complete.\r212280 markers complete.\r212340 markers complete.\r212400 markers complete.\r212460 markers complete.\r212520 markers complete.\r212580 markers complete.\r212640 markers complete.\r212700 markers complete.\r212760 markers complete.\r212820 markers complete.\r212880 markers complete.\r212940 markers complete.\r213000 markers complete.\r213060 markers complete.\r213120 markers complete.\r213180 markers complete.\r213240 markers complete.\r213300 markers complete.\r213360 markers complete.\r213420 markers complete.\r213480 markers complete.\r213540 markers complete.\r213600 markers complete.\r213660 markers complete.\r213720 markers complete.\r213780 markers complete.\r213840 markers complete.\r213900 markers complete.\r213960 markers complete.\r214020 markers complete.\r214080 markers complete.\r214140 markers complete.\r214200 markers complete.\r214260 markers complete.\r214320 markers complete.\r214380 markers complete.\r214440 markers complete.\r214500 markers complete.\r214560 markers complete.\r214620 markers complete.\r214680 markers complete.\r214740 markers complete.\r214800 markers complete.\r214860 markers complete.\r214920 markers complete.\r214980 markers complete.\r215040 markers complete.\r215100 markers complete.\r215160 markers complete.\r215220 markers complete.\r215280 markers complete.\r215340 markers complete.\r215400 markers complete.\r215460 markers complete.\r215520 markers complete.\r215580 markers complete.\r215640 markers complete.\r215700 markers complete.\r215760 markers complete.\r215820 markers complete.\r215880 markers complete.\r215940 markers complete.\r216000 markers complete.\r216060 markers complete.\r216120 markers complete.\r216180 markers complete.\r216240 markers complete.\r216300 markers complete.\r216360 markers complete.\r216420 markers complete.\r216480 markers complete.\r216540 markers complete.\r216600 markers complete.\r216660 markers complete.\r216720 markers complete.\r216780 markers complete.\r216840 markers complete.\r216900 markers complete.\r216960 markers complete.\r217020 markers complete.\r217080 markers complete.\r217140 markers complete.\r217200 markers complete.\r217260 markers complete.\r217320 markers complete.\r217380 markers complete.\r217440 markers complete.\r217500 markers complete.\r217560 markers complete.\r217620 markers complete.\r217680 markers complete.\r217740 markers complete.\r217800 markers complete.\r217860 markers complete.\r217920 markers complete.\r217980 markers complete.\r218040 markers complete.\r218100 markers complete.\r218160 markers complete.\r218220 markers complete.\r218280 markers complete.\r218340 markers complete.\r218400 markers complete.\r218460 markers complete.\r218520 markers complete.\r218580 markers complete.\r218640 markers complete.\r218700 markers complete.\r218760 markers complete.\r218820 markers complete.\r218880 markers complete.\r218940 markers complete.\r219000 markers complete.\r219060 markers complete.\r219120 markers complete.\r219180 markers complete.\r219240 markers complete.\r219300 markers complete.\r219360 markers complete.\r219420 markers complete.\r219480 markers complete.\r219540 markers complete.\r219600 markers complete.\r219660 markers complete.\r219720 markers complete.\r219780 markers complete.\r219840 markers complete.\r219900 markers complete.\r219960 markers complete.\r220020 markers complete.\r220080 markers complete.\r220140 markers complete.\r220200 markers complete.\r220260 markers complete.\r220320 markers complete.\r220380 markers complete.\r220440 markers complete.\r220500 markers complete.\r220560 markers complete.\r220620 markers complete.\r220680 markers complete.\r220740 markers complete.\r220800 markers complete.\r220860 markers complete.\r220920 markers complete.\r220980 markers complete.\r221040 markers complete.\r221100 markers complete.\r221160 markers complete.\r221220 markers complete.\r221280 markers complete.\r221340 markers complete.\r221400 markers complete.\r221460 markers complete.\r221520 markers complete.\r221580 markers complete.\r221640 markers complete.\r221700 markers complete.\r221760 markers complete.\r221820 markers complete.\r221880 markers complete.\r221940 markers complete.\r222000 markers complete.\r222060 markers complete.\r222120 markers complete.\r222180 markers complete.\r222240 markers complete.\r222300 markers complete.\r222360 markers complete.\r222420 markers complete.\r222480 markers complete.\r222540 markers complete.\r222600 markers complete.\r222660 markers complete.\r222720 markers complete.\r222780 markers complete.\r222840 markers complete.\r222900 markers complete.\r222960 markers complete.\r223020 markers complete.\r223080 markers complete.\r223140 markers complete.\r223200 markers complete.\r223260 markers complete.\r223320 markers complete.\r223380 markers complete.\r223440 markers complete.\r223500 markers complete.\r223560 markers complete.\r223620 markers complete.\r223680 markers complete.\r223740 markers complete.\r223800 markers complete.\r223860 markers complete.\r223920 markers complete.\r223980 markers complete.\r224040 markers complete.\r224100 markers complete.\r224160 markers complete.\r224220 markers complete.\r224280 markers complete.\r224340 markers complete.\r224400 markers complete.\r224460 markers complete.\r224520 markers complete.\r224580 markers complete.\r224640 markers complete.\r224700 markers complete.\r224760 markers complete.\r224820 markers complete.\r224880 markers complete.\r224940 markers complete.\r225000 markers complete.\r225060 markers complete.\r225120 markers complete.\r225180 markers complete.\r225240 markers complete.\r225300 markers complete.\r225360 markers complete.\r225420 markers complete.\r225480 markers complete.\r225540 markers complete.\r225600 markers complete.\r225660 markers complete.\r225720 markers complete.\r225780 markers complete.\r225840 markers complete.\r225900 markers complete.\r225960 markers complete.\r226020 markers complete.\r226080 markers complete.\r226140 markers complete.\r226200 markers complete.\r226260 markers complete.\r226320 markers complete.\r226380 markers complete.\r226440 markers complete.\r226500 markers complete.\r226560 markers complete.\r226620 markers complete.\r226680 markers complete.\r226740 markers complete.\r226800 markers complete.\r226860 markers complete.\r226920 markers complete.\r226980 markers complete.\r227040 markers complete.\r227100 markers complete.\r227160 markers complete.\r227220 markers complete.\r227280 markers complete.\r227340 markers complete.\r227400 markers complete.\r227460 markers complete.\r227520 markers complete.\r227580 markers complete.\r227640 markers complete.\r227700 markers complete.\r227760 markers complete.\r227820 markers complete.\r227880 markers complete.\r227940 markers complete.\r228000 markers complete.\r228060 markers complete.\r228120 markers complete.\r228180 markers complete.\r228240 markers complete.\r228300 markers complete.\r228360 markers complete.\r228420 markers complete.\r228480 markers complete.\r228540 markers complete.\r228600 markers complete.\r228660 markers complete.\r228720 markers complete.\r228780 markers complete.\r228840 markers complete.\r228900 markers complete.\r228960 markers complete.\r229020 markers complete.\r229080 markers complete.\r229140 markers complete.\r229200 markers complete.\r229260 markers complete.\r229320 markers complete.\r229380 markers complete.\r229440 markers complete.\r229500 markers complete.\r229560 markers complete.\r229620 markers complete.\r229680 markers complete.\r229740 markers complete.\r229800 markers complete.\r229860 markers complete.\r229920 markers complete.\r229980 markers complete.\r230040 markers complete.\r230100 markers complete.\r230160 markers complete.\r230220 markers complete.\r230280 markers complete.\r230340 markers complete.\r230400 markers complete.\r230460 markers complete.\r230520 markers complete.\r230580 markers complete.\r230640 markers complete.\r230700 markers complete.\r230760 markers complete.\r230820 markers complete.\r230880 markers complete.\r230940 markers complete.\r231000 markers complete.\r231060 markers complete.\r231120 markers complete.\r231180 markers complete.\r231240 markers complete.\r231300 markers complete.\r231360 markers complete.\r231420 markers complete.\r231480 markers complete.\r231540 markers complete.\r231600 markers complete.\r231660 markers complete.\r231720 markers complete.\r231780 markers complete.\r231840 markers complete.\r231900 markers complete.\r231960 markers complete.\r232020 markers complete.\r232080 markers complete.\r232140 markers complete.\r232200 markers complete.\r232260 markers complete.\r232320 markers complete.\r232380 markers complete.\r232440 markers complete.\r232500 markers complete.\r232560 markers complete.\r232620 markers complete.\r232680 markers complete.\r232740 markers complete.\r232800 markers complete.\r232860 markers complete.\r232920 markers complete.\r232980 markers complete.\r233040 markers complete.\r233100 markers complete.\r233160 markers complete.\r233220 markers complete.\r233280 markers complete.\r233340 markers complete.\r233400 markers complete.\r233460 markers complete.\r233520 markers complete.\r233580 markers complete.\r233640 markers complete.\r233700 markers complete.\r233760 markers complete.\r233820 markers complete.\r233880 markers complete.\r233940 markers complete.\r234000 markers complete.\r234060 markers complete.\r234120 markers complete.\r234180 markers complete.\r234240 markers complete.\r234300 markers complete.\r234360 markers complete.\r234420 markers complete.\r234480 markers complete.\r234540 markers complete.\r234600 markers complete.\r234660 markers complete.\r234720 markers complete.\r234780 markers complete.\r234840 markers complete.\r234900 markers complete.\r234960 markers complete.\r235020 markers complete.\r235080 markers complete.\r235140 markers complete.\r235200 markers complete.\r235260 markers complete.\r235320 markers complete.\r235380 markers complete.\r235440 markers complete.\r235500 markers complete.\r235560 markers complete.\r235620 markers complete.\r235680 markers complete.\r235740 markers complete.\r235800 markers complete.\r235860 markers complete.\r235920 markers complete.\r235980 markers complete.\r236040 markers complete.\r236100 markers complete.\r236160 markers complete.\r236220 markers complete.\r236280 markers complete.\r236340 markers complete.\r236400 markers complete.\r236460 markers complete.\r236520 markers complete.\r236580 markers complete.\r236640 markers complete.\r236700 markers complete.\r236760 markers complete.\r236820 markers complete.\r236880 markers complete.\r236940 markers complete.\r237000 markers complete.\r237060 markers complete.\r237120 markers complete.\r237180 markers complete.\r237240 markers complete.\r237300 markers complete.\r237360 markers complete.\r237420 markers complete.\r237480 markers complete.\r237540 markers complete.\r237600 markers complete.\r237660 markers complete.\r237720 markers complete.\r237780 markers complete.\r237840 markers complete.\r237900 markers complete.\r237960 markers complete.\r238020 markers complete.\r238080 markers complete.\r238140 markers complete.\r238200 markers complete.\r238260 markers complete.\r238320 markers complete.\r238380 markers complete.\r238440 markers complete.\r238500 markers complete.\r238560 markers complete.\r238620 markers complete.\r238680 markers complete.\r238740 markers complete.\r238800 markers complete.\r238860 markers complete.\r238920 markers complete.\r238980 markers complete.\r239040 markers complete.\r239100 markers complete.\r239160 markers complete.\r239220 markers complete.\r239280 markers complete.\r239340 markers complete.\r239400 markers complete.\r239460 markers complete.\r239520 markers complete.\r239580 markers complete.\r239640 markers complete.\r239700 markers complete.\r239760 markers complete.\r239820 markers complete.\r239880 markers complete.\r239940 markers complete.\r240000 markers complete.\r240060 markers complete.\r240120 markers complete.\r240180 markers complete.\r240240 markers complete.\r240300 markers complete.\r240360 markers complete.\r240420 markers complete.\r240480 markers complete.\r240540 markers complete.\r240600 markers complete.\r240660 markers complete.\r240720 markers complete.\r240780 markers complete.\r240840 markers complete.\r240900 markers complete.\r240960 markers complete.\r241020 markers complete.\r241080 markers complete.\r241140 markers complete.\r241200 markers complete.\r241260 markers complete.\r241320 markers complete.\r241380 markers complete.\r241440 markers complete.\r241500 markers complete.\r241560 markers complete.\r241620 markers complete.\r241680 markers complete.\r241740 markers complete.\r241800 markers complete.\r241860 markers complete.\r241920 markers complete.\r241980 markers complete.\r242040 markers complete.\r242100 markers complete.\r242160 markers complete.\r242220 markers complete.\r242280 markers complete.\r242340 markers complete.\r242400 markers complete.\r242460 markers complete.\r242520 markers complete.\r242580 markers complete.\r242640 markers complete.\r242700 markers complete.\r242760 markers complete.\r242820 markers complete.\r242880 markers complete.\r242940 markers complete.\r243000 markers complete.\r243060 markers complete.\r243120 markers complete.\r243180 markers complete.\r243240 markers complete.\r243300 markers complete.\r243360 markers complete.\r243420 markers complete.\r243480 markers complete.\r243540 markers complete.\r243600 markers complete.\r243660 markers complete.\r243720 markers complete.\r243780 markers complete.\r243840 markers complete.\r243900 markers complete.\r243960 markers complete.\r244020 markers complete.\r244080 markers complete.\r244140 markers complete.\r244200 markers complete.\r244260 markers complete.\r244320 markers complete.\r244380 markers complete.\r244440 markers complete.\r244500 markers complete.\r244560 markers complete.\r244620 markers complete.\r244680 markers complete.\r244740 markers complete.\r244800 markers complete.\r244860 markers complete.\r244920 markers complete.\r244980 markers complete.\r245040 markers complete.\r245100 markers complete.\r245160 markers complete.\r245220 markers complete.\r245280 markers complete.\r245340 markers complete.\r245400 markers complete.\r245460 markers complete.\r245520 markers complete.\r245580 markers complete.\r245640 markers complete.\r245700 markers complete.\r245760 markers complete.\r245820 markers complete.\r245880 markers complete.\r245940 markers complete.\r246000 markers complete.\r246060 markers complete.\r246120 markers complete.\r246180 markers complete.\r246240 markers complete.\r246300 markers complete.\r246360 markers complete.\r246420 markers complete.\r246480 markers complete.\r246540 markers complete.\r246600 markers complete.\r246660 markers complete.\r246720 markers complete.\r246780 markers complete.\r246840 markers complete.\r246900 markers complete.\r246960 markers complete.\r247020 markers complete.\r247080 markers complete.\r247140 markers complete.\r247200 markers complete.\r247260 markers complete.\r247320 markers complete.\r247380 markers complete.\r247440 markers complete.\r247500 markers complete.\r247560 markers complete.\r247620 markers complete.\r247680 markers complete.\r247740 markers complete.\r247800 markers complete.\r247860 markers complete.\r247920 markers complete.\r247980 markers complete.\r248040 markers complete.\r248100 markers complete.\r248160 markers complete.\r248220 markers complete.\r248280 markers complete.\r248340 markers complete.\r248400 markers complete.\r248460 markers complete.\r248520 markers complete.\r248580 markers complete.\r248640 markers complete.\r248700 markers complete.\r248760 markers complete.\r248820 markers complete.\r248880 markers complete.\r248940 markers complete.\r249000 markers complete.\r249060 markers complete.\r249120 markers complete.\r249180 markers complete.\r249240 markers complete.\r249300 markers complete.\r249360 markers complete.\r249420 markers complete.\r249480 markers complete.\r249540 markers complete.\r249600 markers complete.\r249660 markers complete.\r249720 markers complete.\r249780 markers complete.\r249840 markers complete.\r249900 markers complete.\r249960 markers complete.\r250020 markers complete.\r250080 markers complete.\r250140 markers complete.\r250200 markers complete.\r250260 markers complete.\r250320 markers complete.\r250380 markers complete.\r250440 markers complete.\r250500 markers complete.\r250560 markers complete.\r250620 markers complete.\r250680 markers complete.\r250740 markers complete.\r250800 markers complete.\r250860 markers complete.\r250920 markers complete.\r250980 markers complete.\r251040 markers complete.\r251100 markers complete.\r251160 markers complete.\r251220 markers complete.\r251280 markers complete.\r251340 markers complete.\r251400 markers complete.\r251460 markers complete.\r251520 markers complete.\r251580 markers complete.\r251640 markers complete.\r251700 markers complete.\r251760 markers complete.\r251820 markers complete.\r251880 markers complete.\r251940 markers complete.\r252000 markers complete.\r252060 markers complete.\r252120 markers complete.\r252180 markers complete.\r252240 markers complete.\r252300 markers complete.\r252360 markers complete.\r252420 markers complete.\r252480 markers complete.\r252540 markers complete.\r252600 markers complete.\r252660 markers complete.\r252720 markers complete.\r252780 markers complete.\r252840 markers complete.\r252900 markers complete.\r252960 markers complete.\r253020 markers complete.\r253080 markers complete.\r253140 markers complete.\r253200 markers complete.\r253260 markers complete.\r253320 markers complete.\r253380 markers complete.\r253440 markers complete.\r253500 markers complete.\r253560 markers complete.\r253620 markers complete.\r253680 markers complete.\r253740 markers complete.\r253800 markers complete.\r253860 markers complete.\r253920 markers complete.\r253980 markers complete.\r254040 markers complete.\r254100 markers complete.\r254160 markers complete.\r254220 markers complete.\r254280 markers complete.\r254340 markers complete.\r254400 markers complete.\r254460 markers complete.\r254520 markers complete.\r254580 markers complete.\r254640 markers complete.\r254700 markers complete.\r254760 markers complete.\r254820 markers complete.\r254880 markers complete.\r254940 markers complete.\r255000 markers complete.\r255060 markers complete.\r255120 markers complete.\r255180 markers complete.\r255240 markers complete.\r255300 markers complete.\r255360 markers complete.\r255420 markers complete.\r255480 markers complete.\r255540 markers complete.\r255600 markers complete.\r255660 markers complete.\r255720 markers complete.\r255780 markers complete.\r255840 markers complete.\r255900 markers complete.\r255960 markers complete.\r256020 markers complete.\r256080 markers complete.\r256140 markers complete.\r256200 markers complete.\r256260 markers complete.\r256320 markers complete.\r256380 markers complete.\r256440 markers complete.\r256500 markers complete.\r256560 markers complete.\r256620 markers complete.\r256680 markers complete.\r256740 markers complete.\r256800 markers complete.\r256860 markers complete.\r256920 markers complete.\r256980 markers complete.\r257040 markers complete.\r257100 markers complete.\r257160 markers complete.\r257220 markers complete.\r257280 markers complete.\r257340 markers complete.\r257400 markers complete.\r257460 markers complete.\r257520 markers complete.\r257580 markers complete.\r257640 markers complete.\r257700 markers complete.\r257760 markers complete.\r257820 markers complete.\r257880 markers complete.\r257940 markers complete.\r258000 markers complete.\r258060 markers complete.\r258120 markers complete.\r258180 markers complete.\r258240 markers complete.\r258300 markers complete.\r258360 markers complete.\r258420 markers complete.\r258480 markers complete.\r258540 markers complete.\r258600 markers complete.\r258660 markers complete.\r258720 markers complete.\r258780 markers complete.\r258840 markers complete.\r258900 markers complete.\r258960 markers complete.\r259020 markers complete.\r259080 markers complete.\r259140 markers complete.\r259200 markers complete.\r259260 markers complete.\r259320 markers complete.\r259380 markers complete.\r259440 markers complete.\r259500 markers complete.\r259560 markers complete.\r259620 markers complete.\r259680 markers complete.\r259740 markers complete.\r259800 markers complete.\r259860 markers complete.\r259920 markers complete.\r259980 markers complete.\r260040 markers complete.\r260100 markers complete.\r260160 markers complete.\r260220 markers complete.\r260280 markers complete.\r260340 markers complete.\r260400 markers complete.\r260460 markers complete.\r260520 markers complete.\r260580 markers complete.\r260640 markers complete.\r260700 markers complete.\r260760 markers complete.\r260820 markers complete.\r260880 markers complete.\r260940 markers complete.\r261000 markers complete.\r261060 markers complete.\r261120 markers complete.\r261180 markers complete.\r261240 markers complete.\r261300 markers complete.\r261360 markers complete.\r261420 markers complete.\r261480 markers complete.\r261540 markers complete.\r261600 markers complete.\r261660 markers complete.\r261720 markers complete.\r261780 markers complete.\r261840 markers complete.\r261900 markers complete.\r261960 markers complete.\r262020 markers complete.\r262080 markers complete.\r262140 markers complete.\r262200 markers complete.\r262260 markers complete.\r262320 markers complete.\r262380 markers complete.\r262440 markers complete.\r262500 markers complete.\r262560 markers complete.\r262620 markers complete.\r262680 markers complete.\r262740 markers complete.\r262800 markers complete.\r262860 markers complete.\r262920 markers complete.\r262980 markers complete.\r263040 markers complete.\r263100 markers complete.\r263160 markers complete.\r263220 markers complete.\r263280 markers complete.\r263340 markers complete.\r263400 markers complete.\r263460 markers complete.\r263520 markers complete.\r263580 markers complete.\r263640 markers complete.\r263700 markers complete.\r263760 markers complete.\r263820 markers complete.\r263880 markers complete.\r263940 markers complete.\r264000 markers complete.\r264060 markers complete.\r264120 markers complete.\r264180 markers complete.\r264240 markers complete.\r264300 markers complete.\r264360 markers complete.\r264420 markers complete.\r264480 markers complete.\r264540 markers complete.\r264600 markers complete.\r264660 markers complete.\r264720 markers complete.\r264780 markers complete.\r264840 markers complete.\r264900 markers complete.\r264960 markers complete.\r265020 markers complete.\r265080 markers complete.\r265140 markers complete.\r265200 markers complete.\r265260 markers complete.\r265320 markers complete.\r265380 markers complete.\r265440 markers complete.\r265500 markers complete.\r265560 markers complete.\r265620 markers complete.\r265680 markers complete.\r265740 markers complete.\r265800 markers complete.\r265860 markers complete.\r265920 markers complete.\r265980 markers complete.\r266040 markers complete.\r266100 markers complete.\r266160 markers complete.\r266220 markers complete.\r266280 markers complete.\r266340 markers complete.\r266400 markers complete.\r266460 markers complete.\r266520 markers complete.\r266580 markers complete.\r266640 markers complete.\r266700 markers complete.\r266760 markers complete.\r266820 markers complete.\r266880 markers complete.\r266940 markers complete.\r267000 markers complete.\r267060 markers complete.\r267120 markers complete.\r267180 markers complete.\r267240 markers complete.\r267300 markers complete.\r267360 markers complete.\r267420 markers complete.\r267480 markers complete.\r267540 markers complete.\r267600 markers complete.\r267660 markers complete.\r267720 markers complete.\r267780 markers complete.\r267840 markers complete.\r267900 markers complete.\r267960 markers complete.\r268020 markers complete.\r268080 markers complete.\r268140 markers complete.\r268200 markers complete.\r268260 markers complete.\r268320 markers complete.\r268380 markers complete.\r268440 markers complete.\r268500 markers complete.\r268560 markers complete.\r268620 markers complete.\r268680 markers complete.\r268740 markers complete.\r268800 markers complete.\r268860 markers complete.\r268920 markers complete.\r268980 markers complete.\r269040 markers complete.\r269100 markers complete.\r269160 markers complete.\r269220 markers complete.\r269280 markers complete.\r269340 markers complete.\r269400 markers complete.\r269460 markers complete.\r269520 markers complete.\r269580 markers complete.\r269640 markers complete.\r269700 markers complete.\r269760 markers complete.\r269820 markers complete.\r269880 markers complete.\r269940 markers complete.\r270000 markers complete.\r270060 markers complete.\r270120 markers complete.\r270180 markers complete.\r270240 markers complete.\r270300 markers complete.\r270360 markers complete.\r270420 markers complete.\r270480 markers complete.\r270540 markers complete.\r270600 markers complete.\r270660 markers complete.\r270720 markers complete.\r270780 markers complete.\r270840 markers complete.\r270900 markers complete.\r270960 markers complete.\r271020 markers complete.\r271080 markers complete.\r271140 markers complete.\r271200 markers complete.\r271260 markers complete.\r271320 markers complete.\r271380 markers complete.\r271440 markers complete.\r271500 markers complete.\r271560 markers complete.\r271620 markers complete.\r271680 markers complete.\r271740 markers complete.\r271800 markers complete.\r271860 markers complete.\r271920 markers complete.\r271980 markers complete.\r272040 markers complete.\r272100 markers complete.\r272160 markers complete.\r272220 markers complete.\r272280 markers complete.\r272340 markers complete.\r272400 markers complete.\r272460 markers complete.\r272520 markers complete.\r272580 markers complete.\r272640 markers complete.\r272700 markers complete.\r272760 markers complete.\r272810 markers complete.\rRelationship matrix calculation complete.\n103 people excluded by --rel-cutoff.\nRemaining sample IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.rel.id\n.\n--make-bed to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bed\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.bim\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"}]},{"metadata":{},"id":"dominant-sixth","cell_type":"markdown","source":"Lastly, for most of the genotype-based analysis we may want to remove strongly linked SNPs. With the following command, we scan 200 SNPs-long windows, sliding by 25, and remove one SNP from any SNP-pair having an $R^{2}$ > .4\nNext using the extract command we will take only the SNPs in the *.in file."},{"metadata":{"trusted":true},"id":"expected-prior","cell_type":"code","source":"%%bash \n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved --indep-pairwise 200 25 .4 --allow-no-sex --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD","execution_count":87,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.log.\nOptions in effect:\n  --allow-no-sex\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved\n  --indep-pairwise 200 25 .4\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n125459 variants loaded from .bim file.\n61 people (0 males, 0 females, 61 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.nosex\n.\n61 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 61 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.998761.\n125459 variants and 61 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 61 are controls.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r50%\r51%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 85 variants from chromosome 1, leaving 9867.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 59 variants from chromosome 2, leaving 10213.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 36 variants from chromosome 3, leaving 8132.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 27 variants from chromosome 4, leaving 7333.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r97%\r98%\r99%\rPruned 45 variants from chromosome 5, leaving 8017.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 24 variants from chromosome 6, leaving 7849.\n\r1%\r2%\r3%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 41 variants from chromosome 7, leaving 6854.\n\r1%\r2%\r3%\r4%\r5%\r6%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 39 variants from chromosome 8, leaving 6753.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 29 variants from chromosome 9, leaving 5631.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 34 variants from chromosome 10, leaving 6380.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 47 variants from chromosome 11, leaving 6137.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 33 variants from chromosome 12, leaving 6344.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 11 variants from chromosome 13, leaving 4565.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 22 variants from chromosome 14, leaving 4258.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 43 variants from chromosome 15, leaving 4064.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 24 variants from chromosome 16, leaving 4081.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 30 variants from chromosome 17, leaving 3942.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 17 variants from chromosome 18, leaving 3806.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r12%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r55%\r56%\r57%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r83%\r84%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 18 variants from chromosome 19, leaving 2870.\n\r1%\r2%\r3%\r4%\r5%\r6%\r7%\r8%\r9%\r10%\r11%\r11%\r12%\r13%\r14%\r15%\r16%\r17%\r18%\r19%\r20%\r21%\r22%\r23%\r24%\r25%\r26%\r27%\r28%\r29%\r30%\r31%\r32%\r33%\r34%\r35%\r36%\r37%\r38%\r39%\r40%\r41%\r42%\r42%\r43%\r44%\r45%\r46%\r47%\r48%\r49%\r50%\r51%\r52%\r53%\r54%\r54%\r55%\r56%\r57%\r58%\r59%\r60%\r61%\r62%\r63%\r64%\r65%\r66%\r67%\r68%\r69%\r70%\r71%\r72%\r73%\r74%\r75%\r76%\r77%\r78%\r79%\r80%\r81%\r82%\r83%\r84%\r85%\r85%\r86%\r87%\r88%\r89%\r90%\r91%\r92%\r93%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 20 variants from chromosome 20, leaving 3527.\n\r1%\r2%\r3%\r4%\r6%\r7%\r8%\r9%\r11%\r12%\r13%\r14%\r16%\r17%\r18%\r19%\r20%\r22%\r23%\r24%\r25%\r27%\r28%\r29%\r30%\r32%\r33%\r34%\r35%\r37%\r38%\r39%\r40%\r41%\r43%\r44%\r45%\r46%\r48%\r49%\r50%\r51%\r53%\r54%\r55%\r56%\r57%\r59%\r60%\r61%\r62%\r64%\r65%\r66%\r67%\r69%\r70%\r71%\r72%\r74%\r75%\r76%\r77%\r78%\r80%\r81%\r82%\r83%\r85%\r86%\r87%\r88%\r90%\r91%\r92%\r93%\r95%\r96%\r97%\r98%\r99%\rPruned 5 variants from chromosome 21, leaving 2021.\n\r1%\r2%\r3%\r4%\r5%\r7%\r8%\r9%\r10%\r11%\r12%\r14%\r15%\r16%\r17%\r18%\r19%\r21%\r22%\r23%\r24%\r25%\r27%\r28%\r29%\r30%\r31%\r32%\r34%\r35%\r36%\r37%\r38%\r39%\r41%\r42%\r43%\r44%\r45%\r47%\r48%\r49%\r50%\r51%\r52%\r54%\r55%\r56%\r57%\r58%\r59%\r61%\r62%\r63%\r64%\r65%\r67%\r68%\r69%\r70%\r71%\r72%\r74%\r75%\r76%\r77%\r78%\r79%\r81%\r82%\r83%\r84%\r85%\r87%\r88%\r89%\r90%\r91%\r92%\r94%\r95%\r96%\r97%\r98%\r99%\rPruned 18 variants from chromosome 22, leaving 2108.\nPruning complete.  707 of 125459 variants removed.\nWriting...\rMarker lists written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.in\nand\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.out\n.\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"representative-design","cell_type":"code","source":"%%bash \n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved --extract GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.in --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD ","execution_count":89,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemoved\n  --extract GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.prune.in\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD\n\n29441 MB RAM detected; reserving 14720 MB for main workspace.\n125459 variants loaded from .bim file.\n61 people (0 males, 0 females, 61 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.nosex\n.\n61 phenotype values loaded from .fam.\n--extract: 124752 variants remaining.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 61 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.998763.\n124752 variants and 61 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 61 are controls.\n--make-bed to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{},"id":"settled-corner","cell_type":"markdown","source":"### Great! We now have a cleaned and Quality Controlled file for further analysis"}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.7.10","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat":4,"nbformat_minor":5}